---
title: "hybrids_forpublication"
output: html_document
editor_options: 
  chunk_output_type: inline
---

```{r warning=FALSE}
#Set working dir:
#setwd("/Users/hannahdevens/Desktop/Wray_Lab")
#Load libraries:
library(dplyr)
library(ggplot2)
library(DESeq2)
library(edgeR)
library(apeglm)
library(reshape)
library(cowplot)
library(corrplot)
library(apeglm)
``` 
```{r}
#first i read in the gene transcript files which I got from Dropbox (they were both originally just called "transcript_coords.txt")
Hery_transcript_coords<-read.table("Hery_transcript_coords.txt", header=F)
colnames(Hery_transcript_coords) <- c("HEChr", "start",
                                "end", "orientation", "H_ery_ID")
Hery_transcript_coords2 <- Hery_transcript_coords %>% 
 mutate(start2 = ifelse(orientation == "+", start, end),
        end2 = ifelse(orientation == "+", end, start)) %>% 
 select(HEChr, start2, end2, orientation, H_ery_ID) %>% 
 select(HEChr, start = start2, end = end2, orientation, H_ery_ID)

HEliftedpeakcoordsonly<- read.table("heryandliftedhtubandhyballstages_merged_peaks.narrowPeak")
colnames(HEliftedpeakcoordsonly)<-c("Chr","start","end")
HEliftedpeakcoordsonly$peakno<-rownames(HEliftedpeakcoordsonly)
HEliftedpeakcoordsonly$midpeak <- HEliftedpeakcoordsonly$start +((HEliftedpeakcoordsonly$end-HEliftedpeakcoordsonly$start)/2)
names(HEliftedpeakcoordsonly)[names(HEliftedpeakcoordsonly) == "Chr"] <- "HEChr"

HEliftedpeakcoordsonly_mid<-HEliftedpeakcoordsonly %>% select(peakno, HEChr, midpeak)
Hery_peak_gene_map <- left_join(HEliftedpeakcoordsonly_mid, 
                          Hery_transcript_coords2 %>% select(HEChr, start, H_ery_ID))

Hery_peak_gene_map2 <- Hery_peak_gene_map %>% mutate(peakGeneDist = midpeak-start)
Hery_peak_gene_map_5prime<- Hery_peak_gene_map2 %>% 
 dplyr::filter(peakGeneDist <= 0)
Hery_peak_gene_map_3prime<- Hery_peak_gene_map2 %>% 
 dplyr::filter(peakGeneDist > 0)

Hery_peak_gene_map_5prime_nearest <- Hery_peak_gene_map_5prime %>% 
 group_by(peakno) %>% 
 dplyr::slice(which.max(peakGeneDist))
Hery_peak_gene_map_3prime_nearest <- Hery_peak_gene_map_3prime %>% 
 group_by(peakno) %>% 
 dplyr::slice(which.min(peakGeneDist))

colnames(Hery_peak_gene_map_5prime_nearest) <- c("peak","HEChr_5prime","HEmid","HEstart_5prime",
                                           "H_ery_ID_5prime","HEpeakGeneDist_5prime")
colnames(Hery_peak_gene_map_3prime_nearest) <- c("peak","HEChr_3prime","HEmid","HEstart_3prime",
                                           "H_ery_ID_3prime","HEpeakGeneDist_3prime")

Hery_peak_gene_map_final <- left_join(Hery_peak_gene_map_5prime_nearest,
                                Hery_peak_gene_map_3prime_nearest,
                                by=c("peak","HEmid"))
rm(Hery_peak_gene_map)
rm(Hery_peak_gene_map2)
rm(Hery_peak_gene_map_3prime)
rm(Hery_peak_gene_map_5prime)

Hery_peak_gene_map_final$HEpeakGeneDist_5prime[is.na(Hery_peak_gene_map_final$HEpeakGeneDist_5prime)] <- 9999999
Hery_peak_gene_map_final$HEpeakGeneDist_3prime[is.na(Hery_peak_gene_map_final$HEpeakGeneDist_3prime)] <- 9999999

Hery_nearestgene_bothdir<-for (i in 1:5) {if(
  abs(Hery_peak_gene_map_final[i,6])
  > abs(Hery_peak_gene_map_final[i,10]))
{(Hery_peak_gene_map_final[i,9])} else{(Hery_peak_gene_map_final[i,5])}
}

Hery_nearestgene_bothdir<-data.frame(peak=Hery_peak_gene_map_final$peak,
NearestGene=ifelse(abs(Hery_peak_gene_map_final$HEpeakGeneDist_5prime)>abs(Hery_peak_gene_map_final$HEpeakGeneDist_3prime),
Hery_peak_gene_map_final$H_ery_ID_3prime, Hery_peak_gene_map_final$H_ery_ID_5prime))
#master list of peaks with the nearest 5' and 3' gene in both genomes
```

Set up ase.df, our main table of raw counts
```{r}
ase.df <-read.table("labeled_heryandliftedhtubandhyballstages_multicov.txt", header=TRUE)
ase.df$HybL1<-ase.df$HybHEL1 + ase.df$HybHTL1
ase.df$HybL2<-ase.df$HybHEL2 + ase.df$HybHTL2
ase.df$HybL11<-ase.df$HybHEL11 + ase.df$HybHTL11 
ase.df$HybG7<-ase.df$HybHEG7 + ase.df$HybHTG7 
ase.df$HybG8<-ase.df$HybHEG8 + ase.df$HybHTG8 
ase.df$HybG9<-ase.df$HybHEG9 + ase.df$HybHTG9 
ase.df$HybB12<-ase.df$HybHEB12 + ase.df$HybHTB12 
ase.df$HybB13<-ase.df$HybHEB13 + ase.df$HybHTB13 

```
```{r}
#PCA:
#drop extra samples from htub:
ase.df<-ase.df %>% select(-htubB9, -htubG10, -HeryB1, -htubB1)
#readcounts:
ase.readcounts.for.pca<-as.matrix(ase.df[,c(4:43)])
#drop low counts (<3 counts in >1/3 of samples,only counting combined hybrids and parents for this. so 27 samples and 1/3 of 27 =  9)
ase.readcounts.for.pca.cpm<-cpm(ase.readcounts.for.pca)
ase.readcounts.for.pca.countcheck<-ase.readcounts.for.pca.cpm > 3
ase.readcounts.for.pca.keep<-which(rowSums(ase.readcounts.for.pca.countcheck) >= 9)
ase.readcounts.for.pca <- ase.readcounts.for.pca[ase.readcounts.for.pca.keep,]
dim(ase.readcounts.for.pca.keep)
ase_nolowcounts<-ase.df[ase.readcounts.for.pca.keep,]
ase_allDEseq<-ase_nolowcounts[,4:43]
rownames(ase_allDEseq)<-paste0(ase_nolowcounts$Chr,":",ase_nolowcounts$start, "-",ase_nolowcounts$end)

#read in descriptions file:
descriptions_jan2021<-read.table("descriptions_jan2021.txt", header=T)
descriptions_jan2021 <- descriptions_jan2021 %>% dplyr::filter(!Sample=="HeryB1") %>% dplyr::filter(!Sample=="htubB1")

ase_allDEseq_dds<-DESeqDataSetFromMatrix(ase_allDEseq, descriptions_jan2021, design=~Origin)
ase_allDEseq_dds<-DESeq(ase_allDEseq_dds)
normalized_counts <- counts(ase_allDEseq_dds, normalized=TRUE)

#vst transform: (used vst instead of rlog bc it is faster for large no of samples)
vst.ase.readcounts.for.pca <-vst(round(ase.readcounts.for.pca))
vst.ase.readcounts.for.pca<-as.data.frame(t(vst.ase.readcounts.for.pca)) #transpose
#doing the actual pca analysis:
pca.ase.df.T<-prcomp(vst.ase.readcounts.for.pca, center=T)
pca.ase.df.T<-as.data.frame(pca.ase.df.T$x)

#make plot:
pca.ase.df.plot<-ggplot(pca.ase.df.T, aes(PC1,PC2, colour=descriptions_jan2021$Origin, shape=descriptions_jan2021$Stage))+
        geom_point()+
        geom_text(aes(label=descriptions_jan2021$Sample),hjust=1, vjust=0)
summary(prcomp(vst.ase.readcounts.for.pca, center=T)) #PC1=52% PC2=12%
pca.ase.df.plot #plot for fig. 2C

pca.ase.df.plot23<-ggplot(pca.ase.df.T, aes(PC2,PC3, colour=descriptions_jan2021$Origin, shape=descriptions_jan2021$Stage))+
        geom_point()+
        geom_text(aes(label=descriptions_jan2021$Sample),hjust=1, vjust=0)+
  theme_classic()
summary(prcomp(vst.ase.readcounts.for.pca, center=T)) #PC2=12% PC3=6%
pca.ase.df.plot23 #plot for sfig 2a

pca.ase.df.plot34<-ggplot(pca.ase.df.T, aes(PC3,PC4, colour=descriptions_jan2021$Origin, shape=descriptions_jan2021$Stage))+
        geom_point()+
        geom_text(aes(label=descriptions_jan2021$Sample),hjust=1, vjust=0)+
  theme_classic()
summary(prcomp(vst.ase.readcounts.for.pca, center=T)) #PC3=6% PC4=4%
pca.ase.df.plot34 #plot for sfig 2b
```
Counts for venn diagram (Fig 2b)
```{r}
venn_ase_nolowcounts<-ase_nolowcounts
venn_ase_nolowcounts$HEtot<-rowSums(venn_ase_nolowcounts %>% select(HeryB4,HeryB7,HeryG2,HeryG5,HeryG8,HeryL3,HeryL6,HeryL9))
venn_ase_nolowcounts$HTtot<-rowSums(venn_ase_nolowcounts %>% select(htubB4,htubB7,htubG2,htubG5,htubG8,htubL3,htubL6,htubL11))
venn_ase_nolowcounts$Hybtot<-rowSums(venn_ase_nolowcounts %>% select(HybB12,HybB13,HybG7,HybG8,HybG9,HybL1,HybL2,HybL11))
venn_ase_nolowcounts<-venn_ase_nolowcounts %>% select(Chr,start,end,HEtot,HTtot,Hybtot)
dim(venn_ase_nolowcounts)
dim(venn_ase_nolowcounts %>% dplyr::filter(HEtot != 0))
dim(venn_ase_nolowcounts %>% dplyr::filter(HTtot != 0))
dim(venn_ase_nolowcounts %>% dplyr::filter(Hybtot != 0))
dim(venn_ase_nolowcounts %>% dplyr::filter(HEtot != 0 & HTtot==0 & Hybtot==0))
dim(venn_ase_nolowcounts %>% dplyr::filter(HTtot != 0 & HEtot==0 & Hybtot==0))
dim(venn_ase_nolowcounts %>% dplyr::filter(HEtot != 0 & HTtot!=0 & Hybtot==0))
dim(venn_ase_nolowcounts %>% dplyr::filter(HEtot != 0 & HTtot==0 & Hybtot!=0))
dim(venn_ase_nolowcounts %>% dplyr::filter(HEtot == 0 & HTtot!=0 & Hybtot!=0))
dim(venn_ase_nolowcounts %>% dplyr::filter(HEtot == 0 & HTtot==0 & Hybtot!=0))
dim(venn_ase_nolowcounts %>% dplyr::filter(HEtot != 0 & HTtot!=0 & Hybtot!=0))
```

Create DESeq objects for parents vs. hybrids
```{r}
ase_nolowcounts<-ase.df[ase.readcounts.for.pca.keep,]
ase_pHEvsHyball_readcounts_Bonly<-ase_nolowcounts %>% select (HeryB4,HeryB7,HybB12,HybB13)
rownames(ase_pHEvsHyball_readcounts_Bonly)<-paste0(ase_nolowcounts$Chr,":",ase_nolowcounts$start, "-",ase_nolowcounts$end)
ase_pHEvsHyball_readcounts_Bonly_desc<-data.frame(Sample=c("HeryB4","HeryB7","HybB12","HybB13"),
  Origin=c("Hery","Hery","Hyb","Hyb"))
ase_pHEvsHyball_readcounts_Bonly_desc$Origin<-as.factor(ase_pHEvsHyball_readcounts_Bonly_desc$Origin)
ase_pHEvsHyball_readcounts_Bonly_dds<-DESeqDataSetFromMatrix(ase_pHEvsHyball_readcounts_Bonly, ase_pHEvsHyball_readcounts_Bonly_desc, design=~Origin)
ase_pHEvsHyball_readcounts_Bonly_dds<-DESeq(ase_pHEvsHyball_readcounts_Bonly_dds)
resultsNames(ase_pHEvsHyball_readcounts_Bonly_dds)
ase_pHEvsHyball_readcounts_Bonly_res<-results(ase_pHEvsHyball_readcounts_Bonly_dds, name="Origin_Hyb_vs_Hery")
ase_pHEvsHyball_readcounts_Bonly_res<-as.data.frame(ase_pHEvsHyball_readcounts_Bonly_res)
#negative log2FC means higher in Hery than Hyb

#lifted pHT vs liftedHyball:
ase_pHTvsHyball_readcounts_Bonly<-ase_nolowcounts%>% select (htubB4,htubB7,HybB12,HybB13)
rownames(ase_pHTvsHyball_readcounts_Bonly)<-paste0(ase_nolowcounts$Chr,":",ase_nolowcounts$start, "-",ase_nolowcounts$end)
ase_pHTvsHyball_readcounts_Bonly_desc<-data.frame(Sample=c("htubB4","htubB7","HybB12","HybB13"),
  Origin=c("Htub","Htub","Hyb","Hyb"))
ase_pHTvsHyball_readcounts_Bonly_desc$Origin<-as.factor(ase_pHTvsHyball_readcounts_Bonly_desc$Origin)
ase_pHTvsHyball_readcounts_Bonly_dds<-DESeqDataSetFromMatrix(ase_pHTvsHyball_readcounts_Bonly, ase_pHTvsHyball_readcounts_Bonly_desc, design=~Origin)
ase_pHTvsHyball_readcounts_Bonly_dds<-DESeq(ase_pHTvsHyball_readcounts_Bonly_dds)
resultsNames(ase_pHTvsHyball_readcounts_Bonly_dds)
ase_pHTvsHyball_readcounts_Bonly_res<-results(ase_pHTvsHyball_readcounts_Bonly_dds, name="Origin_Hyb_vs_Htub")
ase_pHTvsHyball_readcounts_Bonly_res<-as.data.frame(ase_pHTvsHyball_readcounts_Bonly_res)

#lifted parents against each other
ase_pHEvspHT_readcounts_Bonly<-ase_nolowcounts %>% select (HeryB4,HeryB7,htubB4,htubB7)
rownames(ase_pHEvspHT_readcounts_Bonly)<-paste0(ase_nolowcounts$Chr,":",ase_nolowcounts$start, "-",ase_nolowcounts$end)
ase_pHEvspHT_readcounts_Bonly_desc<-data.frame(Sample=c("heryB4","heryB7","htubB4","htubB7"),
  Origin=c("Hery","Hery","Htub","Htub"))
ase_pHEvspHT_readcounts_Bonly_desc$Origin<-as.factor(ase_pHEvspHT_readcounts_Bonly_desc$Origin)
ase_pHEvspHT_readcounts_Bonly_dds<-DESeqDataSetFromMatrix(ase_pHEvspHT_readcounts_Bonly, ase_pHEvspHT_readcounts_Bonly_desc, design=~Origin)
ase_pHEvspHT_readcounts_Bonly_dds<-DESeq(ase_pHEvspHT_readcounts_Bonly_dds)
resultsNames(ase_pHEvspHT_readcounts_Bonly_dds)
ase_pHEvspHT_readcounts_Bonly_res<-results(ase_pHEvspHT_readcounts_Bonly_dds,name="Origin_Htub_vs_Hery")
ase_pHEvspHT_readcounts_Bonly_res<-as.data.frame(ase_pHEvspHT_readcounts_Bonly_res)
```

#INHERITANCE TESTS BASED ON ABOVE DESEQs
```{r}
ase_nolowcounts.lfcShrink_Bonly_Hyb_vs_Hery_res<-lfcShrink(ase_pHEvsHyball_readcounts_Bonly_dds, coef="Origin_Hyb_vs_Hery") #lifted hybs vs lifted pHE, w/normalization, negative L2FC means higher in Hery than Hyb
ase_nolowcounts.lfcShrink_Bonly_Hyb_vs_Htub_res<-lfcShrink(ase_pHTvsHyball_readcounts_Bonly_dds, coef="Origin_Hyb_vs_Htub")#lifted hybs vs lifted pHT, w/normalization, negative L2FC means higher in htub than Hyb
ase_nolowcounts.lfcShrink_Bonly_Htub_vs_Hery_res<-lfcShrink(ase_pHEvspHT_readcounts_Bonly_dds,coef="Origin_Htub_vs_Hery") #lifted pHtub vs lifted pHery, w/normalization, negative L2FC means higher in Hery than htub

#replacing NAs with 1s 
ase_nolowcounts.lfcShrink_Bonly_Hyb_vs_Hery_res<-as.data.frame(ase_nolowcounts.lfcShrink_Bonly_Hyb_vs_Hery_res)
ase_nolowcounts.lfcShrink_Bonly_Hyb_vs_Hery_res[is.na(ase_nolowcounts.lfcShrink_Bonly_Hyb_vs_Hery_res)]<-1
ase_nolowcounts.lfcShrink_Bonly_Hyb_vs_Htub_res<-as.data.frame(ase_nolowcounts.lfcShrink_Bonly_Hyb_vs_Htub_res)
ase_nolowcounts.lfcShrink_Bonly_Hyb_vs_Htub_res[is.na(ase_nolowcounts.lfcShrink_Bonly_Hyb_vs_Htub_res)]<-1
ase_nolowcounts.lfcShrink_Bonly_Htub_vs_Hery_res<-as.data.frame(ase_nolowcounts.lfcShrink_Bonly_Htub_vs_Hery_res)
ase_nolowcounts.lfcShrink_Bonly_Htub_vs_Hery_res[is.na(ase_nolowcounts.lfcShrink_Bonly_Htub_vs_Hery_res)]<-1

ase_nolowcounts.inh.hyb.vs.HE.B<-ase_nolowcounts.lfcShrink_Bonly_Hyb_vs_Hery_res %>% select(log2FChyb_pHE=log2FoldChange,HybvspHE_padj=padj)
ase_nolowcounts.inh.hyb.vs.HT.B<-ase_nolowcounts.lfcShrink_Bonly_Hyb_vs_Htub_res %>% select(log2FChyb_pHT=log2FoldChange,HybvspHT_padj=padj)
ase_nolowcounts.inh.HT.vs.HE.B<-ase_nolowcounts.lfcShrink_Bonly_Htub_vs_Hery_res %>% select(log2FCpHT_pHE=log2FoldChange, pHTvspHE_padj=padj)
ase_nolowcounts.inh.B.all<-cbind(ase_nolowcounts.inh.hyb.vs.HE.B,ase_nolowcounts.inh.hyb.vs.HT.B,ase_nolowcounts.inh.HT.vs.HE.B)
#remember that for log2FChyb_hybHE, negative means higher in Hery; for log2FChyb_pHT, negative means higher in htub; for htub vs Hery, negative means higher in Hery

#conserved are when Ht = He; Hy = He; Hy = Ht. ie, pHe not sig diff from pHT; HybTOT not sig diff from pHE; HybTOT not sig diff from pHT
ase_nolowcounts.B_conserved<-ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0.05 & HybvspHT_padj>0.05)

#underdominant are when (Ht = He ; Hy < He ; Hy < Ht) OR 
ase_nolowcounts.B_underdom<-ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE <(0) & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0))
#(Ht < He ; Hy < He ; Hy < Ht underdominant) OR 
ase_nolowcounts.B_underdom<-rbind(ase_nolowcounts.B_underdom, ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE<(0) & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE <(0) & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0)))
#(Ht > He ; Hy < He ; Hy < Ht). 
ase_nolowcounts.B_underdom<-rbind(ase_nolowcounts.B_underdom, ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE>0 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE <(0) & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0)))

#overdominant are when (Ht = He ; Hy > He ; Hy > Ht) OR 
ase_nolowcounts.B_overdom<-ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT>0)
#(Ht < He ; Hy > He ; Hy > Ht) OR 
ase_nolowcounts.B_overdom<-rbind(ase_nolowcounts.B_overdom, ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE<0 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT>0))
#(Ht > He ; Hy > He ; Hy > Ht) 
ase_nolowcounts.B_overdom<-rbind(ase_nolowcounts.B_overdom, ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE>0 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT>0))

#HEdominant are when (Ht < He ; Hy = He ; Hy > Ht) OR 
ase_nolowcounts.B_HEdom<-ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE<(0) & HybvspHE_padj > 0.05 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT>0)
#(Ht > He ; Hy = He ; Hy < Ht)
ase_nolowcounts.B_HEdom<-rbind(ase_nolowcounts.B_HEdom, ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE>0 & HybvspHE_padj > 0.05 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0)))

#HTdominant are when (Ht < He ;Hy < He; Hy = Ht) OR 
ase_nolowcounts.B_HTdom<-ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE<(0) & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE<(0) & HybvspHT_padj>0.05)
#(Ht > He ; Hy > He ; Hy = Ht)
ase_nolowcounts.B_HTdom<-rbind(ase_nolowcounts.B_HTdom, ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE>0 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0.05))

#additive are when (Ht < He ; Hy < He ; Hy > Ht) 
ase_nolowcounts.B_additive<-ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE<(0) & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE<(0) & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT>0)
#or (Ht > He ; Hy > He ; Hy < Ht)
ase_nolowcounts.B_additive<-rbind(ase_nolowcounts.B_additive, ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE>0 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0)))

#ambiguous are when 
#(Ht = He ; Hy = He ; Hy < Ht) OR
ase_nolowcounts.B_ambig<-ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj > 0.05 & HybvspHT_padj >0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0))
#(Ht = He ; Hy = He ; Hy > Ht) OR
ase_nolowcounts.B_ambig<-rbind(ase_nolowcounts.B_ambig, ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj > 0.05 & HybvspHT_padj >0 & HybvspHT_padj<=0.05 & log2FChyb_pHT>0))
#(Ht = He ; Hy < He ; Hy = Ht) OR
ase_nolowcounts.B_ambig<-rbind(ase_nolowcounts.B_ambig, ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE<(0) & HybvspHT_padj>0.05))
#(Ht = He ; Hy > He ; Hy = Ht) OR
ase_nolowcounts.B_ambig<-rbind(ase_nolowcounts.B_ambig, ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0.05))
#(Ht = He ; Hy < He ; Hy > Ht) OR
ase_nolowcounts.B_ambig<-rbind(ase_nolowcounts.B_ambig, ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE<(0) & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT>0))
#(Ht = He ; Hy > He ; Hy < Ht) OR
ase_nolowcounts.B_ambig<-rbind(ase_nolowcounts.B_ambig, ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT <(0)))
#(Ht < He ; Hy = He ; Hy = Ht) OR
ase_nolowcounts.B_ambig<-rbind(ase_nolowcounts.B_ambig, ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj <=0.05 & log2FCpHT_pHE <(0) & HybvspHE_padj > 0.05 & HybvspHT_padj >0.05))
#(Ht > He ; Hy = He ; Hy = Ht) OR
ase_nolowcounts.B_ambig<-rbind(ase_nolowcounts.B_ambig, ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj <=0.05 & log2FCpHT_pHE >0 & HybvspHE_padj > 0.05 & HybvspHT_padj>0.05))
#(Ht < He ; Hy = He ; Hy < Ht) OR
ase_nolowcounts.B_ambig<-rbind(ase_nolowcounts.B_ambig, ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj <=0.05 & log2FCpHT_pHE <(0) & HybvspHE_padj > 0.05 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT <(0)))
#(Ht > He ; Hy = He ; Hy > Ht) OR
ase_nolowcounts.B_ambig<-rbind(ase_nolowcounts.B_ambig, ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj <=0.05 & log2FCpHT_pHE >0 & HybvspHE_padj > 0.05 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT >0))
#(Ht < He ; Hy > He ; Hy = Ht) OR
ase_nolowcounts.B_ambig<-rbind(ase_nolowcounts.B_ambig, ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj <=0.05 & log2FCpHT_pHE <(0) & HybvspHE_padj>0 &HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0.05))
#(Ht > He ; Hy < He ; Hy = Ht) OR
ase_nolowcounts.B_ambig<-rbind(ase_nolowcounts.B_ambig, ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj <=0.05 & log2FCpHT_pHE >0 & HybvspHE_padj>0 &HybvspHE_padj<=0.05 & log2FChyb_pHE<(0) & HybvspHT_padj>0.05))
#(Ht < He ; Hy > He ; Hy < Ht) OR
ase_nolowcounts.B_ambig<-rbind(ase_nolowcounts.B_ambig, ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj <=0.05 & log2FCpHT_pHE <(0) & HybvspHE_padj>0 &HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0)))
#(Ht > He ; Hy < He ; Hy > Ht)
ase_nolowcounts.B_ambig<-rbind(ase_nolowcounts.B_ambig,ase_nolowcounts.inh.B.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE > 2 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 &log2FChyb_pHE<(0) & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT >0))

ase_nolowcounts.inh.test.res<-data.frame(B=as.numeric(c(nrow(ase_nolowcounts.B_conserved),nrow(ase_nolowcounts.B_underdom),nrow(ase_nolowcounts.B_overdom),nrow(ase_nolowcounts.B_HEdom),nrow(ase_nolowcounts.B_HTdom),nrow(ase_nolowcounts.B_additive),nrow(ase_nolowcounts.B_ambig))))
rownames(ase_nolowcounts.inh.test.res)<-c("Conserved","Underdominant","Overdominant","HE dominant", "HT dominant","Additive","Ambiguous")
```

```{r}
#GASTRULA DE SEQ + INHERITANCE TESTS

#lifted pHE vs liftedHyball:
ase_pHEvsHyball_readcounts_Gonly<-ase_nolowcounts %>% select(HeryG2,HeryG5,HeryG8,HybG7,HybG8,HybG9)
rownames(ase_pHEvsHyball_readcounts_Gonly)<-paste0(ase_nolowcounts$Chr,":",ase_nolowcounts$start, "-",ase_nolowcounts$end)
ase_pHEvsHyball_readcounts_Gonly_desc<-data.frame(Sample=c("HeryG2","HeryG5","HeryG8","HybG7","HybG8","HybG9"),
  Origin=c("Hery","Hery","Hery","Hyb","Hyb","Hyb"))
ase_pHEvsHyball_readcounts_Gonly_desc$Origin<-as.factor(ase_pHEvsHyball_readcounts_Gonly_desc$Origin)
ase_pHEvsHyball_readcounts_Gonly_dds<-DESeqDataSetFromMatrix(ase_pHEvsHyball_readcounts_Gonly, ase_pHEvsHyball_readcounts_Gonly_desc, design=~Origin)
ase_pHEvsHyball_readcounts_Gonly_dds<-DESeq(ase_pHEvsHyball_readcounts_Gonly_dds)
resultsNames(ase_pHEvsHyball_readcounts_Gonly_dds)
ase_pHEvsHyball_readcounts_Gonly_res<-results(ase_pHEvsHyball_readcounts_Gonly_dds,name="Origin_Hyb_vs_Hery")
ase_pHEvsHyball_readcounts_Gonly_res<-as.data.frame(ase_pHEvsHyball_readcounts_Gonly_res)

#lifted pHT vs liftedHyball:
ase_pHTvsHyball_readcounts_Gonly<-ase_nolowcounts%>% select(htubG2,htubG5,htubG8,HybG7, HybG8,HybG9)
rownames(ase_pHTvsHyball_readcounts_Gonly)<-paste0(ase_nolowcounts$Chr,":",ase_nolowcounts$start, "-",ase_nolowcounts$end)
ase_pHTvsHyball_readcounts_Gonly_desc<-data.frame(Sample=c("htubG2","htubG5","htubG8","HybG7","HybG8","HybG9"),
  Origin=c("Htub","Htub","Htub","Hyb","Hyb","Hyb"))
ase_pHTvsHyball_readcounts_Gonly_desc$Origin<-as.factor(ase_pHTvsHyball_readcounts_Gonly_desc$Origin)
ase_pHTvsHyball_readcounts_Gonly_dds<-DESeqDataSetFromMatrix(ase_pHTvsHyball_readcounts_Gonly, ase_pHTvsHyball_readcounts_Gonly_desc, design=~Origin)
ase_pHTvsHyball_readcounts_Gonly_dds<-DESeq(ase_pHTvsHyball_readcounts_Gonly_dds)
resultsNames(ase_pHTvsHyball_readcounts_Gonly_dds)
ase_pHTvsHyball_readcounts_Gonly_res<-results(ase_pHTvsHyball_readcounts_Gonly_dds,name="Origin_Hyb_vs_Htub")
ase_pHTvsHyball_readcounts_Gonly_res<-as.data.frame(ase_pHTvsHyball_readcounts_Gonly_res)

#lifted parents against each other
ase_pHEvspHT_readcounts_Gonly<-ase_nolowcounts %>% select (HeryG2, HeryG5,HeryG8,htubG2,htubG5,htubG5,htubG8)
rownames(ase_pHEvspHT_readcounts_Gonly)<-paste0(ase_nolowcounts$Chr,":",ase_nolowcounts$start, "-",ase_nolowcounts$end)
ase_pHEvspHT_readcounts_Gonly_desc<-data.frame(Sample=c("HeryG2","HeryG5", "HeryG8","htubG2","htubG5","htubG8"),
  Origin=c("Hery","Hery","Hery","Htub","Htub","Htub"))
ase_pHEvspHT_readcounts_Gonly_desc$Origin<-as.factor(ase_pHEvspHT_readcounts_Gonly_desc$Origin)
ase_pHEvspHT_readcounts_Gonly_dds<-DESeqDataSetFromMatrix(ase_pHEvspHT_readcounts_Gonly, ase_pHEvspHT_readcounts_Gonly_desc, design=~Origin)
ase_pHEvspHT_readcounts_Gonly_dds<-DESeq(ase_pHEvspHT_readcounts_Gonly_dds)
resultsNames(ase_pHEvspHT_readcounts_Gonly_dds)
ase_pHEvspHT_readcounts_Gonly_res<-results(ase_pHEvspHT_readcounts_Gonly_dds,name="Origin_Htub_vs_Hery")
ase_pHEvspHT_readcounts_Gonly_res<-as.data.frame(ase_pHEvspHT_readcounts_Gonly_res)
#negative means higher in pHE than pHT

#INHERITANCE TESTS BASED ON ABOVE DESEQs

ase_nolowcounts.lfcShrink_Gonly_Hyb_vs_Hery_res<-lfcShrink(ase_pHEvsHyball_readcounts_Gonly_dds, coef="Origin_Hyb_vs_Hery") #lifted hybs vs lifted pHE, w/normalization, negative L2FC means higher in Hery than Hyb
ase_nolowcounts.lfcShrink_Gonly_Hyb_vs_Htub_res<-lfcShrink(ase_pHTvsHyball_readcounts_Gonly_dds, coef="Origin_Hyb_vs_Htub")#lifted hybs vs lifted pHT, w/normalization, negative L2FC means higher in htub than Hyb
ase_nolowcounts.lfcShrink_Gonly_Htub_vs_Hery_res<-lfcShrink(ase_pHEvspHT_readcounts_Gonly_dds,coef="Origin_Htub_vs_Hery") #lifted pHtub vs lifted pHery, w/normalization, negative L2FC means higher in Hery than htub

#replacing NAs with 1s (i think this is ok for now as long as i remember to require padj to be <=.1  and >0)
ase_nolowcounts.lfcShrink_Gonly_Hyb_vs_Hery_res<-as.data.frame(ase_nolowcounts.lfcShrink_Gonly_Hyb_vs_Hery_res)
ase_nolowcounts.lfcShrink_Gonly_Hyb_vs_Hery_res[is.na(ase_nolowcounts.lfcShrink_Gonly_Hyb_vs_Hery_res)]<-1
ase_nolowcounts.lfcShrink_Gonly_Hyb_vs_Htub_res<-as.data.frame(ase_nolowcounts.lfcShrink_Gonly_Hyb_vs_Htub_res)
ase_nolowcounts.lfcShrink_Gonly_Hyb_vs_Htub_res[is.na(ase_nolowcounts.lfcShrink_Gonly_Hyb_vs_Htub_res)]<-1
ase_nolowcounts.lfcShrink_Gonly_Htub_vs_Hery_res<-as.data.frame(ase_nolowcounts.lfcShrink_Gonly_Htub_vs_Hery_res)
ase_nolowcounts.lfcShrink_Gonly_Htub_vs_Hery_res[is.na(ase_nolowcounts.lfcShrink_Gonly_Htub_vs_Hery_res)]<-1

ase_nolowcounts.inh.hyb.vs.HE.G<-ase_nolowcounts.lfcShrink_Gonly_Hyb_vs_Hery_res %>% select(log2FChyb_pHE=log2FoldChange,HybvspHE_padj=padj)
ase_nolowcounts.inh.hyb.vs.HT.G<-ase_nolowcounts.lfcShrink_Gonly_Hyb_vs_Htub_res %>% select(log2FChyb_pHT=log2FoldChange,HybvspHT_padj=padj)
ase_nolowcounts.inh.HT.vs.HE.G<-ase_nolowcounts.lfcShrink_Gonly_Htub_vs_Hery_res %>% select(log2FCpHT_pHE=log2FoldChange, pHTvspHE_padj=padj)
ase_nolowcounts.inh.G.all<-cbind(ase_nolowcounts.inh.hyb.vs.HE.G,ase_nolowcounts.inh.hyb.vs.HT.G,ase_nolowcounts.inh.HT.vs.HE.G)
 #remember that for log2FChyb_hybHE, negative means higher in Hery; for log2FChyb_pHT, negative means higher in htub; for htub vs Hery, negative means higher in Hery

#conserved are when Ht = He; Hy = He; Hy = Ht. ie, pHe not sig diff from pHT; HybTOT not sig diff from pHE; HybTOT not sig diff from pHT
ase_nolowcounts.G_conserved<-ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0.05 & HybvspHT_padj>0.05)

#underdominant are when (Ht = He ; Hy < He ; Hy < Ht) OR 
ase_nolowcounts.G_underdom<-ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE <(0) & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0))
#(Ht < He ; Hy < He ; Hy < Ht underdominant) OR 
ase_nolowcounts.G_underdom<-rbind(ase_nolowcounts.G_underdom, ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE<(0) & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE <(0) & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0)))
#(Ht > He ; Hy < He ; Hy < Ht). 
ase_nolowcounts.G_underdom<-rbind(ase_nolowcounts.G_underdom, ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE>0 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE <(0) & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0)))

#underdominant are when (Ht = He ; Hy < He ; Hy < Ht) OR 
ase_nolowcounts.G_underdom<-ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE <(0) & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0))
#(Ht < He ; Hy < He ; Hy < Ht underdominant) OR 
ase_nolowcounts.G_underdom<-rbind(ase_nolowcounts.G_underdom, ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE<(0) & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE <(0) & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0)))
#(Ht > He ; Hy < He ; Hy < Ht). 
ase_nolowcounts.G_underdom<-rbind(ase_nolowcounts.G_underdom, ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE>0 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE <(0) & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0)))

#overdominant are when (Ht = He ; Hy > He ; Hy > Ht) OR 
ase_nolowcounts.G_overdom<-ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT>0)
#(Ht < He ; Hy > He ; Hy > Ht) OR 
ase_nolowcounts.G_overdom<-rbind(ase_nolowcounts.G_overdom, ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE<(0) & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT>0))
#(Ht > He ; Hy > He ; Hy > Ht) 
ase_nolowcounts.G_overdom<-rbind(ase_nolowcounts.G_overdom, ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE>0 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT>0))

#HEdominant are when (Ht < He ; Hy = He ; Hy > Ht) OR 
ase_nolowcounts.G_HEdom<-ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE<(0) & HybvspHE_padj > 0.05 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT>0)
#(Ht > He ; Hy = He ; Hy < Ht)
ase_nolowcounts.G_HEdom<-rbind(ase_nolowcounts.G_HEdom, ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE>0 & HybvspHE_padj > 0.05 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0)))

#HTdominant are when (Ht < He ;Hy < He; Hy = Ht) OR 
ase_nolowcounts.G_HTdom<-ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE<(0) & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE<(0) & HybvspHT_padj>0.05)
#(Ht > He ; Hy > He ; Hy = Ht)
ase_nolowcounts.G_HTdom<-rbind(ase_nolowcounts.G_HTdom, ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE>0 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0.05))

#additive are when (Ht < He ; Hy < He ; Hy > Ht) 
ase_nolowcounts.G_additive<-ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE<(0) & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE<(0) & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT>0)
#or (Ht > He ; Hy > He ; Hy < Ht)
ase_nolowcounts.G_additive<-rbind(ase_nolowcounts.G_additive, ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE>0 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0)))

#ambiguous are when 
#(Ht = He ; Hy = He ; Hy < Ht) OR
ase_nolowcounts.G_ambig<-ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj > 0.05 & HybvspHT_padj >0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0))
#(Ht = He ; Hy = He ; Hy > Ht) OR
ase_nolowcounts.G_ambig<-rbind(ase_nolowcounts.G_ambig, ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj > 0.05 & HybvspHT_padj >0 & HybvspHT_padj<=0.05 & log2FChyb_pHT>0))
#(Ht = He ; Hy < He ; Hy = Ht) OR
ase_nolowcounts.G_ambig<-rbind(ase_nolowcounts.G_ambig, ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE<(0) & HybvspHT_padj>0.05))
#(Ht = He ; Hy > He ; Hy = Ht) OR
ase_nolowcounts.G_ambig<-rbind(ase_nolowcounts.G_ambig, ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0.05))
#(Ht = He ; Hy < He ; Hy > Ht) OR
ase_nolowcounts.G_ambig<-rbind(ase_nolowcounts.G_ambig, ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE<(0) & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT>0))
#(Ht = He ; Hy > He ; Hy < Ht) OR
ase_nolowcounts.G_ambig<-rbind(ase_nolowcounts.G_ambig, ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT <(0)))
#(Ht < He ; Hy = He ; Hy = Ht) OR
ase_nolowcounts.G_ambig<-rbind(ase_nolowcounts.G_ambig, ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj <=0.05 & log2FCpHT_pHE <(0) & HybvspHE_padj > 0.05 & HybvspHT_padj >0.05))
#(Ht > He ; Hy = He ; Hy = Ht) OR
ase_nolowcounts.G_ambig<-rbind(ase_nolowcounts.G_ambig, ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj <=0.05 & log2FCpHT_pHE >0 & HybvspHE_padj > 0.05 & HybvspHT_padj>0.05))
#(Ht < He ; Hy = He ; Hy < Ht) OR
ase_nolowcounts.G_ambig<-rbind(ase_nolowcounts.G_ambig, ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj <=0.05 & log2FCpHT_pHE <(0) & HybvspHE_padj > 0.05 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT <(0)))
#(Ht > He ; Hy = He ; Hy > Ht) OR
ase_nolowcounts.G_ambig<-rbind(ase_nolowcounts.G_ambig, ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj <=0.05 & log2FCpHT_pHE >0 & HybvspHE_padj > 0.05 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT >0))
#(Ht < He ; Hy > He ; Hy = Ht) OR
ase_nolowcounts.G_ambig<-rbind(ase_nolowcounts.G_ambig, ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj <=0.05 & log2FCpHT_pHE <0 & HybvspHE_padj>0 &HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0.05))
#(Ht > He ; Hy < He ; Hy = Ht) OR
ase_nolowcounts.G_ambig<-rbind(ase_nolowcounts.G_ambig, ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj <=0.05 & log2FCpHT_pHE >0 & HybvspHE_padj>0 &HybvspHE_padj<=0.05 & log2FChyb_pHE<(0) & HybvspHT_padj>0.05))
#(Ht < He ; Hy > He ; Hy < Ht) OR
ase_nolowcounts.G_ambig<-rbind(ase_nolowcounts.G_ambig, ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj <=0.05 & log2FCpHT_pHE <(0) & HybvspHE_padj>0 &HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0)))
#(Ht > He ; Hy < He ; Hy > Ht)
ase_nolowcounts.G_ambig<-rbind(ase_nolowcounts.G_ambig,ase_nolowcounts.inh.G.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE > 2 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 &log2FChyb_pHE<(0) & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT >0))

ase_nolowcounts.inh.test.res$G<-as.numeric(c(nrow(ase_nolowcounts.G_conserved),nrow(ase_nolowcounts.G_underdom),nrow(ase_nolowcounts.G_overdom),nrow(ase_nolowcounts.G_HEdom),nrow(ase_nolowcounts.G_HTdom),nrow(ase_nolowcounts.G_additive),nrow(ase_nolowcounts.G_ambig)))

```


```{r}
#Larva DE SEQ + INHERITANCE TESTS

#lifted pHE vs liftedHyball:
ase_pHEvsHyball_readcounts_Lonly<-ase_nolowcounts %>% select(HeryL3,HeryL6,HeryL9, HybL1,HybL2,HybL11)
rownames(ase_pHEvsHyball_readcounts_Lonly)<-paste0(ase_nolowcounts$Chr,":",ase_nolowcounts$start, "-",ase_nolowcounts$end)

ase_pHEvsHyball_readcounts_Lonly_desc<-data.frame(Sample=c("HeryL3","HeryL6","HeryL9","HybL1","HybL2","HybL11"),
  Origin=c("Hery","Hery","Hery","Hyb","Hyb","Hyb"))
ase_pHEvsHyball_readcounts_Lonly_desc$Origin<-as.factor(ase_pHEvsHyball_readcounts_Lonly_desc$Origin)
ase_pHEvsHyball_readcounts_Lonly_dds<-DESeqDataSetFromMatrix(ase_pHEvsHyball_readcounts_Lonly, ase_pHEvsHyball_readcounts_Lonly_desc, design=~Origin)
ase_pHEvsHyball_readcounts_Lonly_dds<-DESeq(ase_pHEvsHyball_readcounts_Lonly_dds)
resultsNames(ase_pHEvsHyball_readcounts_Lonly_dds)
ase_pHEvsHyball_readcounts_Lonly_res<-results(ase_pHEvsHyball_readcounts_Lonly_dds,name="Origin_Hyb_vs_Hery")
ase_pHEvsHyball_readcounts_Lonly_res<-as.data.frame(ase_pHEvsHyball_readcounts_Lonly_res)

#lifted pHT vs liftedHyball:
ase_pHTvsHyball_readcounts_Lonly<-ase_nolowcounts%>% select(htubL3,htubL6,htubL11,HybL1,HybL2,HybL11)
rownames(ase_pHTvsHyball_readcounts_Lonly)<-paste0(ase_nolowcounts$Chr,":",ase_nolowcounts$start, "-",ase_nolowcounts$end)

ase_pHTvsHyball_readcounts_Lonly_desc<-data.frame(Sample=c("htubL3","htubL6","htubL11","HybL1","HybL2","HybL11"),
  Origin=c("Htub","Htub","Htub","Hyb","Hyb","Hyb"))
ase_pHTvsHyball_readcounts_Lonly_desc$Origin<-as.factor(ase_pHTvsHyball_readcounts_Lonly_desc$Origin)
ase_pHTvsHyball_readcounts_Lonly_dds<-DESeqDataSetFromMatrix(ase_pHTvsHyball_readcounts_Lonly, ase_pHTvsHyball_readcounts_Lonly_desc, design=~Origin)
ase_pHTvsHyball_readcounts_Lonly_dds<-DESeq(ase_pHTvsHyball_readcounts_Lonly_dds)
resultsNames(ase_pHTvsHyball_readcounts_Lonly_dds)
ase_pHTvsHyball_readcounts_Lonly_res<-results(ase_pHTvsHyball_readcounts_Lonly_dds,name="Origin_Hyb_vs_Htub")
ase_pHTvsHyball_readcounts_Lonly_res<-as.data.frame(ase_pHTvsHyball_readcounts_Lonly_res)
#i think negative log2FC means higher in Htub than Hyb

#lifted parents against each other
ase_pHEvspHT_readcounts_Lonly<-ase_nolowcounts%>% select (HeryL3, HeryL6,HeryL9,htubL3,htubL3,htubL6,htubL11)
rownames(ase_pHEvspHT_readcounts_Lonly)<-paste0(ase_nolowcounts$Chr,":",ase_nolowcounts$start, "-",ase_nolowcounts$end)

ase_pHEvspHT_readcounts_Lonly_desc<-data.frame(Sample=c("HeryL3","HeryL6", "HeryL9","htubL3","htubL6","htubL11"),
  Origin=c("Hery","Hery","Hery","Htub","Htub","Htub"))
ase_pHEvspHT_readcounts_Lonly_desc$Origin<-as.factor(ase_pHEvspHT_readcounts_Lonly_desc$Origin)
ase_pHEvspHT_readcounts_Lonly_dds<-DESeqDataSetFromMatrix(ase_pHEvspHT_readcounts_Lonly, ase_pHEvspHT_readcounts_Lonly_desc, design=~Origin)
ase_pHEvspHT_readcounts_Lonly_dds<-DESeq(ase_pHEvspHT_readcounts_Lonly_dds)
resultsNames(ase_pHEvspHT_readcounts_Lonly_dds)
ase_pHEvspHT_readcounts_Lonly_res<-results(ase_pHEvspHT_readcounts_Lonly_dds,name="Origin_Htub_vs_Hery")
ase_pHEvspHT_readcounts_Lonly_res<-as.data.frame(ase_pHEvspHT_readcounts_Lonly_res)
#negative means higher in pHE than pHT

#INHERITANCE TESTS BASED ON ABOVE DESEQs

ase_nolowcounts.lfcShrink_Lonly_Hyb_vs_Hery_res<-lfcShrink(ase_pHEvsHyball_readcounts_Lonly_dds, coef="Origin_Hyb_vs_Hery") #lifted hybs vs lifted pHE, w/normalization, negative L2FC means higher in Hery than Hyb
ase_nolowcounts.lfcShrink_Lonly_Hyb_vs_Htub_res<-lfcShrink(ase_pHTvsHyball_readcounts_Lonly_dds, coef="Origin_Hyb_vs_Htub")#lifted hybs vs lifted pHT, w/normalization, negative L2FC means higher in htub than Hyb
ase_nolowcounts.lfcShrink_Lonly_Htub_vs_Hery_res<-lfcShrink(ase_pHEvspHT_readcounts_Lonly_dds,coef="Origin_Htub_vs_Hery") #lifted pHtub vs lifted pHery, w/normalization, negative L2FC means higher in Hery than htub

#replacing NAs with 1s
ase_nolowcounts.lfcShrink_Lonly_Hyb_vs_Hery_res<-as.data.frame(ase_nolowcounts.lfcShrink_Lonly_Hyb_vs_Hery_res)
ase_nolowcounts.lfcShrink_Lonly_Hyb_vs_Hery_res[is.na(ase_nolowcounts.lfcShrink_Lonly_Hyb_vs_Hery_res)]<-1
ase_nolowcounts.lfcShrink_Lonly_Hyb_vs_Htub_res<-as.data.frame(ase_nolowcounts.lfcShrink_Lonly_Hyb_vs_Htub_res)
ase_nolowcounts.lfcShrink_Lonly_Hyb_vs_Htub_res[is.na(ase_nolowcounts.lfcShrink_Lonly_Hyb_vs_Htub_res)]<-1
ase_nolowcounts.lfcShrink_Lonly_Htub_vs_Hery_res<-as.data.frame(ase_nolowcounts.lfcShrink_Lonly_Htub_vs_Hery_res)
ase_nolowcounts.lfcShrink_Lonly_Htub_vs_Hery_res[is.na(ase_nolowcounts.lfcShrink_Lonly_Htub_vs_Hery_res)]<-1

ase_nolowcounts.inh.hyb.vs.HE.L<-ase_nolowcounts.lfcShrink_Lonly_Hyb_vs_Hery_res %>% select(log2FChyb_pHE=log2FoldChange,HybvspHE_padj=padj)
ase_nolowcounts.inh.hyb.vs.HT.L<-ase_nolowcounts.lfcShrink_Lonly_Hyb_vs_Htub_res %>% select(log2FChyb_pHT=log2FoldChange,HybvspHT_padj=padj)
ase_nolowcounts.inh.HT.vs.HE.L<-ase_nolowcounts.lfcShrink_Lonly_Htub_vs_Hery_res %>% select(log2FCpHT_pHE=log2FoldChange, pHTvspHE_padj=padj)
ase_nolowcounts.inh.L.all<-cbind(ase_nolowcounts.inh.hyb.vs.HE.L,ase_nolowcounts.inh.hyb.vs.HT.L,ase_nolowcounts.inh.HT.vs.HE.L)
#remember that for log2FChyb_hybHE, negative means higher in Hery; for log2FChyb_pHT, negative means higher in htub; for htub vs Hery, negative means higher in Hery

#conserved are when Ht = He; Hy = He; Hy = Ht. ie, pHe not sig diff from pHT; HybTOT not sig diff from pHE; HybTOT not sig diff from pHT
ase_nolowcounts.L_conserved<-ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0.05 & HybvspHT_padj>0.05)

#underdominant are when (Ht = He ; Hy < He ; Hy < Ht) OR 
ase_nolowcounts.L_underdom<-ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE <(0) & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0))
#(Ht < He ; Hy < He ; Hy < Ht underdominant) OR 
ase_nolowcounts.L_underdom<-rbind(ase_nolowcounts.L_underdom, ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE<(0) & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE <(0) & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0)))
#(Ht > He ; Hy < He ; Hy < Ht). 
ase_nolowcounts.L_underdom<-rbind(ase_nolowcounts.L_underdom, ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE>0 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE <(0) & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0)))

#overdominant are when (Ht = He ; Hy > He ; Hy > Ht) OR 
ase_nolowcounts.L_overdom<-ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT>0)
#(Ht < He ; Hy > He ; Hy > Ht) OR 
ase_nolowcounts.L_overdom<-rbind(ase_nolowcounts.L_overdom, ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE<(0) & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT>0))
#(Ht > He ; Hy > He ; Hy > Ht) 
ase_nolowcounts.L_overdom<-rbind(ase_nolowcounts.L_overdom, ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE>0 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT>0))

#HEdominant are when (Ht < He ; Hy = He ; Hy > Ht) OR 
ase_nolowcounts.L_HEdom<-ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE<(0) & HybvspHE_padj > 0.05 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT>0)
#(Ht > He ; Hy = He ; Hy < Ht)
ase_nolowcounts.L_HEdom<-rbind(ase_nolowcounts.L_HEdom, ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE>0 & HybvspHE_padj > 0.05 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0)))

#HTdominant are when (Ht < He ;Hy < He; Hy = Ht) OR 
ase_nolowcounts.L_HTdom<-ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE<(0) & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE<(0) & HybvspHT_padj>0.05)
#(Ht > He ; Hy > He ; Hy = Ht)
ase_nolowcounts.L_HTdom<-rbind(ase_nolowcounts.L_HTdom, ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE>0 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0.05))

#additive are when (Ht < He ; Hy < He ; Hy > Ht) 
ase_nolowcounts.L_additive<-ase_nolowcounts.inh.L.all%>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE<(0) & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE<(0) & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT>0)
#or (Ht > He ; Hy > He ; Hy < Ht)
ase_nolowcounts.L_additive<-rbind(ase_nolowcounts.L_additive, ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE>0 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0)))

#ambiguous are when 
#(Ht = He ; Hy = He ; Hy < Ht) OR
ase_nolowcounts.L_ambig<-ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj > 0.05 & HybvspHT_padj >0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0))
#(Ht = He ; Hy = He ; Hy > Ht) OR
ase_nolowcounts.L_ambig<-rbind(ase_nolowcounts.L_ambig, ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj > 0.05 & HybvspHT_padj >0 & HybvspHT_padj<=0.05 & log2FChyb_pHT>0))
#(Ht = He ; Hy < He ; Hy = Ht) OR
ase_nolowcounts.L_ambig<-rbind(ase_nolowcounts.L_ambig, ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE<(0) & HybvspHT_padj>0.05))
#(Ht = He ; Hy > He ; Hy = Ht) OR
ase_nolowcounts.L_ambig<-rbind(ase_nolowcounts.L_ambig, ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0.05))
#(Ht = He ; Hy < He ; Hy > Ht) OR
ase_nolowcounts.L_ambig<-rbind(ase_nolowcounts.L_ambig, ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE<0 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT>0))
#(Ht = He ; Hy > He ; Hy < Ht) OR
ase_nolowcounts.L_ambig<-rbind(ase_nolowcounts.L_ambig, ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0.05 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT <(0)))
#(Ht < He ; Hy = He ; Hy = Ht) OR
ase_nolowcounts.L_ambig<-rbind(ase_nolowcounts.L_ambig, ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj <=0.05 & log2FCpHT_pHE <(0) & HybvspHE_padj > 0.05 & HybvspHT_padj >0.05))
#(Ht > He ; Hy = He ; Hy = Ht) OR
ase_nolowcounts.L_ambig<-rbind(ase_nolowcounts.L_ambig, ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj <=0.05 & log2FCpHT_pHE >0 & HybvspHE_padj > 0.05 & HybvspHT_padj>0.05))
#(Ht < He ; Hy = He ; Hy < Ht) OR
ase_nolowcounts.L_ambig<-rbind(ase_nolowcounts.L_ambig, ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj <=0.05 & log2FCpHT_pHE <(0) & HybvspHE_padj > 0.05 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT <(0)))
#(Ht > He ; Hy = He ; Hy > Ht) OR
ase_nolowcounts.L_ambig<-rbind(ase_nolowcounts.L_ambig, ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj <=0.05 & log2FCpHT_pHE >0 & HybvspHE_padj > 0.05 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT >0))
#(Ht < He ; Hy > He ; Hy = Ht) OR
ase_nolowcounts.L_ambig<-rbind(ase_nolowcounts.L_ambig, ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj <=0.05 & log2FCpHT_pHE <(0) & HybvspHE_padj>0 &HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0.05))
#(Ht > He ; Hy < He ; Hy = Ht) OR
ase_nolowcounts.L_ambig<-rbind(ase_nolowcounts.L_ambig, ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj <=0.05 & log2FCpHT_pHE >0 & HybvspHE_padj>0 &HybvspHE_padj<=0.05 & log2FChyb_pHE<(0) & HybvspHT_padj>0.05))
#(Ht < He ; Hy > He ; Hy < Ht) OR
ase_nolowcounts.L_ambig<-rbind(ase_nolowcounts.L_ambig, ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj <=0.05 & log2FCpHT_pHE <(0) & HybvspHE_padj>0 &HybvspHE_padj<=0.05 & log2FChyb_pHE>0 & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT<(0)))
#(Ht > He ; Hy < He ; Hy > Ht)
ase_nolowcounts.L_ambig<-rbind(ase_nolowcounts.L_ambig,ase_nolowcounts.inh.L.all %>% dplyr::filter(pHTvspHE_padj>0 & pHTvspHE_padj<=0.05 & log2FCpHT_pHE > 2 & HybvspHE_padj>0 & HybvspHE_padj<=0.05 &log2FChyb_pHE<(0) & HybvspHT_padj>0 & HybvspHT_padj<=0.05 & log2FChyb_pHT >0))

ase_nolowcounts.inh.test.res$L<-as.numeric(c(nrow(ase_nolowcounts.L_conserved),nrow(ase_nolowcounts.L_underdom),nrow(ase_nolowcounts.L_overdom),nrow(ase_nolowcounts.L_HEdom),nrow(ase_nolowcounts.L_HTdom),nrow(ase_nolowcounts.L_additive),nrow(ase_nolowcounts.L_ambig)))
ase_nolowcounts.inh.test.res$Inheritance_Type<-rownames(ase_nolowcounts.inh.test.res)
ase_nolowcounts.inh.test.res<-ase_nolowcounts.inh.test.res[c(4,1:3)]

library(reshape)
ase_nolowcounts.inh.test.res.melt2<-melt(ase_nolowcounts.inh.test.res)
ase_nolowcounts.inh.test.res.plot<-ggplot(ase_nolowcounts.inh.test.res.melt2, aes(x=variable, y=(value),color=Inheritance_Type, group=Inheritance_Type))+
  geom_line(size=2)+theme_classic()+theme(panel.background = element_rect(fill = "transparent",colour = NA), plot.background = element_rect(fill = "transparent",colour = NA), legend.background = element_rect(fill = "transparent",colour = NA))+ 
  xlab("") + ylab("Number of Peaks") + ggtitle("") + theme(plot.title = element_text(size=24, vjust = 1.5, face = "bold"), axis.text = element_text(size=20), axis.title=element_text(size=22),legend.text = element_text(size=19), legend.title=element_text(size=19), legend.key = element_rect(size = 5, colour = NA), legend.key.size = unit(1.5, 'lines'))
ase_nolowcounts.inh.test.res.plot
#ggsave("./ase_nolowcounts.inh.test.res.plot.pdf", ase_nolowcounts.inh.test.res.plot, width=8, height=6)

ase_nolowcounts.inh.test.res.prop<-ase_nolowcounts.inh.test.res
ase_nolowcounts.inh.test.res.prop$B<-ase_nolowcounts.inh.test.res$B/sum(ase_nolowcounts.inh.test.res$B)
ase_nolowcounts.inh.test.res.prop$G<-ase_nolowcounts.inh.test.res$G/sum(ase_nolowcounts.inh.test.res$G)
ase_nolowcounts.inh.test.res.prop$L<-ase_nolowcounts.inh.test.res$L/sum(ase_nolowcounts.inh.test.res$L)
ase_nolowcounts.inh.test.res.prop.melt2<-melt(ase_nolowcounts.inh.test.res.prop)
ase_nolowcounts.inh.test.res.prop.plot<-ggplot(ase_nolowcounts.inh.test.res.prop.melt2, aes(x=variable, y=(value),color=Inheritance_Type, group=Inheritance_Type))+
  geom_line(size=2)+theme_classic()+theme(panel.background = element_rect(fill = "transparent",colour = NA), plot.background = element_rect(fill = "transparent",colour = NA), legend.background = element_rect(fill = "transparent",colour = NA))+ 
  xlab("") + ylab("Proportion of Peaks") + ggtitle("") + theme(plot.title = element_text(size=24, vjust = 1.5, face = "bold"), axis.text = element_text(size=20), axis.title=element_text(size=22),legend.text = element_text(size=19), legend.title=element_text(size=19), legend.key = element_rect(size = 5, colour = NA), legend.key.size = unit(1.5, 'lines'))+
  ylim(0,1.0)
ase_nolowcounts.inh.test.res.prop.plot
#ggsave("./ase_nolowcounts.inh.test.res.prop.plot.pdf", ase_nolowcounts.inh.test.res.prop.plot, width=8, height=6)
```

CIS/TRANS TESTS and assocd DEseq
```{r}
ase_nolowcounts.CTliftedHyb_readcounts_Bonly<-(ase_nolowcounts %>% 
                               select(HybHEB12,HybHEB13,HybHTB12,HybHTB13))
rownames(ase_nolowcounts.CTliftedHyb_readcounts_Bonly)<-paste0(ase_nolowcounts$Chr,":",ase_nolowcounts$start, "-",ase_nolowcounts$end)
ase_nolowcounts.CTliftedHyb_readcounts_Bonly<-round(ase_nolowcounts.CTliftedHyb_readcounts_Bonly)

ase_nolowcounts.CT_liftedHyb.B.desc<-data.frame(individual=factor(c(12,13,12,13)),
                  allele=factor(c("HybHE","HybHE","HybHT","HybHT")))
ase_nolowcounts.CT_liftedHyb_Bonlydds<-DESeqDataSetFromMatrix(ase_nolowcounts.CTliftedHyb_readcounts_Bonly,
                                        ase_nolowcounts.CT_liftedHyb.B.desc,
                                        design = model.matrix(~0+ase_nolowcounts.CT_liftedHyb.B.desc$individual+ase_nolowcounts.CT_liftedHyb.B.desc$allele))
sizeFactors(ase_nolowcounts.CT_liftedHyb_Bonlydds)<-rep(1,4) #setting size factors to 1 so library size isn't controlled btwn hyb reads since they come from same sample
ase_nolowcounts.CT_liftedHyb_Bonlydds<-DESeq(ase_nolowcounts.CT_liftedHyb_Bonlydds)
resultsNames(ase_nolowcounts.CT_liftedHyb_Bonlydds) #negative means higher in hybHE than hybHT

#on comparing the parent he's to the parent ht's: i can't just randomly pair them up. so i am first normalizing to library size, then doing all pairwise comparisons and taking the average of those.
ase_nolowcounts.CTliftedTT_readcounts_Bonly<-ase_nolowcounts
rownames(ase_nolowcounts.CTliftedTT_readcounts_Bonly)<-paste0(ase_nolowcounts$Chr,":",ase_nolowcounts$start, "-",ase_nolowcounts$end)
ase_nolowcounts.CTliftedTT_readcounts_Bonly[4:43]<-ase_nolowcounts.CTliftedTT_readcounts_Bonly[4:43]+1
ase_nolowcounts.CTliftedTT_readcounts_Bonly$"HybHEB12/HybHTB12"<-ase_nolowcounts.CTliftedTT_readcounts_Bonly$HybHEB12/ase_nolowcounts.CTliftedTT_readcounts_Bonly$HybHTB12
ase_nolowcounts.CTliftedTT_readcounts_Bonly$"HybHEB13/HybHTB13"<-ase_nolowcounts.CTliftedTT_readcounts_Bonly$HybHEB13/ase_nolowcounts.CTliftedTT_readcounts_Bonly$HybHTB13
ase_nolowcounts.CTliftedTT_readcounts_Bonly$"HeryB4/HtubB4"<-(ase_nolowcounts.CTliftedTT_readcounts_Bonly$HeryB4/sum(ase_nolowcounts.CTliftedTT_readcounts_Bonly$HeryB4))/(ase_nolowcounts.CTliftedTT_readcounts_Bonly$htubB4/sum(ase_nolowcounts.CTliftedTT_readcounts_Bonly$htubB4))
ase_nolowcounts.CTliftedTT_readcounts_Bonly$"HeryB4/HtubB7"<-(ase_nolowcounts.CTliftedTT_readcounts_Bonly$HeryB4/sum(ase_nolowcounts.CTliftedTT_readcounts_Bonly$HeryB4))/(ase_nolowcounts.CTliftedTT_readcounts_Bonly$htubB7/sum(ase_nolowcounts.CTliftedTT_readcounts_Bonly$htubB7))
ase_nolowcounts.CTliftedTT_readcounts_Bonly$"HeryB7/HtubB4"<-(ase_nolowcounts.CTliftedTT_readcounts_Bonly$HeryB7/sum(ase_nolowcounts.CTliftedTT_readcounts_Bonly$HeryB7))/(ase_nolowcounts.CTliftedTT_readcounts_Bonly$htubB4/sum(ase_nolowcounts.CTliftedTT_readcounts_Bonly$htubB4))
ase_nolowcounts.CTliftedTT_readcounts_Bonly$"HeryB7/HtubB7"<-(ase_nolowcounts.CTliftedTT_readcounts_Bonly$HeryB7/sum(ase_nolowcounts.CTliftedTT_readcounts_Bonly$HeryB7))/(ase_nolowcounts.CTliftedTT_readcounts_Bonly$htubB7/sum(ase_nolowcounts.CTliftedTT_readcounts_Bonly$htubB7))
ase_nolowcounts.CTliftedTT_readcounts_Bonly<-ase_nolowcounts.CTliftedTT_readcounts_Bonly[c(44:49)]
ase_nolowcounts.CTliftedTT_readcounts_Bonly<-ase_nolowcounts.CTliftedTT_readcounts_Bonly*100
ase_nolowcounts.CTliftedTT_readcounts_Bonly<-round(ase_nolowcounts.CTliftedTT_readcounts_Bonly) #deseq makes me put in integers

ase_nolowcounts.CTliftedTT_readcounts_Bonly_desc<-data_frame(Sample=colnames(ase_nolowcounts.CTliftedTT_readcounts_Bonly),
                                             Origin=as.factor(c("HybvHyb","HybvHyb",rep("pHEvpHT",4))))

ase_nolowcounts.CTliftedTT_readcounts_Bonlydds<-DESeqDataSetFromMatrix(ase_nolowcounts.CTliftedTT_readcounts_Bonly,
                                        ase_nolowcounts.CTliftedTT_readcounts_Bonly_desc,
                                        design = ~Origin)
ase_nolowcounts.CTliftedTT_readcounts_Bonlydds<-DESeq(ase_nolowcounts.CTliftedTT_readcounts_Bonlydds)
resultsNames(ase_nolowcounts.CTliftedTT_readcounts_Bonlydds)
#negative should mean HE/HT is greater than HybHE/HybHT

#lifted parents vs each other was already done during Inh tests:
#need to lfcshrink ase_nolowcounts.CT_liftedHyb_Bonlydds and ase_nolowcounts.CTliftedTT_readcounts_Bonlydds:
lfcase_nolowcounts.CT_liftedHyb_Bonlyres<-lfcShrink(ase_nolowcounts.CT_liftedHyb_Bonlydds,coef="ase_nolowcounts.CT_liftedHyb.B.desc.alleleHybHT")
lfcase_nolowcounts.CTliftedTT_readcounts_Bonlyres<-lfcShrink(ase_nolowcounts.CTliftedTT_readcounts_Bonlydds, coef="Origin_pHEvpHT_vs_HybvHyb")
lfcase_nolowcounts.CT_liftedHyb_Bonlyres<-as.data.frame(lfcase_nolowcounts.CT_liftedHyb_Bonlyres)
lfcase_nolowcounts.CT_liftedHyb_Bonlyres[is.na(lfcase_nolowcounts.CT_liftedHyb_Bonlyres)]<-1
lfcase_nolowcounts.CTliftedTT_readcounts_Bonlyres<-as.data.frame(lfcase_nolowcounts.CTliftedTT_readcounts_Bonlyres)
lfcase_nolowcounts.CTliftedTT_readcounts_Bonlyres[is.na(lfcase_nolowcounts.CTliftedTT_readcounts_Bonlyres)]<-1

ase_nolowcounts.CT.hyb.B<-lfcase_nolowcounts.CT_liftedHyb_Bonlyres %>% select("log2FChybHE/hybHT"=log2FoldChange,"HybHE/HybHT_padj"=padj)
ase_nolowcounts.CT.TT.B<-lfcase_nolowcounts.CTliftedTT_readcounts_Bonlyres %>% select("log2FC_TT_pHE/pHTvsHybHE/HybHT"=log2FoldChange,"TT_pHE/pHTvsHybHE/HybHT_padj"=padj)
ase_nolowcounts.CT.TT.HEvsHT<-ase_nolowcounts.lfcShrink_Bonly_Htub_vs_Hery_res %>% select("log2FC_TT_HEvsHT"=log2FoldChange, "TT_HEvsHTpadj"=padj)
ase_nolowcounts.CT.B.all<-cbind(ase_nolowcounts.CT.hyb.B,ase_nolowcounts.CT.TT.B,ase_nolowcounts.CT.TT.HEvsHT)

#doing the actual tests:
#conserved: P1 = P2; A1 = A2; P1/P2 = A1/A2
ase_nolowcounts.CT.B.conserved<-ase_nolowcounts.CT.B.all %>% dplyr::filter(ase_nolowcounts.CT.B.all$"TT_HEvsHTpadj">0.05 & ase_nolowcounts.CT.B.all$"HybHE/HybHT_padj">0.05 & ase_nolowcounts.CT.B.all$"TT_pHE/pHTvsHybHE/HybHT_padj">0.05)

#compensatory: P1 = P2; A1  A2; P1/P2  A1/A2
ase_nolowcounts.CT.B.comp<-ase_nolowcounts.CT.B.all %>% dplyr::filter(ase_nolowcounts.CT.B.all$"TT_HEvsHTpadj">0.05 & ase_nolowcounts.CT.B.all$"HybHE/HybHT_padj"<=0.05 & ase_nolowcounts.CT.B.all$"TT_pHE/pHTvsHybHE/HybHT_padj"<=0.05)

#all trans: P1  P2; A1 = A2; P1/P2  A1/A2
ase_nolowcounts.CT.B.alltrans<-ase_nolowcounts.CT.B.all %>% dplyr::filter(ase_nolowcounts.CT.B.all$"TT_HEvsHTpadj"<=0.05 & ase_nolowcounts.CT.B.all$"HybHE/HybHT_padj">0.05 & ase_nolowcounts.CT.B.all$"TT_pHE/pHTvsHybHE/HybHT_padj"<=0.05)

#all cis: P1  P2; A1  A2; P1/P2 = A1/A2
ase_nolowcounts.CT.B.allcis<-ase_nolowcounts.CT.B.all %>% dplyr::filter(ase_nolowcounts.CT.B.all$"TT_HEvsHTpadj"<=0.05 & ase_nolowcounts.CT.B.all$"HybHE/HybHT_padj"<=0.05 & ase_nolowcounts.CT.B.all$"TT_pHE/pHTvsHybHE/HybHT_padj">0.05)

#cis + trans: P1  P2; A1  A2; P1/P2 > A1/A2
ase_nolowcounts.CT.B.CplusT<-ase_nolowcounts.CT.B.all %>% dplyr::filter(ase_nolowcounts.CT.B.all$"TT_HEvsHTpadj"<=0.05 & ase_nolowcounts.CT.B.all$"HybHE/HybHT_padj"<=0.05 & ase_nolowcounts.CT.B.all$"TT_pHE/pHTvsHybHE/HybHT_padj"<=0.05 & `log2FC_TT_pHE/pHTvsHybHE/HybHT`<0)

#cis x trans: P1  P2; A1  A2; P1/P2 < A1/A2
ase_nolowcounts.CT.B.CbyT<-ase_nolowcounts.CT.B.all %>% dplyr::filter(ase_nolowcounts.CT.B.all$"TT_HEvsHTpadj"<=0.05 & ase_nolowcounts.CT.B.all$"HybHE/HybHT_padj"<=0.05 & ase_nolowcounts.CT.B.all$"TT_pHE/pHTvsHybHE/HybHT_padj"<=0.05 & `log2FC_TT_pHE/pHTvsHybHE/HybHT`>0)

#ambiguous: P1 = P2; A1 = A2; P1/P2  A1/A2 OR
ase_nolowcounts.CT.B.ambig<-ase_nolowcounts.CT.B.all %>% dplyr::filter(ase_nolowcounts.CT.B.all$"TT_HEvsHTpadj">0.05 & ase_nolowcounts.CT.B.all$"HybHE/HybHT_padj">0.05 & ase_nolowcounts.CT.B.all$"TT_pHE/pHTvsHybHE/HybHT_padj"<=0.05)
#P1 = P2 A1  A2 P1/P2 = A1/A2 OR
ase_nolowcounts.CT.B.ambig<-rbind(ase_nolowcounts.CT.B.ambig, ase_nolowcounts.CT.B.all %>% dplyr::filter(ase_nolowcounts.CT.B.all$"TT_HEvsHTpadj">0.05 & ase_nolowcounts.CT.B.all$"HybHE/HybHT_padj"<=0.05 & ase_nolowcounts.CT.B.all$"TT_pHE/pHTvsHybHE/HybHT_padj">0.05))
#P1  P2 A1 = A2 P1/P2 = A1/A2
ase_nolowcounts.CT.B.ambig<-rbind(ase_nolowcounts.CT.B.ambig, ase_nolowcounts.CT.B.all %>% dplyr::filter(ase_nolowcounts.CT.B.all$"TT_HEvsHTpadj"<=0.05 & ase_nolowcounts.CT.B.all$"HybHE/HybHT_padj">0.05 & ase_nolowcounts.CT.B.all$"TT_pHE/pHTvsHybHE/HybHT_padj">0.05))

ase_nolowcounts.CT.test.res<-data_frame(B=as.numeric(c(nrow(ase_nolowcounts.CT.B.conserved),nrow(ase_nolowcounts.CT.B.comp),nrow(ase_nolowcounts.CT.B.alltrans),nrow(ase_nolowcounts.CT.B.allcis),nrow(ase_nolowcounts.CT.B.CplusT),nrow(ase_nolowcounts.CT.B.CbyT),nrow(ase_nolowcounts.CT.B.ambig))))
ase_nolowcounts.CT.test.res$Type<-c("Conserved","Compensatory","All trans", "All cis", "Cis + trans","Cis x trans","ambig")
ase_nolowcounts.CT.test.res<-ase_nolowcounts.CT.test.res[c(2,1)]
```

#Gastrula an.CT TESTS:

```{r}
#Gastrula ase_nolowcounts.CT Tests and assoc'd DEseq

ase_nolowcounts.CTliftedHyb_readcounts_Gonly<-(ase_nolowcounts %>% 
                               select(HybHEG7,HybHEG8,HybHEG9,HybHTG7,HybHTG8,HybHTG9))
rownames(ase_nolowcounts.CTliftedHyb_readcounts_Gonly)<-paste0(ase_nolowcounts$Chr,":",ase_nolowcounts$start, "-",ase_nolowcounts$end)
ase_nolowcounts.CTliftedHyb_readcounts_Gonly<-round(ase_nolowcounts.CTliftedHyb_readcounts_Gonly)

ase_nolowcounts.CT_liftedHyb.G.desc<-data.frame(individual=factor(c(7,8,9,7,8,9)),
                  allele=factor(c("HybHE","HybHE","HybHE","HybHT","HybHT","HybHT")))
ase_nolowcounts.CT_liftedHyb_Gonlydds<-DESeqDataSetFromMatrix(ase_nolowcounts.CTliftedHyb_readcounts_Gonly,
                                        ase_nolowcounts.CT_liftedHyb.G.desc,
                                        design = model.matrix(~0+ase_nolowcounts.CT_liftedHyb.G.desc$individual+ase_nolowcounts.CT_liftedHyb.G.desc$allele))
sizeFactors(ase_nolowcounts.CT_liftedHyb_Gonlydds)<-rep(1,6) #setting size factors to 1 so library size isn't controlled btwn hyb reads since they come from same sample #number of reps should be nuumer of samples
ase_nolowcounts.CT_liftedHyb_Gonlydds<-DESeq(ase_nolowcounts.CT_liftedHyb_Gonlydds)
resultsNames(ase_nolowcounts.CT_liftedHyb_Gonlydds) #negative means higher in hybHE than hybHT

ase_nolowcounts.CTliftedTT_readcounts_Gonly<-ase_nolowcounts
rownames(ase_nolowcounts.CTliftedTT_readcounts_Gonly)<-paste0(ase_nolowcounts$Chr,":",ase_nolowcounts$start, "-",ase_nolowcounts$end)
ase_nolowcounts.CTliftedTT_readcounts_Gonly[4:43]<-ase_nolowcounts.CTliftedTT_readcounts_Gonly[4:43]+1
ase_nolowcounts.CTliftedTT_readcounts_Gonly$"HybHEG7/HybHTG7"<-ase_nolowcounts.CTliftedTT_readcounts_Gonly$HybHEG7/ase_nolowcounts.CTliftedTT_readcounts_Gonly$HybHTG7
ase_nolowcounts.CTliftedTT_readcounts_Gonly$"HybHEG8/HybHTG8"<-ase_nolowcounts.CTliftedTT_readcounts_Gonly$HybHEG8/ase_nolowcounts.CTliftedTT_readcounts_Gonly$HybHTG8
ase_nolowcounts.CTliftedTT_readcounts_Gonly$"HybHEG9/HybHTG9"<-ase_nolowcounts.CTliftedTT_readcounts_Gonly$HybHEG9/ase_nolowcounts.CTliftedTT_readcounts_Gonly$HybHTG9
ase_nolowcounts.CTliftedTT_readcounts_Gonly$"HeryG2/HtubG2"<-(ase_nolowcounts.CTliftedTT_readcounts_Gonly$HeryG2/sum(ase_nolowcounts.CTliftedTT_readcounts_Gonly$HeryG2))/(ase_nolowcounts.CTliftedTT_readcounts_Gonly$htubG2/sum(ase_nolowcounts.CTliftedTT_readcounts_Gonly$htubG2))
ase_nolowcounts.CTliftedTT_readcounts_Gonly$"HeryG2/HtubG5"<-(ase_nolowcounts.CTliftedTT_readcounts_Gonly$HeryG2/sum(ase_nolowcounts.CTliftedTT_readcounts_Gonly$HeryG2))/(ase_nolowcounts.CTliftedTT_readcounts_Gonly$htubG5/sum(ase_nolowcounts.CTliftedTT_readcounts_Gonly$htubG5))
ase_nolowcounts.CTliftedTT_readcounts_Gonly$"HeryG2/HtubG8"<-(ase_nolowcounts.CTliftedTT_readcounts_Gonly$HeryG2/sum(ase_nolowcounts.CTliftedTT_readcounts_Gonly$HeryG2))/(ase_nolowcounts.CTliftedTT_readcounts_Gonly$htubG8/sum(ase_nolowcounts.CTliftedTT_readcounts_Gonly$htubG8))
ase_nolowcounts.CTliftedTT_readcounts_Gonly$"HeryG5/HtubG2"<-(ase_nolowcounts.CTliftedTT_readcounts_Gonly$HeryG5/sum(ase_nolowcounts.CTliftedTT_readcounts_Gonly$HeryG5))/(ase_nolowcounts.CTliftedTT_readcounts_Gonly$htubG2/sum(ase_nolowcounts.CTliftedTT_readcounts_Gonly$htubG2))
ase_nolowcounts.CTliftedTT_readcounts_Gonly$"HeryG5/HtubG5"<-(ase_nolowcounts.CTliftedTT_readcounts_Gonly$HeryG5/sum(ase_nolowcounts.CTliftedTT_readcounts_Gonly$HeryG5))/(ase_nolowcounts.CTliftedTT_readcounts_Gonly$htubG5/sum(ase_nolowcounts.CTliftedTT_readcounts_Gonly$htubG5))
ase_nolowcounts.CTliftedTT_readcounts_Gonly$"HeryG5/HtubG8"<-(ase_nolowcounts.CTliftedTT_readcounts_Gonly$HeryG5/sum(ase_nolowcounts.CTliftedTT_readcounts_Gonly$HeryG5))/(ase_nolowcounts.CTliftedTT_readcounts_Gonly$htubG8/sum(ase_nolowcounts.CTliftedTT_readcounts_Gonly$htubG8))
ase_nolowcounts.CTliftedTT_readcounts_Gonly$"HeryG8/HtubG2"<-(ase_nolowcounts.CTliftedTT_readcounts_Gonly$HeryG8/sum(ase_nolowcounts.CTliftedTT_readcounts_Gonly$HeryG8))/(ase_nolowcounts.CTliftedTT_readcounts_Gonly$htubG2/sum(ase_nolowcounts.CTliftedTT_readcounts_Gonly$htubG2))
ase_nolowcounts.CTliftedTT_readcounts_Gonly$"HeryG8/HtubG5"<-(ase_nolowcounts.CTliftedTT_readcounts_Gonly$HeryG8/sum(ase_nolowcounts.CTliftedTT_readcounts_Gonly$HeryG8))/(ase_nolowcounts.CTliftedTT_readcounts_Gonly$htubG5/sum(ase_nolowcounts.CTliftedTT_readcounts_Gonly$htubG5))
ase_nolowcounts.CTliftedTT_readcounts_Gonly$"HeryG8/HtubG8"<-(ase_nolowcounts.CTliftedTT_readcounts_Gonly$HeryG8/sum(ase_nolowcounts.CTliftedTT_readcounts_Gonly$HeryG8))/(ase_nolowcounts.CTliftedTT_readcounts_Gonly$htubG8/sum(ase_nolowcounts.CTliftedTT_readcounts_Gonly$htubG8))
ase_nolowcounts.CTliftedTT_readcounts_Gonly<-ase_nolowcounts.CTliftedTT_readcounts_Gonly[c(44:55)]
ase_nolowcounts.CTliftedTT_readcounts_Gonly<-ase_nolowcounts.CTliftedTT_readcounts_Gonly*100
ase_nolowcounts.CTliftedTT_readcounts_Gonly<-round(ase_nolowcounts.CTliftedTT_readcounts_Gonly) #deseq makes me put in integers
ase_nolowcounts.CTliftedTT_readcounts_Gonly_desc<-data_frame(Sample=colnames(ase_nolowcounts.CTliftedTT_readcounts_Gonly),
                                             Origin=as.factor(c("HybvHyb","HybvHyb","HybvHyb",rep("pHEvpHT",9))))

ase_nolowcounts.CTliftedTT_readcounts_Gonlydds<-DESeqDataSetFromMatrix(ase_nolowcounts.CTliftedTT_readcounts_Gonly,
                                        ase_nolowcounts.CTliftedTT_readcounts_Gonly_desc,
                                        design = ~Origin)
ase_nolowcounts.CTliftedTT_readcounts_Gonlydds<-DESeq(ase_nolowcounts.CTliftedTT_readcounts_Gonlydds)
resultsNames(ase_nolowcounts.CTliftedTT_readcounts_Gonlydds)
#negative should mean HE/HT is greater than HybHE/HybHT i think. NOT POSITIVE NEED TO CHECK. also not sure if i should have forced sizefactors to 1 here as well

#lifted parents vs each other was already done during Inh tests (i think):
#View(ase_nolowcounts.lfcShrink_Gonly_Htub_vs_Hery_res )
#need to lfcshrink ase_nolowcounts.CT_liftedHyb_Gonlydds and ase_nolowcounts.CTliftedTT_readcounts_Gonlydds:
lfcase_nolowcounts.CT_liftedHyb_Gonlyres<-lfcShrink(ase_nolowcounts.CT_liftedHyb_Gonlydds,coef="ase_nolowcounts.CT_liftedHyb.G.desc.alleleHybHT")
lfcase_nolowcounts.CTliftedTT_readcounts_Gonlyres<-lfcShrink(ase_nolowcounts.CTliftedTT_readcounts_Gonlydds, coef="Origin_pHEvpHT_vs_HybvHyb")
lfcase_nolowcounts.CT_liftedHyb_Gonlyres<-as.data.frame(lfcase_nolowcounts.CT_liftedHyb_Gonlyres)
lfcase_nolowcounts.CT_liftedHyb_Gonlyres[is.na(lfcase_nolowcounts.CT_liftedHyb_Gonlyres)]<-1
lfcase_nolowcounts.CTliftedTT_readcounts_Gonlyres<-as.data.frame(lfcase_nolowcounts.CTliftedTT_readcounts_Gonlyres)
lfcase_nolowcounts.CTliftedTT_readcounts_Gonlyres[is.na(lfcase_nolowcounts.CTliftedTT_readcounts_Gonlyres)]<-1

ase_nolowcounts.CT.hyb.G<-lfcase_nolowcounts.CT_liftedHyb_Gonlyres %>% select("log2FChybHE/hybHT"=log2FoldChange,"HybHE/HybHT_padj"=padj)
ase_nolowcounts.CT.TT.G<-lfcase_nolowcounts.CTliftedTT_readcounts_Gonlyres %>% select("log2FC_TT_pHE/pHTvsHybHE/HybHT"=log2FoldChange,"TT_pHE/pHTvsHybHE/HybHT_padj"=padj)
Ct.TT.HEvsHT<-ase_nolowcounts.lfcShrink_Gonly_Htub_vs_Hery_res %>% select("log2FC_TT_HEvsHT"=log2FoldChange, "TT_HEvsHTpadj"=padj)
ase_nolowcounts.CT.G.all<-cbind(ase_nolowcounts.CT.hyb.G,ase_nolowcounts.CT.TT.G,Ct.TT.HEvsHT)

#doing the actual tests:
#conserved: P1 = P2; A1 = A2; P1/P2 = A1/A2
ase_nolowcounts.CT.G.conserved<-ase_nolowcounts.CT.G.all %>% dplyr::filter(ase_nolowcounts.CT.G.all$"TT_HEvsHTpadj">0.05 & ase_nolowcounts.CT.G.all$"HybHE/HybHT_padj">0.05 & ase_nolowcounts.CT.G.all$"TT_pHE/pHTvsHybHE/HybHT_padj">0.05)

#compensatory: P1 = P2; A1  A2; P1/P2  A1/A2
ase_nolowcounts.CT.G.comp<-ase_nolowcounts.CT.G.all %>% dplyr::filter(ase_nolowcounts.CT.G.all$"TT_HEvsHTpadj">0.05 & ase_nolowcounts.CT.G.all$"HybHE/HybHT_padj"<=0.05 & ase_nolowcounts.CT.G.all$"TT_pHE/pHTvsHybHE/HybHT_padj"<=0.05)

#all trans: P1  P2; A1 = A2; P1/P2  A1/A2
ase_nolowcounts.CT.G.alltrans<-ase_nolowcounts.CT.G.all %>% dplyr::filter(ase_nolowcounts.CT.G.all$"TT_HEvsHTpadj"<=0.05 & ase_nolowcounts.CT.G.all$"HybHE/HybHT_padj">0.05 & ase_nolowcounts.CT.G.all$"TT_pHE/pHTvsHybHE/HybHT_padj"<=0.05)

#all cis: P1  P2; A1  A2; P1/P2 = A1/A2
ase_nolowcounts.CT.G.allcis<-ase_nolowcounts.CT.G.all %>% dplyr::filter(ase_nolowcounts.CT.G.all$"TT_HEvsHTpadj"<=0.05 & ase_nolowcounts.CT.G.all$"HybHE/HybHT_padj"<=0.05 & ase_nolowcounts.CT.G.all$"TT_pHE/pHTvsHybHE/HybHT_padj">0.05)

#cis + trans: P1  P2; A1  A2; P1/P2 > A1/A2
ase_nolowcounts.CT.G.CplusT<-ase_nolowcounts.CT.G.all %>% dplyr::filter(ase_nolowcounts.CT.G.all$"TT_HEvsHTpadj"<=0.05 & ase_nolowcounts.CT.G.all$"HybHE/HybHT_padj"<=0.05 & ase_nolowcounts.CT.G.all$"TT_pHE/pHTvsHybHE/HybHT_padj"<=0.05 & `log2FC_TT_pHE/pHTvsHybHE/HybHT`<0)

#cis x trans: P1  P2; A1  A2; P1/P2 < A1/A2
ase_nolowcounts.CT.G.CbyT<-ase_nolowcounts.CT.G.all %>% dplyr::filter(ase_nolowcounts.CT.G.all$"TT_HEvsHTpadj"<=0.05 & ase_nolowcounts.CT.G.all$"HybHE/HybHT_padj"<=0.05 & ase_nolowcounts.CT.G.all$"TT_pHE/pHTvsHybHE/HybHT_padj"<=0.05 & `log2FC_TT_pHE/pHTvsHybHE/HybHT`>0)

#ambiguous: P1 = P2; A1 = A2; P1/P2  A1/A2 OR
ase_nolowcounts.CT.G.ambig<-ase_nolowcounts.CT.G.all %>% dplyr::filter(ase_nolowcounts.CT.G.all$"TT_HEvsHTpadj">0.05 & ase_nolowcounts.CT.G.all$"HybHE/HybHT_padj">0.05 & ase_nolowcounts.CT.G.all$"TT_pHE/pHTvsHybHE/HybHT_padj"<=0.05)
#P1 = P2 A1  A2 P1/P2 = A1/A2 OR
ase_nolowcounts.CT.G.ambig<-rbind(ase_nolowcounts.CT.G.ambig, ase_nolowcounts.CT.G.all %>% dplyr::filter(ase_nolowcounts.CT.G.all$"TT_HEvsHTpadj">0.05 & ase_nolowcounts.CT.G.all$"HybHE/HybHT_padj"<=0.05 & ase_nolowcounts.CT.G.all$"TT_pHE/pHTvsHybHE/HybHT_padj">0.05))
#P1  P2 A1 = A2 P1/P2 = A1/A2
ase_nolowcounts.CT.G.ambig<-rbind(ase_nolowcounts.CT.G.ambig, ase_nolowcounts.CT.G.all %>% dplyr::filter(ase_nolowcounts.CT.G.all$"TT_HEvsHTpadj"<=0.05 & ase_nolowcounts.CT.G.all$"HybHE/HybHT_padj">0.05 & ase_nolowcounts.CT.G.all$"TT_pHE/pHTvsHybHE/HybHT_padj">0.05))


ase_nolowcounts.CT.test.res$G<-as.numeric(c(nrow(ase_nolowcounts.CT.G.conserved),nrow(ase_nolowcounts.CT.G.comp),nrow(ase_nolowcounts.CT.G.alltrans),nrow(ase_nolowcounts.CT.G.allcis),nrow(ase_nolowcounts.CT.G.CplusT),nrow(ase_nolowcounts.CT.G.CbyT),nrow(ase_nolowcounts.CT.G.ambig)))
```

#LARVA ase_nolowcounts.CT TESTS
```{r}
#Larva ase_nolowcounts.CT Tests and assoc'd DEseq
ase_nolowcounts.CTliftedHyb_readcounts_Lonly<-(ase_nolowcounts %>% 
                               select(HybHEL1,HybHEL2,HybHEL11,HybHTL1,HybHTL2,HybHTL11))
rownames(ase_nolowcounts.CTliftedHyb_readcounts_Lonly)<-paste0(ase_nolowcounts$Chr,":",ase_nolowcounts$start, "-",ase_nolowcounts$end)
ase_nolowcounts.CTliftedHyb_readcounts_Lonly<-round(ase_nolowcounts.CTliftedHyb_readcounts_Lonly)

ase_nolowcounts.CT_liftedHyb.L.desc<-data.frame(individual=factor(c(1,2,11,1,2,11)),
                  allele=factor(c("HybHE","HybHE","HybHE","HybHT","HybHT","HybHT")))
ase_nolowcounts.CT_liftedHyb_Lonlydds<-DESeqDataSetFromMatrix(ase_nolowcounts.CTliftedHyb_readcounts_Lonly,
                                        ase_nolowcounts.CT_liftedHyb.L.desc,
                                        design = model.matrix(~0+ase_nolowcounts.CT_liftedHyb.L.desc$individual+ase_nolowcounts.CT_liftedHyb.L.desc$allele))
sizeFactors(ase_nolowcounts.CT_liftedHyb_Lonlydds)<-rep(1,6) #setting size factors to 1 so library size isn't controlled btwn hyb reads since they come from same sample #number of reps should be number of samples
ase_nolowcounts.CT_liftedHyb_Lonlydds<-DESeq(ase_nolowcounts.CT_liftedHyb_Lonlydds)
resultsNames(ase_nolowcounts.CT_liftedHyb_Lonlydds) #negative means higher in hybHE than hybHT

ase_nolowcounts.CTliftedTT_readcounts_Lonly<-ase_nolowcounts
rownames(ase_nolowcounts.CTliftedTT_readcounts_Lonly)<-paste0(ase_nolowcounts$Chr,":",ase_nolowcounts$start, "-",ase_nolowcounts$end)
ase_nolowcounts.CTliftedTT_readcounts_Lonly[4:43]<-ase_nolowcounts.CTliftedTT_readcounts_Lonly[4:43]+1
ase_nolowcounts.CTliftedTT_readcounts_Lonly$"HybHEL1/HybHTL1"<-ase_nolowcounts.CTliftedTT_readcounts_Lonly$HybHEL1/ase_nolowcounts.CTliftedTT_readcounts_Lonly$HybHTL1
ase_nolowcounts.CTliftedTT_readcounts_Lonly$"HybHEL2/HybHTL2"<-ase_nolowcounts.CTliftedTT_readcounts_Lonly$HybHEL2/ase_nolowcounts.CTliftedTT_readcounts_Lonly$HybHTL2
ase_nolowcounts.CTliftedTT_readcounts_Lonly$"HybHEL11/HybHTL11"<-ase_nolowcounts.CTliftedTT_readcounts_Lonly$HybHEL11/ase_nolowcounts.CTliftedTT_readcounts_Lonly$HybHTL11
ase_nolowcounts.CTliftedTT_readcounts_Lonly$"HeryL3/HtubL3"<-(ase_nolowcounts.CTliftedTT_readcounts_Lonly$HeryL3/sum(ase_nolowcounts.CTliftedTT_readcounts_Lonly$HeryL3))/(ase_nolowcounts.CTliftedTT_readcounts_Lonly$htubL3/sum(ase_nolowcounts.CTliftedTT_readcounts_Lonly$htubL3))
ase_nolowcounts.CTliftedTT_readcounts_Lonly$"HeryL3/HtubL6"<-(ase_nolowcounts.CTliftedTT_readcounts_Lonly$HeryL3/sum(ase_nolowcounts.CTliftedTT_readcounts_Lonly$HeryL3))/(ase_nolowcounts.CTliftedTT_readcounts_Lonly$htubL6/sum(ase_nolowcounts.CTliftedTT_readcounts_Lonly$htubL6))
ase_nolowcounts.CTliftedTT_readcounts_Lonly$"HeryL3/HtubL11"<-(ase_nolowcounts.CTliftedTT_readcounts_Lonly$HeryL3/sum(ase_nolowcounts.CTliftedTT_readcounts_Lonly$HeryL3))/(ase_nolowcounts.CTliftedTT_readcounts_Lonly$htubL11/sum(ase_nolowcounts.CTliftedTT_readcounts_Lonly$htubL11))
ase_nolowcounts.CTliftedTT_readcounts_Lonly$"HeryL6/HtubL3"<-(ase_nolowcounts.CTliftedTT_readcounts_Lonly$HeryL6/sum(ase_nolowcounts.CTliftedTT_readcounts_Lonly$HeryL6))/(ase_nolowcounts.CTliftedTT_readcounts_Lonly$htubL3/sum(ase_nolowcounts.CTliftedTT_readcounts_Lonly$htubL3))
ase_nolowcounts.CTliftedTT_readcounts_Lonly$"HeryL6/HtubL6"<-(ase_nolowcounts.CTliftedTT_readcounts_Lonly$HeryL6/sum(ase_nolowcounts.CTliftedTT_readcounts_Lonly$HeryL6))/(ase_nolowcounts.CTliftedTT_readcounts_Lonly$htubL6/sum(ase_nolowcounts.CTliftedTT_readcounts_Lonly$htubL6))
ase_nolowcounts.CTliftedTT_readcounts_Lonly$"HeryL6/HtubL11"<-(ase_nolowcounts.CTliftedTT_readcounts_Lonly$HeryL6/sum(ase_nolowcounts.CTliftedTT_readcounts_Lonly$HeryL6))/(ase_nolowcounts.CTliftedTT_readcounts_Lonly$htubL11/sum(ase_nolowcounts.CTliftedTT_readcounts_Lonly$htubL11))
ase_nolowcounts.CTliftedTT_readcounts_Lonly$"HeryL9/HtubL3"<-(ase_nolowcounts.CTliftedTT_readcounts_Lonly$HeryL9/sum(ase_nolowcounts.CTliftedTT_readcounts_Lonly$HeryL9))/(ase_nolowcounts.CTliftedTT_readcounts_Lonly$htubL3/sum(ase_nolowcounts.CTliftedTT_readcounts_Lonly$htubL3))
ase_nolowcounts.CTliftedTT_readcounts_Lonly$"HeryL9/HtubL6"<-(ase_nolowcounts.CTliftedTT_readcounts_Lonly$HeryL9/sum(ase_nolowcounts.CTliftedTT_readcounts_Lonly$HeryL9))/(ase_nolowcounts.CTliftedTT_readcounts_Lonly$htubL6/sum(ase_nolowcounts.CTliftedTT_readcounts_Lonly$htubL6))
ase_nolowcounts.CTliftedTT_readcounts_Lonly$"HeryL9/HtubL11"<-(ase_nolowcounts.CTliftedTT_readcounts_Lonly$HeryL9/sum(ase_nolowcounts.CTliftedTT_readcounts_Lonly$HeryL9))/(ase_nolowcounts.CTliftedTT_readcounts_Lonly$htubL11/sum(ase_nolowcounts.CTliftedTT_readcounts_Lonly$htubL11))
ase_nolowcounts.CTliftedTT_readcounts_Lonly<-ase_nolowcounts.CTliftedTT_readcounts_Lonly[c(44:55)]
ase_nolowcounts.CTliftedTT_readcounts_Lonly<-ase_nolowcounts.CTliftedTT_readcounts_Lonly*100
ase_nolowcounts.CTliftedTT_readcounts_Lonly<-round(ase_nolowcounts.CTliftedTT_readcounts_Lonly) #deseq makes me put in integers
ase_nolowcounts.CTliftedTT_readcounts_Lonly_desc<-data_frame(Sample=colnames(ase_nolowcounts.CTliftedTT_readcounts_Lonly),
                                             Origin=as.factor(c("HybvHyb","HybvHyb","HybvHyb",rep("pHEvpHT",9))))

ase_nolowcounts.CTliftedTT_readcounts_Lonlydds<-DESeqDataSetFromMatrix(ase_nolowcounts.CTliftedTT_readcounts_Lonly,
                                        ase_nolowcounts.CTliftedTT_readcounts_Lonly_desc,
                                        design = ~Origin)
ase_nolowcounts.CTliftedTT_readcounts_Lonlydds<-DESeq(ase_nolowcounts.CTliftedTT_readcounts_Lonlydds)
resultsNames(ase_nolowcounts.CTliftedTT_readcounts_Lonlydds)

#need to lfcshrink ase_nolowcounts.CT_liftedHyb_Lonlydds and ase_nolowcounts.CTliftedTT_readcounts_Lonlydds:
lfcase_nolowcounts.CT_liftedHyb_Lonlyres<-lfcShrink(ase_nolowcounts.CT_liftedHyb_Lonlydds,coef="ase_nolowcounts.CT_liftedHyb.L.desc.alleleHybHT")
lfcase_nolowcounts.CTliftedTT_readcounts_Lonlyres<-lfcShrink(ase_nolowcounts.CTliftedTT_readcounts_Lonlydds, coef="Origin_pHEvpHT_vs_HybvHyb")
lfcase_nolowcounts.CT_liftedHyb_Lonlyres<-as.data.frame(lfcase_nolowcounts.CT_liftedHyb_Lonlyres)
lfcase_nolowcounts.CT_liftedHyb_Lonlyres[is.na(lfcase_nolowcounts.CT_liftedHyb_Lonlyres)]<-1
lfcase_nolowcounts.CTliftedTT_readcounts_Lonlyres<-as.data.frame(lfcase_nolowcounts.CTliftedTT_readcounts_Lonlyres)
lfcase_nolowcounts.CTliftedTT_readcounts_Lonlyres[is.na(lfcase_nolowcounts.CTliftedTT_readcounts_Lonlyres)]<-1

ase_nolowcounts.CT.hyb.L<-lfcase_nolowcounts.CT_liftedHyb_Lonlyres %>% select("log2FChybHE/hybHT"=log2FoldChange,"HybHE/HybHT_padj"=padj)
ase_nolowcounts.CT.TT.L<-lfcase_nolowcounts.CTliftedTT_readcounts_Lonlyres %>% select("log2FC_TT_pHE/pHTvsHybHE/HybHT"=log2FoldChange,"TT_pHE/pHTvsHybHE/HybHT_padj"=padj)
Ct.TT.HEvsHT<-ase_nolowcounts.lfcShrink_Lonly_Htub_vs_Hery_res %>% select("log2FC_TT_HEvsHT"=log2FoldChange, "TT_HEvsHTpadj"=padj)
ase_nolowcounts.CT.L.all<-cbind(ase_nolowcounts.CT.hyb.L,ase_nolowcounts.CT.TT.L,Ct.TT.HEvsHT)

#doing the actual tests:
#conserved: P1 = P2; A1 = A2; P1/P2 = A1/A2
ase_nolowcounts.CT.L.conserved<-ase_nolowcounts.CT.L.all %>% dplyr::filter(ase_nolowcounts.CT.L.all$"TT_HEvsHTpadj">0.05 & ase_nolowcounts.CT.L.all$"HybHE/HybHT_padj">0.05 & ase_nolowcounts.CT.L.all$"TT_pHE/pHTvsHybHE/HybHT_padj">0.05)

#compensatory: P1 = P2; A1  A2; P1/P2  A1/A2
ase_nolowcounts.CT.L.comp<-ase_nolowcounts.CT.L.all %>% dplyr::filter(ase_nolowcounts.CT.L.all$"TT_HEvsHTpadj">0.05 & ase_nolowcounts.CT.L.all$"HybHE/HybHT_padj"<=0.05 & ase_nolowcounts.CT.L.all$"TT_pHE/pHTvsHybHE/HybHT_padj"<=0.05)

#all trans: P1  P2; A1 = A2; P1/P2  A1/A2
ase_nolowcounts.CT.L.alltrans<-ase_nolowcounts.CT.L.all %>% dplyr::filter(ase_nolowcounts.CT.L.all$"TT_HEvsHTpadj"<=0.05 & ase_nolowcounts.CT.L.all$"HybHE/HybHT_padj">0.05 & ase_nolowcounts.CT.L.all$"TT_pHE/pHTvsHybHE/HybHT_padj"<=0.05)

#all cis: P1  P2; A1  A2; P1/P2 = A1/A2
ase_nolowcounts.CT.L.allcis<-ase_nolowcounts.CT.L.all %>% dplyr::filter(ase_nolowcounts.CT.L.all$"TT_HEvsHTpadj"<=0.05 & ase_nolowcounts.CT.L.all$"HybHE/HybHT_padj"<=0.05 & ase_nolowcounts.CT.L.all$"TT_pHE/pHTvsHybHE/HybHT_padj">0.05)

#cis + trans: P1  P2; A1  A2; P1/P2 > A1/A2
ase_nolowcounts.CT.L.CplusT<-ase_nolowcounts.CT.L.all %>% dplyr::filter(ase_nolowcounts.CT.L.all$"TT_HEvsHTpadj"<=0.05 & ase_nolowcounts.CT.L.all$"HybHE/HybHT_padj"<=0.05 & ase_nolowcounts.CT.L.all$"TT_pHE/pHTvsHybHE/HybHT_padj"<=0.05 & `log2FC_TT_pHE/pHTvsHybHE/HybHT`<0)

#cis x trans: P1  P2; A1  A2; P1/P2 < A1/A2
ase_nolowcounts.CT.L.CbyT<-ase_nolowcounts.CT.L.all %>% dplyr::filter(ase_nolowcounts.CT.L.all$"TT_HEvsHTpadj"<=0.05 & ase_nolowcounts.CT.L.all$"HybHE/HybHT_padj"<=0.05 & ase_nolowcounts.CT.L.all$"TT_pHE/pHTvsHybHE/HybHT_padj"<=0.05 & `log2FC_TT_pHE/pHTvsHybHE/HybHT`>0)

#ambiguous: P1 = P2; A1 = A2; P1/P2  A1/A2 OR
ase_nolowcounts.CT.L.ambig<-ase_nolowcounts.CT.L.all %>% dplyr::filter(ase_nolowcounts.CT.L.all$"TT_HEvsHTpadj">0.05 & ase_nolowcounts.CT.L.all$"HybHE/HybHT_padj">0.05 & ase_nolowcounts.CT.L.all$"TT_pHE/pHTvsHybHE/HybHT_padj"<=0.05)
#P1 = P2 A1  A2 P1/P2 = A1/A2 OR
ase_nolowcounts.CT.L.ambig<-rbind(ase_nolowcounts.CT.L.ambig, ase_nolowcounts.CT.L.all %>% dplyr::filter(ase_nolowcounts.CT.L.all$"TT_HEvsHTpadj">0.05 & ase_nolowcounts.CT.L.all$"HybHE/HybHT_padj"<=0.05 & ase_nolowcounts.CT.L.all$"TT_pHE/pHTvsHybHE/HybHT_padj">0.05))
#P1  P2 A1 = A2 P1/P2 = A1/A2
ase_nolowcounts.CT.L.ambig<-rbind(ase_nolowcounts.CT.L.ambig, ase_nolowcounts.CT.L.all %>% dplyr::filter(ase_nolowcounts.CT.L.all$"TT_HEvsHTpadj"<=0.05 & ase_nolowcounts.CT.L.all$"HybHE/HybHT_padj">0.05 & ase_nolowcounts.CT.L.all$"TT_pHE/pHTvsHybHE/HybHT_padj">0.05))

ase_nolowcounts.CT.test.res$L<-as.numeric(c(nrow(ase_nolowcounts.CT.L.conserved),nrow(ase_nolowcounts.CT.L.comp),nrow(ase_nolowcounts.CT.L.alltrans),nrow(ase_nolowcounts.CT.L.allcis),nrow(ase_nolowcounts.CT.L.CplusT),nrow(ase_nolowcounts.CT.L.CbyT),nrow(ase_nolowcounts.CT.L.ambig)))
```
```{r}
ase_nolowcountsCT.test.res<-as.data.frame(ase_nolowcounts.CT.test.res)
ase_nolowcountsCT.test.res<-melt(ase_nolowcountsCT.test.res)
ase_nolowcountsCT.test.res.plot<-ggplot(ase_nolowcountsCT.test.res, aes(x=variable, y=value,color=Type, group=Type))+
  geom_line(size=2)+theme_classic()+theme(panel.background = element_rect(fill = "transparent",colour = NA), plot.background = element_rect(fill = "transparent",colour = NA), legend.background = element_rect(fill = "transparent",colour = NA))+ 
  xlab("") + ylab("Number of Peaks") + ggtitle("") + theme(plot.title = element_text(size=24, vjust = 1.5, face = "bold"), axis.text = element_text(size=20), axis.title=element_text(size=22),legend.text = element_text(size=19), legend.title=element_text(size=19), legend.key = element_rect(size = 5, colour = NA), legend.key.size = unit(1.5, 'lines'))
ase_nolowcountsCT.test.res.plot
#ggsave("./ase_nolowcountsCT.test.res.plot.pdf", ase_nolowcountsCT.test.res.plot, width=8, height=6)

ase_nolowcountsCT.test.res.prop<-as.data.frame(ase_nolowcounts.CT.test.res)
#make into proportions:
ase_nolowcountsCT.test.res.prop$B<-ase_nolowcountsCT.test.res.prop$B/sum(ase_nolowcountsCT.test.res.prop$B)
ase_nolowcountsCT.test.res.prop$G<-ase_nolowcountsCT.test.res.prop$G/sum(ase_nolowcountsCT.test.res.prop$G)
ase_nolowcountsCT.test.res.prop$L<-ase_nolowcountsCT.test.res.prop$L/sum(ase_nolowcountsCT.test.res.prop$L)
ase_nolowcountsCT.test.res.prop<-melt(ase_nolowcountsCT.test.res.prop)
ase_nolowcountsCT.test.res.prop.plot<-ggplot(ase_nolowcountsCT.test.res.prop, aes(x=variable, y=value,color=Type, group=Type))+
  geom_line(size=2)+theme_classic()+theme(panel.background = element_rect(fill = "transparent",colour = NA), plot.background = element_rect(fill = "transparent",colour = NA), legend.background = element_rect(fill = "transparent",colour = NA))+ 
  xlab("") + ylab("Proportion of Peaks") + ggtitle("") + theme(plot.title = element_text(size=24, vjust = 1.5, face = "bold"), axis.text = element_text(size=20), axis.title=element_text(size=22),legend.text = element_text(size=19), legend.title=element_text(size=19), legend.key = element_rect(size = 5, colour = NA), legend.key.size = unit(1.5, 'lines'))+
  ylim(0,1.0)
ase_nolowcountsCT.test.res.prop.plot
#ggsave("./ase_nolowcountsCT.test.res.prop.plot.pdf", ase_nolowcountsCT.test.res.prop.plot, width=8, height=6)
```

```{r}
#What percent of peaks are DA between species across all stages?
Hery_Htub.DA.B<-ase_nolowcounts.lfcShrink_Bonly_Htub_vs_Hery_res %>% dplyr::filter(padj<=0.05)
Hery_Htub.DA.G<-ase_nolowcounts.lfcShrink_Gonly_Htub_vs_Hery_res %>% dplyr::filter(padj<=0.05)
Hery_Htub.DA.L<-ase_nolowcounts.lfcShrink_Lonly_Htub_vs_Hery_res %>% dplyr::filter(padj<=0.05)

Hery_Htub.DA.B$Peak<-row.names(Hery_Htub.DA.B)
Hery_Htub.DA.G$Peak<-row.names(Hery_Htub.DA.G)
Hery_Htub.DA.L$Peak<-row.names(Hery_Htub.DA.L)

Hery_Htub.DA.BGL<-Hery_Htub.DA.B %>% select(Peak,padj)
Hery_Htub.DA.BGL<-Hery_Htub.DA.BGL %>% dplyr::rename(B_padj=padj)
Hery_Htub.DA.BGL<-left_join(Hery_Htub.DA.BGL, Hery_Htub.DA.G %>% select(Peak, padj) %>% dplyr::rename(G_padj=padj))
Hery_Htub.DA.BGL<-left_join(Hery_Htub.DA.BGL, Hery_Htub.DA.L %>% select(Peak, padj) %>% dplyr::rename(L_padj=padj))

#View(Hery_Htub.DA.BGL) #37,010 peaks, out of 117,391 total
```


```{r}
#Build data frame of inh classification for each peak at each stage:
ase_nolowcounts.B_inhconserved.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.B_conserved), Inh.B=rep("conserved"))
ase_nolowcounts.B_underdom.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.B_underdom), Inh.B=rep("underdom"))
ase_nolowcounts.B_overdom.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.B_overdom), Inh.B=rep("overdom"))
ase_nolowcounts.B_HEdom.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.B_HEdom), Inh.B=rep("HEdom"))
ase_nolowcounts.B_HTdom.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.B_HTdom), Inh.B=rep("HTdom"))
ase_nolowcounts.B_additive.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.B_additive), Inh.B=rep("additive"))
ase_nolowcounts.B_inhambiguous.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.B_ambig), Inh.B=rep("ambig"))

ase_nolowcounts.B_inh.peaks<-rbind(ase_nolowcounts.B_inhconserved.peaks,ase_nolowcounts.B_underdom.peaks,ase_nolowcounts.B_overdom.peaks,ase_nolowcounts.B_HEdom.peaks,ase_nolowcounts.B_HTdom.peaks,ase_nolowcounts.B_additive.peaks,ase_nolowcounts.B_inhambiguous.peaks)

ase_nolowcounts.G_inhconserved.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.G_conserved), Inh.G=rep("conserved"))
ase_nolowcounts.G_underdom.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.G_underdom), Inh.G=rep("underdom"))
ase_nolowcounts.G_overdom.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.G_overdom), Inh.G=rep("overdom"))
ase_nolowcounts.G_HEdom.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.G_HEdom), Inh.G=rep("HEdom"))
ase_nolowcounts.G_HTdom.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.G_HTdom), Inh.G=rep("HTdom"))
ase_nolowcounts.G_additive.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.G_additive), Inh.G=rep("additive"))
ase_nolowcounts.G_inhambiguous.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.G_ambig), Inh.G=rep("ambig"))

ase_nolowcounts.G_inh.peaks<-rbind(ase_nolowcounts.G_inhconserved.peaks,ase_nolowcounts.G_underdom.peaks,ase_nolowcounts.G_overdom.peaks,ase_nolowcounts.G_HEdom.peaks,ase_nolowcounts.G_HTdom.peaks,ase_nolowcounts.G_additive.peaks,ase_nolowcounts.G_inhambiguous.peaks)

ase_nolowcounts.L_inhconserved.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.L_conserved), Inh.L=rep("conserved"))
ase_nolowcounts.L_underdom.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.L_underdom), Inh.L=rep("underdom"))
ase_nolowcounts.L_overdom.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.L_overdom), Inh.L=rep("overdom"))
ase_nolowcounts.L_HEdom.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.L_HEdom), Inh.L=rep("HEdom"))
ase_nolowcounts.L_HTdom.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.L_HTdom), Inh.L=rep("HTdom"))
ase_nolowcounts.L_additive.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.L_additive), Inh.L=rep("additive"))
ase_nolowcounts.L_inhambiguous.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.L_ambig), Inh.L=rep("ambig"))

ase_nolowcounts.L_inh.peaks<-rbind(ase_nolowcounts.L_inhconserved.peaks,ase_nolowcounts.L_underdom.peaks,ase_nolowcounts.L_overdom.peaks,ase_nolowcounts.L_HEdom.peaks,ase_nolowcounts.L_HTdom.peaks,ase_nolowcounts.L_additive.peaks,ase_nolowcounts.L_inhambiguous.peaks)

ase_nolowcounts.inh.peaks.all<-full_join(ase_nolowcounts.B_inh.peaks,ase_nolowcounts.G_inh.peaks, by="PeakCoord")
ase_nolowcounts.inh.peaks.all<-full_join(ase_nolowcounts.inh.peaks.all, ase_nolowcounts.L_inh.peaks, by="PeakCoord")
```
```{r}
#Build data frame of CT classification for each peak at each stage:
ase_nolowcounts.B_CTconserved.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.CT.B.conserved), CT.B=rep("conserved"))
ase_nolowcounts.B_allcis.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.CT.B.allcis), CT.B=rep("allcis"))
ase_nolowcounts.B_alltrans.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.CT.B.alltrans), CT.B=rep("alltrans"))
ase_nolowcounts.B_CplusT.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.CT.B.CplusT), CT.B=rep("CplusT"))
ase_nolowcounts.B_CbyT.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.CT.B.CbyT), CT.B=rep("CbyT"))
ase_nolowcounts.B_comp.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.CT.B.comp), CT.B=rep("comp"))
ase_nolowcounts.B_CTambig.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.CT.B.ambig), CT.B=rep("ambig"))


ase_nolowcounts.B_CT.peaks<-rbind(ase_nolowcounts.B_CTconserved.peaks,ase_nolowcounts.B_allcis.peaks,ase_nolowcounts.B_alltrans.peaks,ase_nolowcounts.B_CplusT.peaks,ase_nolowcounts.B_CbyT.peaks,ase_nolowcounts.B_comp.peaks,ase_nolowcounts.B_CTambig.peaks)

ase_nolowcounts.G_CTconserved.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.CT.G.conserved), CT.G=rep("conserved"))
ase_nolowcounts.G_allcis.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.CT.G.allcis), CT.G=rep("allcis"))
ase_nolowcounts.G_alltrans.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.CT.G.alltrans), CT.G=rep("alltrans"))
ase_nolowcounts.G_CplusT.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.CT.G.CplusT), CT.G=rep("CplusT"))
ase_nolowcounts.G_CbyT.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.CT.G.CbyT), CT.G=rep("CbyT"))
ase_nolowcounts.G_comp.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.CT.G.comp), CT.G=rep("comp"))
ase_nolowcounts.G_CTambig.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.CT.G.ambig), CT.G=rep("ambig"))

ase_nolowcounts.G_CT.peaks<-rbind(ase_nolowcounts.G_CTconserved.peaks,ase_nolowcounts.G_allcis.peaks,ase_nolowcounts.G_alltrans.peaks,ase_nolowcounts.G_CplusT.peaks,ase_nolowcounts.G_CbyT.peaks,ase_nolowcounts.G_comp.peaks,ase_nolowcounts.G_CTambig.peaks)

ase_nolowcounts.L_CTconserved.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.CT.L.conserved), CT.L=rep("conserved"))
ase_nolowcounts.L_allcis.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.CT.L.allcis), CT.L=rep("allcis"))
ase_nolowcounts.L_alltrans.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.CT.L.alltrans), CT.L=rep("alltrans"))
ase_nolowcounts.L_CplusT.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.CT.L.CplusT), CT.L=rep("CplusT"))
ase_nolowcounts.L_CbyT.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.CT.L.CbyT), CT.L=rep("CbyT"))
ase_nolowcounts.L_comp.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.CT.L.comp), CT.L=rep("comp"))
ase_nolowcounts.L_CTambig.peaks<-data_frame(PeakCoord=rownames(ase_nolowcounts.CT.L.ambig), CT.L=rep("ambig"))

ase_nolowcounts.L_CT.peaks<-rbind(ase_nolowcounts.L_CTconserved.peaks,ase_nolowcounts.L_allcis.peaks,ase_nolowcounts.L_alltrans.peaks,ase_nolowcounts.L_CplusT.peaks,ase_nolowcounts.L_CbyT.peaks,ase_nolowcounts.L_comp.peaks,ase_nolowcounts.L_CTambig.peaks)
ase_nolowcounts.CT.peaks.all<-full_join(ase_nolowcounts.B_CT.peaks,ase_nolowcounts.G_CT.peaks, by="PeakCoord")
ase_nolowcounts.CT.peaks.all<-full_join(ase_nolowcounts.CT.peaks.all, ase_nolowcounts.L_CT.peaks, by="PeakCoord")

```

Alluvial plots (Supp. Figure 4)
```{r}
library(ggalluvial)
ase_nolowcounts.inh.peaks.all<-as.data.frame(ase_nolowcounts.inh.peaks.all)
ase_nolowcounts.inh.peaks.all.melt<-melt(ase_nolowcounts.inh.peaks.all,id="PeakCoord")
#rename to game the system to make alluvial plot:
ase_nolowcounts.inh.peaks.all.melt$rename<-case_when(
  ase_nolowcounts.inh.peaks.all.melt$value == "conserved"~"Z"
  ,ase_nolowcounts.inh.peaks.all.melt$value == "HEdom"~"A"
  ,ase_nolowcounts.inh.peaks.all.melt$value == "HTdom"~"B"
  ,ase_nolowcounts.inh.peaks.all.melt$value == "additive" ~ "C"
  ,ase_nolowcounts.inh.peaks.all.melt$value == "overdom" ~ "D"
  ,ase_nolowcounts.inh.peaks.all.melt$value == "underdom" ~ "E"
)
ase_nolowcounts.inh.peaks.all.melt$value<-paste0(ase_nolowcounts.inh.peaks.all.melt$rename,ase_nolowcounts.inh.peaks.all.melt$value)
ase_nolowcounts.inh.peaks.all.melt<-ase_nolowcounts.inh.peaks.all.melt %>% select(-rename)

inhcolors<- c("#E41A1C","#007857","#E2E21B","#08F4F4","#F2AAED","#A09C98")

ase_nolowcounts.inh.alluvial<-ggplot(ase_nolowcounts.inh.peaks.all.melt %>% dplyr::filter(!value=="NAambig"),aes(x=variable,stratum=value,alluvium=PeakCoord,fill=value, label=value))+
  geom_flow()+
  geom_stratum(alpha=.8)+
  geom_text(stat = "stratum", aes(label = after_stat(stratum)))+
  scale_fill_manual(values=inhcolors)+
  theme_classic()+
  theme(legend.position='none')

#ggsave("./ase.inh.alluvial.pdf",ase_nolowcounts.inh.alluvial,width=12,height=8,units="in")
ase_nolowcounts.CT.peaks.all<-as.data.frame(ase_nolowcounts.CT.peaks.all)
ase_nolowcounts.CT.peaks.all.melt<-melt(ase_nolowcounts.CT.peaks.all,id="PeakCoord")
ase_nolowcounts.CT.peaks.all.melt$value<-gsub("CbyT", "cis by trans",ase_nolowcounts.CT.peaks.all.melt$value)
ase_nolowcounts.CT.peaks.all.melt$value<-gsub("CplusT", "cis plus trans",ase_nolowcounts.CT.peaks.all.melt$value)

customregorder<-c('conserved', 'alltrans', 'allcis', 'cis by trans', 'cis plus trans', 'comp','ambig')
ase_nolowcounts.CT.peaks.all.melt<-ase_nolowcounts.CT.peaks.all.melt%>% arrange(factor(value, levels=customregorder))

regcolors<- c("#A09C98","#58B8F9","#CC3770","#2B1066","#7A7AC4","#984EA3")
regcolors<- c("#CC3770","#58B8F9","#2B1066","#7A7AC4","#984EA3","#A09C98")

ase_nolowcounts.CT.alluvial<-ggplot(ase_nolowcounts.CT.peaks.all.melt %>% dplyr::filter(!value=="ambig"),aes(x=variable,stratum=value,alluvium=PeakCoord,fill=value, label=value))+
  geom_flow()+
  geom_stratum(alpha=.8)+
  geom_text(stat = "stratum", aes(label = after_stat(stratum)))+
  scale_fill_manual(values=regcolors)+
  theme_classic()+
  theme(legend.position='none')
#ggsave("./ase.CT.alluvial.pdf",ase_nolowcounts.CT.alluvial,width=12,height=8,units="in")
```

```{r}
#grn genes list:
GRNgeneslist<-read.csv("grngenesfromphilspaper_SPUsonly.csv", header=T)
```

```{r}
B.cis.meanopennesschange<-data.frame(abslog2FC=abs(ase_nolowcounts.CT.B.allcis$log2FC_TT_HEvsHT),
                                        RegulatoryMode=rep("Allcis",length(ase_nolowcounts.CT.B.allcis$log2FC_TT_HEvsHT)))
B.trans.meanopennesschange<-data.frame(abslog2FC=abs(ase_nolowcounts.CT.B.alltrans$log2FC_TT_HEvsHT),
                                        RegulatoryMode=rep("Alltrans",length(ase_nolowcounts.CT.B.alltrans$log2FC_TT_HEvsHT)))
B.cistrans.meanopennesschange<-rbind(B.cis.meanopennesschange,B.trans.meanopennesschange)
B.cistrans.meanopennesschange.plot<-ggplot(B.cistrans.meanopennesschange, aes(x=RegulatoryMode, y=abslog2FC,colour=RegulatoryMode))+
  scale_color_manual(values=c("red","blue"),name = "Regulatory Mode")+
  geom_boxplot(width=0.1,color="black")+
  geom_violin(fill=NA)+
  theme_bw()+
  ylim(0,10)+
  theme(legend.position='none')+
  theme(axis.title.x=element_blank(),
        axis.ticks.x=element_blank(),
        text = element_text(size=20))+
  background_grid(major="none",minor="none")

G.cis.meanopennesschange<-data.frame(abslog2FC=abs(ase_nolowcounts.CT.G.allcis$log2FC_TT_HEvsHT),
                                        RegulatoryMode=rep("Allcis",length(ase_nolowcounts.CT.G.allcis$log2FC_TT_HEvsHT)))
G.trans.meanopennesschange<-data.frame(abslog2FC=abs(ase_nolowcounts.CT.G.alltrans$log2FC_TT_HEvsHT),
                                        RegulatoryMode=rep("Alltrans",length(ase_nolowcounts.CT.G.alltrans$log2FC_TT_HEvsHT)))
G.cistrans.meanopennesschange<-rbind(G.cis.meanopennesschange,G.trans.meanopennesschange)
G.cistrans.meanopennesschange.plot<-ggplot(G.cistrans.meanopennesschange, aes(x=RegulatoryMode, y=abslog2FC, colour=RegulatoryMode))+
  scale_color_manual(values=c("red","blue"),name = "Regulatory Mode")+
  geom_boxplot(width=0.1,color="black")+
  geom_violin(fill=NA)+
  theme_bw()+
  ylim(0,10)+
  theme(legend.position='none')+
  theme(axis.title.x=element_blank(),
        axis.ticks.x=element_blank(),
        text = element_text(size=20))+
  background_grid(major="none",minor="none")


L.cis.meanopennesschange<-data.frame(abslog2FC=abs(ase_nolowcounts.CT.L.allcis$log2FC_TT_HEvsHT),
                                        RegulatoryMode=rep("Allcis",length(ase_nolowcounts.CT.L.allcis$log2FC_TT_HEvsHT)))
L.trans.meanopennesschange<-data.frame(abslog2FC=abs(ase_nolowcounts.CT.L.alltrans$log2FC_TT_HEvsHT),
                                        RegulatoryMode=rep("Alltrans",length(ase_nolowcounts.CT.L.alltrans$log2FC_TT_HEvsHT)))
L.cistrans.meanopennesschange<-rbind(L.cis.meanopennesschange,L.trans.meanopennesschange)

L.cistrans.meanopennesschange.plot<-ggplot(L.cistrans.meanopennesschange, aes(x=RegulatoryMode, y=abslog2FC, colour=RegulatoryMode))+
    scale_color_manual(values=c("red","blue"),name = "Regulatory Mode")+
  geom_boxplot(width=0.1,color="black")+
  geom_violin(fill=NA)+
  ylim(0,10)+
  theme_bw()+
  theme(legend.position='none')+
  theme(axis.title.x=element_blank(),
        axis.ticks.x=element_blank(),
        text = element_text(size=20))+
  background_grid(major="none",minor="none")

#calculate (absolute) mean opennessession change at each stage:
var(B.cis.meanopennesschange)
var(B.trans.meanopennesschange)
var(G.cis.meanopennesschange)
var(G.trans.meanopennesschange)
var(L.cis.meanopennesschange)
var(L.trans.meanopennesschange)
#because the variances of cis vs trans are different, i will use a welch's test of unequal variance to see if the means are statistically different:
t.test(B.cis.meanopennesschange$abslog2FC,B.trans.meanopennesschange$abslog2FC) #stat diff
t.test(G.cis.meanopennesschange$abslog2FC,G.trans.meanopennesschange$abslog2FC) #stat diff
t.test(L.cis.meanopennesschange$abslog2FC,L.trans.meanopennesschange$abslog2FC) #stat diff


#Supp Fig 5A:
#ggsave("B.cistrans.meanopennesschange.plot.pdf",B.cistrans.meanopennesschange.plot,width=8,height=10,units="in")
#ggsave("G.cistrans.meanopennesschange.plot.pdf",G.cistrans.meanopennesschange.plot,width=8,height=10,units="in")
#ggsave("L.cistrans.meanopennesschange.plot.pdf",L.cistrans.meanopennesschange.plot,width=8,height=10,units="in")
```

Supp Fig 5B and 5C:
```{r}
#ALL CT CATEGORIES VOLCANO PLOTS
CT.colors<-c("#CC3770","#58B8F9","#2B1066","#984EA3","#A09C98","#7A7AC4")
B.CT.volc<-ggplot()+
  geom_point(data=inner_join(ase_nolowcounts.CT.B.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.B.all)) ,ase_nolowcounts.CT.peaks.all %>% filter(!CT.B=="ambig")), alpha=1,shape=4,size=.1,aes(x=log2FC_TT_HEvsHT, y=-log10(TT_HEvsHTpadj), colour=CT.B))+
  xlim(-15, 15)+
  ylim(0, 100)+
  ggtitle("Blastula")+
  xlab("log2FC")+
  ylab("-log10(p-adj)")+
  theme_bw()+
  theme(plot.title = element_text(hjust=0.5))+
    theme(legend.position="none")+
  background_grid(major = "none", minor = "none")+
  scale_color_manual(values=CT.colors)
G.CT.volc<-ggplot()+
  geom_point(data=left_join(ase_nolowcounts.CT.G.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.G.all)),ase_nolowcounts.CT.peaks.all), alpha=01,size=.5,aes(x=log2FC_TT_HEvsHT, y=-log10(TT_HEvsHTpadj), colour=CT.G, shape=CT.G))+
  xlim(-15, 15)+
  ylim(0, 100)+
  ggtitle("Gastrula")+
  xlab("log2FC")+
  ylab("-log10(p-adj)")+
  theme_bw()+
  theme(plot.title = element_text(hjust=0.5))+
    theme(legend.position="none")+
  background_grid(major = "none", minor = "none")
L.CT.volc<-ggplot()+
  geom_point(data=left_join(ase_nolowcounts.CT.L.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.L.all)),ase_nolowcounts.CT.peaks.all), alpha=01,size=.5,aes(x=log2FC_TT_HEvsHT, y=-log10(TT_HEvsHTpadj), colour=CT.L, shape=CT.L))+
  xlim(-15, 15)+
  ylim(0, 100)+
  ggtitle("Larva")+
  xlab("log2FC")+
  ylab("-log10(p-adj)")+
  theme_bw()+
  theme(plot.title = element_text(hjust=0.5))+
    theme(legend.position="none")+
  background_grid(major = "none", minor = "none")

#ggsave("BGL.CT.volc.pdf",plot_grid(B.CT.volc, G.CT.volc,L.CT.volc, nrow=3), width=6, height=15)

B.CT.volc.top10<-ggplot()+
  geom_point(data=inner_join(ase_nolowcounts.CT.B.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.B.all)) ,ase_nolowcounts.CT.peaks.all %>% filter(!CT.B=="ambig")), alpha=1,size=.1,aes(x=log2FC_TT_HEvsHT, y=-log10(TT_HEvsHTpadj),colour=CT.B))+
  geom_point(data=inner_join(ase_nolowcounts.CT.B.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.B.all)) ,ase_nolowcounts.CT.peaks.all) %>% mutate(absHEvsHTlog2FC=abs(log2FC_TT_HEvsHT)) %>% arrange(absHEvsHTlog2FC) %>% top_frac(n = -0.9), alpha=1,size=.1,colour="#D3CFCF",aes(x=log2FC_TT_HEvsHT, y=-log10(TT_HEvsHTpadj)))+
  xlim(-15, 15)+
  ylim(0, 100)+
  ggtitle("Blastula")+
  xlab("log2FC")+
  ylab("-log10(p-adj)")+
  theme_bw()+
  theme(plot.title = element_text(hjust=0.5))+
    theme(legend.position="none")+
  background_grid(major = "none", minor = "none")+
  scale_color_manual(values=CT.colors)

#ggsave("B.CT.volc.top10.svg",B.CT.volc.top10, width=6, height=5) #Supp Fig 5B

G.CT.volc.top10<-ggplot()+
  geom_point(data=inner_join(ase_nolowcounts.CT.G.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.G.all)) ,ase_nolowcounts.CT.peaks.all %>% filter(!CT.G=="ambig")), alpha=1,size=.1,aes(x=log2FC_TT_HEvsHT, y=-log10(TT_HEvsHTpadj),colour=CT.G))+
  geom_point(data=inner_join(ase_nolowcounts.CT.G.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.G.all)) ,ase_nolowcounts.CT.peaks.all) %>% mutate(absHEvsHTlog2FC=abs(log2FC_TT_HEvsHT)) %>% arrange(absHEvsHTlog2FC) %>% top_frac(n = -0.9), alpha=1,size=.1,colour="#D3CFCF",aes(x=log2FC_TT_HEvsHT, y=-log10(TT_HEvsHTpadj)))+
  xlim(-15, 15)+
  ylim(0, 100)+
  ggtitle("Gastrula")+
  xlab("log2FC")+
  ylab("-log10(p-adj)")+
  theme_bw()+
  theme(plot.title = element_text(hjust=0.5))+
    theme(legend.position="none")+
  background_grid(major = "none", minor = "none")+
  scale_color_manual(values=CT.colors)

#ggsave("G.CT.volc.top10.pdf",G.CT.volc.top10, width=6, height=5) #Supp Fig 5B

L.CT.volc.top10<-ggplot()+
  geom_point(data=inner_join(ase_nolowcounts.CT.L.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.L.all)) ,ase_nolowcounts.CT.peaks.all %>% filter(!CT.L=="ambig")), alpha=1,size=.1,aes(x=log2FC_TT_HEvsHT, y=-log10(TT_HEvsHTpadj),colour=CT.L))+
  geom_point(data=inner_join(ase_nolowcounts.CT.L.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.L.all)) ,ase_nolowcounts.CT.peaks.all) %>% mutate(absHEvsHTlog2FC=abs(log2FC_TT_HEvsHT)) %>% arrange(absHEvsHTlog2FC) %>% top_frac(n = -0.9), alpha=1,size=.1,colour="#D3CFCF",aes(x=log2FC_TT_HEvsHT, y=-log10(TT_HEvsHTpadj)))+
  xlim(-15, 15)+
  ylim(0, 100)+
  ggtitle("Larva")+
  xlab("log2FC")+
  ylab("-log10(p-adj)")+
  theme_bw()+
  theme(plot.title = element_text(hjust=0.5))+
    theme(legend.position="none")+
  background_grid(major = "none", minor = "none")+
  scale_color_manual(values=CT.colors)

#ggsave("L.CT.volc.top10.pdf",L.CT.volc.top10, width=6, height=5) #Supp Fig 5B

#ggsave("BGL.CT.volc.top10.jpg",plot_grid(B.CT.volc.top10, G.CT.volc.top10,L.CT.volc.top10, nrow=3), width=6, height=15)
# (>0 means higher in Ht than He)

#now do these as bar plots for Supp Fig 5C:
B.CT.bar.top10<-ggplot()+
  geom_histogram(data=inner_join(ase_nolowcounts.CT.B.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.B.all)) ,ase_nolowcounts.CT.peaks.all %>% filter(!CT.B=="ambig"))%>% mutate(absHEvsHTlog2FC=abs(log2FC_TT_HEvsHT))%>% arrange(absHEvsHTlog2FC) %>% top_frac(n = 0.1), alpha=1,size=.1,aes(x=CT.B,fill=CT.B),stat="count")+
  ggtitle("Blastula")+
  theme_bw()+
  theme(plot.title = element_text(hjust=0.5))+
    theme(legend.position="bottom")+
  background_grid(major = "none", minor = "none")+
  scale_fill_manual(values=CT.colors)
G.CT.bar.top10<-ggplot()+
  geom_histogram(data=inner_join(ase_nolowcounts.CT.G.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.G.all)) ,ase_nolowcounts.CT.peaks.all %>% filter(!CT.G=="ambig"))%>% mutate(absHEvsHTlog2FC=abs(log2FC_TT_HEvsHT))%>% arrange(absHEvsHTlog2FC) %>% top_frac(n = 0.1), alpha=1,size=.1,aes(x=CT.G,fill=CT.G),stat="count")+
  ggtitle("Gastrula")+
  theme_bw()+
  theme(plot.title = element_text(hjust=0.5))+
    theme(legend.position="bottom")+
  background_grid(major = "none", minor = "none")+
  scale_fill_manual(values=CT.colors)
L.CT.bar.top10<-ggplot()+
  geom_histogram(data=inner_join(ase_nolowcounts.CT.L.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.L.all)) ,ase_nolowcounts.CT.peaks.all %>% filter(!CT.L=="ambig"))%>% mutate(absHEvsHTlog2FC=abs(log2FC_TT_HEvsHT))%>% arrange(absHEvsHTlog2FC) %>% top_frac(n = 0.1), alpha=1,size=.1,aes(x=CT.L,fill=CT.L),stat="count")+
  ggtitle("Larva")+
  theme_bw()+
  theme(plot.title = element_text(hjust=0.5))+
    theme(legend.position="bottom")+
  background_grid(major = "none", minor = "none")+
  scale_fill_manual(values=CT.colors)

#ggsave("BGL.CT.bar.top10.pdf",plot_grid(B.CT.bar.top10, G.CT.bar.top10,L.CT.bar.top10, nrow=3), width=6, height=15)

#numbers for bar plot (reported in body of paper)
#blastula:
nrow(inner_join(ase_nolowcounts.CT.B.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.B.all)) ,ase_nolowcounts.CT.peaks.all %>% filter(!CT.B=="ambig"))%>% mutate(absHEvsHTlog2FC=abs(log2FC_TT_HEvsHT))%>% arrange(absHEvsHTlog2FC) %>% top_frac(n = 0.1) %>% filter(CT.B=="allcis")) #4523 out of...
nrow(inner_join(ase_nolowcounts.CT.B.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.B.all)) ,ase_nolowcounts.CT.peaks.all %>% filter(!CT.B=="ambig"))%>% mutate(absHEvsHTlog2FC=abs(log2FC_TT_HEvsHT))%>% arrange(absHEvsHTlog2FC) %>% top_frac(n = 0.1))
 #...8995
4523/8995
#Gastrula
nrow(inner_join(ase_nolowcounts.CT.G.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.G.all)) ,ase_nolowcounts.CT.peaks.all %>% filter(!CT.G=="ambig"))%>% mutate(absHEvsHTlog2FC=abs(log2FC_TT_HEvsHT))%>% arrange(absHEvsHTlog2FC) %>% top_frac(n = 0.1) %>% filter(CT.G=="allcis")) #5317 out of...
nrow(inner_join(ase_nolowcounts.CT.G.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.G.all)) ,ase_nolowcounts.CT.peaks.all %>% filter(!CT.G=="ambig"))%>% mutate(absHEvsHTlog2FC=abs(log2FC_TT_HEvsHT))%>% arrange(absHEvsHTlog2FC) %>% top_frac(n = 0.1))
 #...9141
5317/8141

#Larva
nrow(inner_join(ase_nolowcounts.CT.L.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.L.all)) ,ase_nolowcounts.CT.peaks.all %>% filter(!CT.L=="ambig"))%>% mutate(absHEvsHTlog2FC=abs(log2FC_TT_HEvsHT))%>% arrange(absHEvsHTlog2FC) %>% top_frac(n = 0.1) %>% filter(CT.L=="allcis")) #7075 out of...
nrow(inner_join(ase_nolowcounts.CT.L.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.L.all)) ,ase_nolowcounts.CT.peaks.all %>% filter(!CT.L=="ambig"))%>% mutate(absHEvsHTlog2FC=abs(log2FC_TT_HEvsHT))%>% arrange(absHEvsHTlog2FC) %>% top_frac(n = 0.1))
 #...9330
7075/9330

(4523/8995+5317/8141+7075/9330)/3 #an average of 63% of the top peaks were cis
```
```{r}
#labeling genes near peaks
names(Hery_nearestgene_bothdir)[names(Hery_nearestgene_bothdir) == "peak"] <- "peakno"
genesnearpeaks<-left_join(HEliftedpeakcoordsonly %>% select(peakno, HEChr, start, end),  Hery_nearestgene_bothdir, by="peakno")
HEgenes<-read.csv("Hery_master_top_hits_final.csv", header=TRUE)
genesnearpeaks$PeakCoord<-paste0(genesnearpeaks$HEChr,":",genesnearpeaks$start, "-",genesnearpeaks$end)
genesnearpeaks<-genesnearpeaks %>% select(-HEChr, -start, -end)
genesnearpeaks$NearestGene<-gsub("_.*", "",genesnearpeaks$NearestGene)
gene_functions<-read.table("functional_category_table.txt")
gene_functions<-gene_functions %>% select(V1,V2,V3)
colnames(gene_functions)<-c("SPU.hit","Function","Subfunction")
HEgenes<-left_join(HEgenes,gene_functions,by="SPU.hit")

genesnearpeaks<-genesnearpeaks %>% dplyr::rename(H_ery_ID=NearestGene)
genesnearpeaks<-left_join(genesnearpeaks, HEgenes %>% select(H_ery_ID, SPU.hit, SPU.common_name, Function), by="H_ery_ID")
CT.peaks.all.genefunct<-left_join(ase_nolowcounts.CT.peaks.all, genesnearpeaks, by="PeakCoord")
#View(CT.peaks.all.genefunct %>% filter(CT.B=="allcis") %>% filter(CT.G=="allcis") %>% filter(CT.L=="allcis") %>% group_by(Function) %>% tally())

Inh.peaks.all.genefunct<-left_join(ase_nolowcounts.inh.peaks.all, genesnearpeaks, by="PeakCoord")
```

What inh/regulatory mode are peaks near GRN genes?
```{r}
Type<-c("All cis", "All trans", "Ambig", "Cis by trans", "Compensatory", "Conserved", "Cis plus trans")
GRN.CT.peaks.all<-right_join(CT.peaks.all.genefunct,GRNgeneslist, by="SPU.hit")
GRN.CT.peaks.test.B<-GRN.CT.peaks.all %>% group_by(CT.B) %>% tally()
GRN.CT.peaks.test.B<-GRN.CT.peaks.test.B$n
GRN.CT.peaks.test.G<-GRN.CT.peaks.all %>% group_by(CT.G) %>% tally()
GRN.CT.peaks.test.G<-GRN.CT.peaks.test.G$n
GRN.CT.peaks.test.L<-GRN.CT.peaks.all %>% group_by(CT.L) %>% tally()
GRN.CT.peaks.test.L<-GRN.CT.peaks.test.L$n
GRN.CT.peaks.test.res<-cbind(Type,GRN.CT.peaks.test.B,GRN.CT.peaks.test.G,GRN.CT.peaks.test.L)
colnames(GRN.CT.peaks.test.res)<-c("Type","CT.B","CT.G","CT.L")
GRN.CT.peaks.test.res<-GRN.CT.peaks.test.res[c(1:3,7,4:6),]
GRN.CT.peaks.test.res<-as.data.frame(GRN.CT.peaks.test.res)
GRN.CT.peaks.test.res$CT.B<-as.numeric(GRN.CT.peaks.test.res$CT.B)
GRN.CT.peaks.test.res$CT.G<-as.numeric(GRN.CT.peaks.test.res$CT.G)
GRN.CT.peaks.test.res$CT.L<-as.numeric(GRN.CT.peaks.test.res$CT.L)
#make into proportions for direct comparison to all CREs:
GRN.CT.peaks.test.res.prop<-GRN.CT.peaks.test.res$CT.B/sum(GRN.CT.peaks.test.res$CT.B)
GRN.CT.peaks.test.res.prop<-as.data.frame(GRN.CT.peaks.test.res.prop)
colnames(GRN.CT.peaks.test.res.prop)
GRN.CT.peaks.test.res.prop$CT.G<-GRN.CT.peaks.test.res$CT.G/sum(GRN.CT.peaks.test.res$CT.G)
GRN.CT.peaks.test.res.prop$CT.L<-GRN.CT.peaks.test.res$CT.L/sum(GRN.CT.peaks.test.res$CT.L)
GRN.CT.peaks.test.res.prop$Type<-GRN.CT.peaks.test.res$Type
GRN.CT.test.res.prop<-melt(GRN.CT.peaks.test.res.prop, id="Type")
GRN.CT.test.res.prop.plot<-ggplot(GRN.CT.test.res.prop, aes(x=variable, y=value,color=Type, group=Type))+
  geom_line(size=2)+theme_classic()+theme(panel.background = element_rect(fill = "transparent",colour = NA), plot.background = element_rect(fill = "transparent",colour = NA), legend.background = element_rect(fill = "transparent",colour = NA))+ 
  xlab("") + ylab("Proportion of Peaks") + ggtitle("") + theme(plot.title = element_text(size=24, vjust = 1.5, face = "bold"), axis.text = element_text(size=20), axis.title=element_text(size=22),legend.text = element_text(size=19), legend.title=element_text(size=19), legend.key = element_rect(size = 5, colour = NA), legend.key.size = unit(1.5, 'lines'))+
  ylim(0,1.0)
GRN.CT.test.res.prop.plot
#ggsave("./GRN.CT.test.res.plot.pdf", GRN.CT.test.res.plot, width=8, height=6)


GRN.Inh.peaks.all<-right_join(Inh.peaks.all.genefunct,GRNgeneslist, by="SPU.hit")
GRN.Inh.peaks.test.B<-GRN.Inh.peaks.all %>% group_by(Inh.B) %>% tally()
colnames(GRN.Inh.peaks.test.B)<-c("Type", "Inh.B")
GRN.Inh.peaks.test.B[7,1]<-"overdom"
GRN.Inh.peaks.test.B[7,2]<-0
GRN.Inh.peaks.test.B[8,1]<-"underdom"
GRN.Inh.peaks.test.B[8,2]<-0
GRN.Inh.peaks.test.B<-GRN.Inh.peaks.test.B[c(1:5,7,8,6),]
GRN.Inh.peaks.test.G<-GRN.Inh.peaks.all %>% group_by(Inh.G) %>% tally()
GRN.Inh.peaks.test.G<-GRN.Inh.peaks.test.G$n
GRN.Inh.peaks.test.L<-GRN.Inh.peaks.all %>% group_by(Inh.L) %>% tally()
GRN.Inh.peaks.test.L<-GRN.Inh.peaks.test.L$n
GRN.Inh.peaks.test.res<-cbind(GRN.Inh.peaks.test.B,GRN.Inh.peaks.test.G,GRN.Inh.peaks.test.L)
colnames(GRN.Inh.peaks.test.res)<-c("Type","Inh.B","Inh.G","Inh.L")
GRN.Inh.peaks.test.res<-GRN.Inh.peaks.test.res[c(1:3,7,4:6),]
GRN.Inh.peaks.test.res<-as.data.frame(GRN.Inh.peaks.test.res)
GRN.Inh.peaks.test.res$Inh.B<-as.numeric(GRN.Inh.peaks.test.res$Inh.B)
GRN.Inh.peaks.test.res$Inh.G<-as.numeric(GRN.Inh.peaks.test.res$Inh.G)
GRN.Inh.peaks.test.res$Inh.L<-as.numeric(GRN.Inh.peaks.test.res$Inh.L)
#make into proportions for direct comparison to all CREs:
GRN.Inh.peaks.test.res$Inh.B<-GRN.Inh.peaks.test.res$Inh.B/sum(GRN.Inh.peaks.test.res$Inh.B)
GRN.Inh.peaks.test.res$Inh.G<-GRN.Inh.peaks.test.res$Inh.G/sum(GRN.Inh.peaks.test.res$Inh.G)
GRN.Inh.peaks.test.res$Inh.L<-GRN.Inh.peaks.test.res$Inh.L/sum(GRN.Inh.peaks.test.res$Inh.L)
GRN.Inh.test.res<-melt(GRN.Inh.peaks.test.res, id="Type")
GRN.Inh.test.res.plot<-ggplot(GRN.Inh.test.res, aes(x=variable, y=value,color=Type, group=Type))+
  geom_line(size=2)+theme_classic()+theme(panel.background = element_rect(fill = "transparent",colour = NA), plot.background = element_rect(fill = "transparent",colour = NA), legend.background = element_rect(fill = "transparent",colour = NA))+ 
  scale_y_continuous(breaks = seq(from = 0, to = 1.0, by = 0.2))+
  xlab("") + ylab("Proportion of Peaks") + ggtitle("") + theme(plot.title = element_text(size=24, vjust = 1.5, face = "bold"), axis.text = element_text(size=20), axis.title=element_text(size=22),legend.text = element_text(size=19), legend.title=element_text(size=19), legend.key = element_rect(size = 5, colour = NA), legend.key.size = unit(1.5, 'lines'))+
  ylim(0,1.0)
GRN.Inh.test.res.plot
#ggsave("./GRN.Inh.test.res.plot.pdf", GRN.Inh.test.res.plot, width=8, height=6)


#now do TF lineplots for direct comparison to GRN genes:
#CT:
TF.CT.peaks.all<-CT.peaks.all.genefunct %>% filter(Function=="TranscriptionFactor")
TF.CT.peaks.test.B<-TF.CT.peaks.all %>% group_by(CT.B) %>% tally()
TF.CT.peaks.test.B<-TF.CT.peaks.test.B$n
TF.CT.peaks.test.G<-TF.CT.peaks.all %>% group_by(CT.G) %>% tally()
TF.CT.peaks.test.G<-TF.CT.peaks.test.G$n
TF.CT.peaks.test.L<-TF.CT.peaks.all %>% group_by(CT.L) %>% tally()
TF.CT.peaks.test.L<-TF.CT.peaks.test.L$n
TF.CT.peaks.test.res<-cbind(Type,TF.CT.peaks.test.B,TF.CT.peaks.test.G,TF.CT.peaks.test.L)
colnames(TF.CT.peaks.test.res)<-c("Type","CT.B","CT.G","CT.L")
TF.CT.peaks.test.res<-TF.CT.peaks.test.res[c(1:3,7,4:6),]
TF.CT.peaks.test.res<-as.data.frame(TF.CT.peaks.test.res)
TF.CT.peaks.test.res$CT.B<-as.numeric(TF.CT.peaks.test.res$CT.B)
TF.CT.peaks.test.res$CT.G<-as.numeric(TF.CT.peaks.test.res$CT.G)
TF.CT.peaks.test.res$CT.L<-as.numeric(TF.CT.peaks.test.res$CT.L)
#make into proportions for direct comparison to all CREs:
TF.CT.peaks.test.res$CT.B<-TF.CT.peaks.test.res$CT.B/sum(TF.CT.peaks.test.res$CT.B)
TF.CT.peaks.test.res$CT.G<-TF.CT.peaks.test.res$CT.G/sum(TF.CT.peaks.test.res$CT.G)
TF.CT.peaks.test.res$CT.L<-TF.CT.peaks.test.res$CT.L/sum(TF.CT.peaks.test.res$CT.L)
TF.CT.test.res<-melt(TF.CT.peaks.test.res, id="Type")
TF.CT.test.res.plot<-ggplot(TF.CT.test.res, aes(x=variable, y=value,color=Type, group=Type))+
  geom_line(size=2)+theme_classic()+theme(panel.background = element_rect(fill = "transparent",colour = NA), plot.background = element_rect(fill = "transparent",colour = NA), legend.background = element_rect(fill = "transparent",colour = NA))+ 
  xlab("") + ylab("Proportion of Peaks") + ggtitle("") + theme(plot.title = element_text(size=24, vjust = 1.5, face = "bold"), axis.text = element_text(size=20), axis.title=element_text(size=22),legend.text = element_text(size=19), legend.title=element_text(size=19), legend.key = element_rect(size = 5, colour = NA), legend.key.size = unit(1.5, 'lines'))+
  ylim(0,1.0)
TF.CT.test.res.plot
#ggsave("./TF.CT.test.res.plot.pdf", TF.CT.test.res.plot, width=8, height=6)

#inh:
TF.Inh.peaks.all<-Inh.peaks.all.genefunct %>% filter(Function=="TranscriptionFactor")
TF.Inh.peaks.test.B<-TF.Inh.peaks.all %>% group_by(Inh.B) %>% tally()
colnames(TF.Inh.peaks.test.B)<-c("Type", "Inh.B")
TF.Inh.peaks.test.B[7,1]<-"underdom"
TF.Inh.peaks.test.B[7,2]<-0
TF.Inh.peaks.test.B<-TF.Inh.peaks.test.B[c(1:5,7,6),]
TF.Inh.peaks.test.G<-TF.Inh.peaks.all %>% group_by(Inh.G) %>% tally()
TF.Inh.peaks.test.G<-TF.Inh.peaks.test.G$n
TF.Inh.peaks.test.L<-TF.Inh.peaks.all %>% group_by(Inh.L) %>% tally()
TF.Inh.peaks.test.L<-TF.Inh.peaks.test.L$n
TF.Inh.peaks.test.res<-cbind(TF.Inh.peaks.test.B,TF.Inh.peaks.test.G,TF.Inh.peaks.test.L)
colnames(TF.Inh.peaks.test.res)<-c("Type","Inh.B","Inh.G","Inh.L")
TF.Inh.peaks.test.res<-TF.Inh.peaks.test.res[c(1:3,7,4:6),]
TF.Inh.peaks.test.res<-as.data.frame(TF.Inh.peaks.test.res)
TF.Inh.peaks.test.res$Inh.B<-as.numeric(TF.Inh.peaks.test.res$Inh.B)
TF.Inh.peaks.test.res$Inh.G<-as.numeric(TF.Inh.peaks.test.res$Inh.G)
TF.Inh.peaks.test.res$Inh.L<-as.numeric(TF.Inh.peaks.test.res$Inh.L)
#make into proportions for direct comparison to all CREs:
TF.Inh.peaks.test.res$Inh.B<-TF.Inh.peaks.test.res$Inh.B/sum(TF.Inh.peaks.test.res$Inh.B)
TF.Inh.peaks.test.res$Inh.G<-TF.Inh.peaks.test.res$Inh.G/sum(TF.Inh.peaks.test.res$Inh.G)
TF.Inh.peaks.test.res$Inh.L<-TF.Inh.peaks.test.res$Inh.L/sum(TF.Inh.peaks.test.res$Inh.L)
TF.Inh.test.res<-melt(TF.Inh.peaks.test.res, id="Type")
TF.Inh.test.res.plot<-ggplot(TF.Inh.test.res, aes(x=variable, y=value,color=Type, group=Type))+
  geom_line(size=2)+theme_classic()+theme(panel.background = element_rect(fill = "transparent",colour = NA), plot.background = element_rect(fill = "transparent",colour = NA), legend.background = element_rect(fill = "transparent",colour = NA))+ 
  scale_y_continuous(breaks = seq(from = 0, to = 1.0, by = 0.2))+
  xlab("") + ylab("Proportion of Peaks") + ggtitle("") + theme(plot.title = element_text(size=24, vjust = 1.5, face = "bold"), axis.text = element_text(size=20), axis.title=element_text(size=22),legend.text = element_text(size=19), legend.title=element_text(size=19), legend.key = element_rect(size = 5, colour = NA), legend.key.size = unit(1.5, 'lines'))+
  ylim(0,1.0)
TF.Inh.test.res.plot
#ggsave("./TF.Inh.test.res.plot.pdf", TF.Inh.test.res.plot, width=8, height=6)
```

Compare mean openness changes of cis OCRs near GRN genes vs cis OCRs near transcription factors vs cis OCRs as a whole
```{r}
B.GRNvsall.cis.grn.meanopennesschange<-data.frame(log2FC=(ase_nolowcounts.CT.B.allcis$log2FC_TT_HEvsHT),
                                          PeakCoord=row.names(ase_nolowcounts.CT.B.allcis))

B.GRNvsall.cis.grn.meanopennesschange<-B.GRNvsall.cis.grn.meanopennesschange %>% filter(PeakCoord %in% (GRN.Inh.peaks.all$PeakCoord))
B.GRNvsall.cis.grn.meanopennesschange<-B.GRNvsall.cis.grn.meanopennesschange %>% select(log2FC)
B.GRNvsall.cis.grn.meanopennesschange$Type<-rep("Cis OCRs near GRN gene", length(B.GRNvsall.cis.grn.meanopennesschange))
B.GRNvsall.cis.all.meanopennesschange<-data.frame(log2FC=(ase_nolowcounts.CT.B.allcis$log2FC_TT_HEvsHT),
                                          Type=rep("all cis OCRs", length(ase_nolowcounts.CT.B.allcis$log2FC_TT_HEvsHT)))
B.GRNvsall.cis.meanopennesschange<-rbind(B.GRNvsall.cis.all.meanopennesschange,B.GRNvsall.cis.grn.meanopennesschange)

G.GRNvsall.cis.grn.meanopennesschange<-data.frame(log2FC=(ase_nolowcounts.CT.G.allcis$log2FC_TT_HEvsHT),
                                          PeakCoord=row.names(ase_nolowcounts.CT.G.allcis))

G.GRNvsall.cis.grn.meanopennesschange<-G.GRNvsall.cis.grn.meanopennesschange %>% filter(PeakCoord %in% (GRN.Inh.peaks.all$PeakCoord))
G.GRNvsall.cis.grn.meanopennesschange<-G.GRNvsall.cis.grn.meanopennesschange %>% select(log2FC)
G.GRNvsall.cis.grn.meanopennesschange$Type<-rep("Cis OCRs near GRN gene", length(G.GRNvsall.cis.grn.meanopennesschange))
G.GRNvsall.cis.all.meanopennesschange<-data.frame(log2FC=(ase_nolowcounts.CT.G.allcis$log2FC_TT_HEvsHT),
                                          Type=rep("all cis OCRs", length(ase_nolowcounts.CT.G.allcis$log2FC_TT_HEvsHT)))
G.GRNvsall.cis.meanopennesschange<-rbind(G.GRNvsall.cis.all.meanopennesschange,G.GRNvsall.cis.grn.meanopennesschange)

L.GRNvsall.cis.grn.meanopennesschange<-data.frame(log2FC=(ase_nolowcounts.CT.L.allcis$log2FC_TT_HEvsHT),
                                          PeakCoord=row.names(ase_nolowcounts.CT.L.allcis))

L.GRNvsall.cis.grn.meanopennesschange<-L.GRNvsall.cis.grn.meanopennesschange %>% filter(PeakCoord %in% (GRN.Inh.peaks.all$PeakCoord))
L.GRNvsall.cis.grn.meanopennesschange<-L.GRNvsall.cis.grn.meanopennesschange %>% select(log2FC)
L.GRNvsall.cis.grn.meanopennesschange$Type<-rep("Cis OCRs near GRN gene", length(L.GRNvsall.cis.grn.meanopennesschange))
L.GRNvsall.cis.all.meanopennesschange<-data.frame(log2FC=(ase_nolowcounts.CT.L.allcis$log2FC_TT_HEvsHT),
                                          Type=rep("all cis OCRs", length(ase_nolowcounts.CT.L.allcis$log2FC_TT_HEvsHT)))
L.GRNvsall.cis.meanopennesschange<-rbind(L.GRNvsall.cis.all.meanopennesschange,L.GRNvsall.cis.grn.meanopennesschange)


BGL.GRNvsall.cis.meanopennesschange<-rbind(B.GRNvsall.cis.meanopennesschange %>% mutate(Stage=rep("B", nrow(B.GRNvsall.cis.meanopennesschange))),G.GRNvsall.cis.meanopennesschange %>% mutate(Stage=rep("G", nrow(G.GRNvsall.cis.meanopennesschange))),L.GRNvsall.cis.meanopennesschange %>% mutate(Stage=rep("L", nrow(L.GRNvsall.cis.meanopennesschange))))

B.GRNvsallTF.ciscis.allnearTF.meanopennesschange<-data.frame(log2FC=(ase_nolowcounts.CT.B.allcis$log2FC_TT_HEvsHT),
                                          PeakCoord=row.names(ase_nolowcounts.CT.B.allcis))
B.GRNvsallTF.ciscis.allnearTF.meanopennesschange<-left_join(B.GRNvsallTF.ciscis.allnearTF.meanopennesschange, CT.peaks.all.genefunct, by="PeakCoord")
B.GRNvsallTF.ciscis.allnearTF.meanopennesschange <-B.GRNvsallTF.ciscis.allnearTF.meanopennesschange %>% filter(Function=="TranscriptionFactor") #81 cis peaks near TF (as compared to 45 for cis peaks near GRN only). need to bootstrap to 45.
G.GRNvsallTF.ciscis.allnearTF.meanopennesschange<-data.frame(log2FC=(ase_nolowcounts.CT.G.allcis$log2FC_TT_HEvsHT),
                                          PeakCoord=row.names(ase_nolowcounts.CT.G.allcis))
G.GRNvsallTF.ciscis.allnearTF.meanopennesschange<-left_join(G.GRNvsallTF.ciscis.allnearTF.meanopennesschange, CT.peaks.all.genefunct, by="PeakCoord")
G.GRNvsallTF.ciscis.allnearTF.meanopennesschange <-G.GRNvsallTF.ciscis.allnearTF.meanopennesschange %>% filter(Function=="TranscriptionFactor")
L.GRNvsallTF.ciscis.allnearTF.meanopennesschange<-data.frame(log2FC=(ase_nolowcounts.CT.L.allcis$log2FC_TT_HEvsHT),
                                          PeakCoord=row.names(ase_nolowcounts.CT.L.allcis))
L.GRNvsallTF.ciscis.allnearTF.meanopennesschange<-left_join(L.GRNvsallTF.ciscis.allnearTF.meanopennesschange, CT.peaks.all.genefunct, by="PeakCoord")
L.GRNvsallTF.ciscis.allnearTF.meanopennesschange <-L.GRNvsallTF.ciscis.allnearTF.meanopennesschange %>% filter(Function=="TranscriptionFactor")

B.GRNvsallTF.ciscis.allnearTF.meanopennesschange<-data.frame(log2FC=B.GRNvsallTF.ciscis.allnearTF.meanopennesschange %>% select(log2FC), Type=rep("Cis OCRs near all TFs", nrow(B.GRNvsallTF.ciscis.allnearTF.meanopennesschange %>% select(log2FC))))
B.GRNvsallTF.cismeanopennesschange<-rbind(B.GRNvsallTF.ciscis.allnearTF.meanopennesschange,B.GRNvsall.cis.grn.meanopennesschange)

G.GRNvsallTF.ciscis.allnearTF.meanopennesschange<-data.frame(log2FC=G.GRNvsallTF.ciscis.allnearTF.meanopennesschange %>% select(log2FC), Type=rep("Cis OCRs near all TFs", nrow(G.GRNvsallTF.ciscis.allnearTF.meanopennesschange %>% select(log2FC))))
G.GRNvsallTF.cismeanopennesschange<-rbind(G.GRNvsallTF.ciscis.allnearTF.meanopennesschange,G.GRNvsall.cis.grn.meanopennesschange)

L.GRNvsallTF.ciscis.allnearTF.meanopennesschange<-data.frame(log2FC=L.GRNvsallTF.ciscis.allnearTF.meanopennesschange %>% select(log2FC), Type=rep("Cis OCRs near all TFs", nrow(L.GRNvsallTF.ciscis.allnearTF.meanopennesschange %>% select(log2FC))))
L.GRNvsallTF.cismeanopennesschange<-rbind(L.GRNvsallTF.ciscis.allnearTF.meanopennesschange,L.GRNvsall.cis.grn.meanopennesschange)

BGL.GRNvsallTF.cismeanopennesschange<-rbind(B.GRNvsallTF.cismeanopennesschange %>% mutate(Stage=rep("B", nrow(B.GRNvsallTF.cismeanopennesschange))),G.GRNvsallTF.cismeanopennesschange %>% mutate(Stage=rep("G", nrow(G.GRNvsallTF.cismeanopennesschange))),L.GRNvsallTF.cismeanopennesschange %>% mutate(Stage=rep("L", nrow(L.GRNvsallTF.cismeanopennesschange))))

BGL.GRNvsallTFvsall.cismeanopennesschange<-rbind(BGL.GRNvsall.cis.meanopennesschange,BGL.GRNvsallTF.cismeanopennesschange)
BGL.GRNvsallTFvsall.cis.meanopennesschange<-unique(BGL.GRNvsallTFvsall.cismeanopennesschange)
BGL.GRNvsallTFvsall.cismeanopennesschange.plot<-ggplot(BGL.GRNvsallTFvsall.cismeanopennesschange, aes(x=Type, y=abs(log2FC), colour=Type))+
    geom_violin()+
  geom_boxplot(width=0.1, color="black")+
  ylim(0,15)+
    theme_bw()+
  background_grid(major="none", minor="none")+
  theme(legend.position = "none")

#For Figure 7C:
#ggsave("BGL.GRNvsallTFvsall.cismeanopennesschange.plot.pdf",BGL.GRNvsallTFvsall.cismeanopennesschange.plot,width=10, height=8, units="in")

#for Figure 5:
#because the variances of cis vs trans are different, i will use a welch's test of unequal variance to see if the means are statistically different:
oneway.test(abs(log2FC) ~ Type, BGL.GRNvsallTFvsall.cismeanopennesschange) #this tells us that the means of the 3 types DO differ, even when we don't assume the variances are the same. i got this from https://www.statology.org/welchs-anova-in-r/. then we need to figure out which means are different, which i do below. 

library(DescTools)
ScheffeTest(aov(abs(log2FC) ~ Type, BGL.GRNvsallTFvsall.cismeanopennesschange))
#the above says the same as the below but i think the above is the correct test.

t.test(abs(BGL.GRNvsallTFvsall.cismeanopennesschange %>% filter(Type=="all cis OCRs") %>% select(log2FC)),abs(BGL.GRNvsallTFvsall.cismeanopennesschange %>% filter(Type=="Cis OCRs near GRN gene") %>% select(log2FC))) #Cis OCRs near GRN genes are statistically more open than cis OCRs near all genes 
t.test(abs(BGL.GRNvsallTFvsall.cismeanopennesschange %>% filter(Type=="Cis OCRs near all TFs") %>% select(log2FC)),abs(BGL.GRNvsallTFvsall.cismeanopennesschange %>% filter(Type=="Cis OCRs near GRN gene") %>% select(log2FC))) #Cis OCRS near GRN genes are statistically more open than cis OCRs near TFs
t.test(abs(BGL.GRNvsallTFvsall.cismeanopennesschange %>% filter(Type=="all cis OCRs") %>% select(log2FC)),abs(BGL.GRNvsallTFvsall.cismeanopennesschange %>% filter(Type=="Cis OCRs near all TFs") %>% select(log2FC))) #Cis OCRs near TFs are NOT different from cis OCRs near all genes

#are they different from stage to stage:
ScheffeTest(aov(abs(log2FC) ~ Stage, BGL.GRNvsallTFvsall.cismeanopennesschange %>% filter(Type=="Cis OCRs near GRN gene"))) #no
ScheffeTest(aov(abs(log2FC) ~ Stage, BGL.GRNvsallTFvsall.cismeanopennesschange %>% filter(Type=="Cis OCRs near all TFs"))) #no
ScheffeTest(aov(abs(log2FC) ~ Stage, BGL.GRNvsallTFvsall.cismeanopennesschange %>% filter(Type=="all cis OCRs"))) #yes at all stages

```
make beds for HOMER analysis of cis vs trans peaks (Table S11)
```{r}
ase_nolowcounts.BGL_allcis.peaks<-full_join(ase_nolowcounts.B_allcis.peaks, ase_nolowcounts.B_allcis.peaks %>% select(PeakCoord), by="PeakCoord")
ase_nolowcounts.BGL_allcis.peaks<-full_join(ase_nolowcounts.BGL_allcis.peaks, ase_nolowcounts.L_allcis.peaks %>% select(PeakCoord),by="PeakCoord")
ase_nolowcounts.BGL_allcis.peaks<-unique(ase_nolowcounts.BGL_allcis.peaks)
#makehomerbedD(ase_nolowcounts.BGL_allcis.peaks)

ase_nolowcounts.BGL_alltrans.peaks<-full_join(ase_nolowcounts.B_alltrans.peaks, ase_nolowcounts.B_alltrans.peaks %>% select(PeakCoord), by="PeakCoord")
ase_nolowcounts.BGL_alltrans.peaks<-full_join(ase_nolowcounts.BGL_alltrans.peaks, ase_nolowcounts.L_alltrans.peaks %>% select(PeakCoord),by="PeakCoord")
ase_nolowcounts.BGL_alltrans.peaks<-unique(ase_nolowcounts.BGL_alltrans.peaks)
#makehomerbedD(ase_nolowcounts.BGL_alltrans.peaks)
```

Compare mean openness changes of trans OCRs near GRN genes vs trans OCRs near transcription factors vs trans OCRs as a whole
```{r}
B.GRNvsall.trans.grn.meanopennesschange<-data.frame(log2FC=(ase_nolowcounts.CT.B.alltrans$log2FC_TT_HEvsHT),
                                          PeakCoord=row.names(ase_nolowcounts.CT.B.alltrans))

B.GRNvsall.trans.grn.meanopennesschange<-B.GRNvsall.trans.grn.meanopennesschange %>% filter(PeakCoord %in% (GRN.Inh.peaks.all$PeakCoord))
B.GRNvsall.trans.grn.meanopennesschange<-B.GRNvsall.trans.grn.meanopennesschange %>% select(log2FC)
B.GRNvsall.trans.grn.meanopennesschange$Type<-rep("trans OCRs near GRN gene", length(B.GRNvsall.trans.grn.meanopennesschange))
B.GRNvsall.trans.all.meanopennesschange<-data.frame(log2FC=(ase_nolowcounts.CT.B.alltrans$log2FC_TT_HEvsHT),
                                          Type=rep("all trans OCRs", length(ase_nolowcounts.CT.B.alltrans$log2FC_TT_HEvsHT)))
B.GRNvsall.trans.meanopennesschange<-rbind(B.GRNvsall.trans.all.meanopennesschange,B.GRNvsall.trans.grn.meanopennesschange)


G.GRNvsall.trans.grn.meanopennesschange<-data.frame(log2FC=(ase_nolowcounts.CT.G.alltrans$log2FC_TT_HEvsHT),
                                          PeakCoord=row.names(ase_nolowcounts.CT.G.alltrans))

G.GRNvsall.trans.grn.meanopennesschange<-G.GRNvsall.trans.grn.meanopennesschange %>% filter(PeakCoord %in% (GRN.Inh.peaks.all$PeakCoord))
G.GRNvsall.trans.grn.meanopennesschange<-G.GRNvsall.trans.grn.meanopennesschange %>% select(log2FC)
G.GRNvsall.trans.grn.meanopennesschange$Type<-rep("trans OCRs near GRN gene", length(G.GRNvsall.trans.grn.meanopennesschange))
G.GRNvsall.trans.all.meanopennesschange<-data.frame(log2FC=(ase_nolowcounts.CT.G.alltrans$log2FC_TT_HEvsHT),
                                          Type=rep("all trans OCRs", length(ase_nolowcounts.CT.G.alltrans$log2FC_TT_HEvsHT)))
G.GRNvsall.trans.meanopennesschange<-rbind(G.GRNvsall.trans.all.meanopennesschange,G.GRNvsall.trans.grn.meanopennesschange)

L.GRNvsall.trans.grn.meanopennesschange<-data.frame(log2FC=(ase_nolowcounts.CT.L.alltrans$log2FC_TT_HEvsHT),
                                          PeakCoord=row.names(ase_nolowcounts.CT.L.alltrans))

L.GRNvsall.trans.grn.meanopennesschange<-L.GRNvsall.trans.grn.meanopennesschange %>% filter(PeakCoord %in% (GRN.Inh.peaks.all$PeakCoord))
L.GRNvsall.trans.grn.meanopennesschange<-L.GRNvsall.trans.grn.meanopennesschange %>% select(log2FC)
L.GRNvsall.trans.grn.meanopennesschange$Type<-rep("trans OCRs near GRN gene", length(L.GRNvsall.trans.grn.meanopennesschange))
L.GRNvsall.trans.all.meanopennesschange<-data.frame(log2FC=(ase_nolowcounts.CT.L.alltrans$log2FC_TT_HEvsHT),
                                          Type=rep("all trans OCRs", length(ase_nolowcounts.CT.L.alltrans$log2FC_TT_HEvsHT)))
L.GRNvsall.trans.meanopennesschange<-rbind(L.GRNvsall.trans.all.meanopennesschange,L.GRNvsall.trans.grn.meanopennesschange)


#now only near all TFs:
B.GRNvsallTF.trans.allnearTF.meanopennesschange<-data.frame(log2FC=(ase_nolowcounts.CT.B.alltrans$log2FC_TT_HEvsHT),
                                          PeakCoord=row.names(ase_nolowcounts.CT.B.alltrans))
B.GRNvsallTF.trans.allnearTF.meanopennesschange<-left_join(B.GRNvsallTF.trans.allnearTF.meanopennesschange, CT.peaks.all.genefunct, by="PeakCoord")
B.GRNvsallTF.trans.allnearTF.meanopennesschange <-B.GRNvsallTF.trans.allnearTF.meanopennesschange %>% filter(Function=="TranscriptionFactor") 
G.GRNvsallTF.trans.allnearTF.meanopennesschange<-data.frame(log2FC=(ase_nolowcounts.CT.G.alltrans$log2FC_TT_HEvsHT),
                                          PeakCoord=row.names(ase_nolowcounts.CT.G.alltrans))
G.GRNvsallTF.trans.allnearTF.meanopennesschange<-left_join(G.GRNvsallTF.trans.allnearTF.meanopennesschange, CT.peaks.all.genefunct, by="PeakCoord")
G.GRNvsallTF.trans.allnearTF.meanopennesschange <-G.GRNvsallTF.trans.allnearTF.meanopennesschange %>% filter(Function=="TranscriptionFactor") 
L.GRNvsallTF.trans.allnearTF.meanopennesschange<-data.frame(log2FC=(ase_nolowcounts.CT.L.alltrans$log2FC_TT_HEvsHT),
                                          PeakCoord=row.names(ase_nolowcounts.CT.L.alltrans))
L.GRNvsallTF.trans.allnearTF.meanopennesschange<-left_join(L.GRNvsallTF.trans.allnearTF.meanopennesschange, CT.peaks.all.genefunct, by="PeakCoord")
L.GRNvsallTF.trans.allnearTF.meanopennesschange <-L.GRNvsallTF.trans.allnearTF.meanopennesschange %>% filter(Function=="TranscriptionFactor") 

B.GRNvsallTF.trans.allnearTF.meanopennesschange<-data.frame(log2FC=B.GRNvsallTF.trans.allnearTF.meanopennesschange %>% select(log2FC), Type=rep("trans OCRs near all TFs", nrow(B.GRNvsallTF.trans.allnearTF.meanopennesschange %>% select(log2FC))))
B.GRNvsallTF.trans.meanopennesschange<-rbind(B.GRNvsallTF.trans.allnearTF.meanopennesschange,B.GRNvsall.trans.grn.meanopennesschange)


G.GRNvsallTF.trans.allnearTF.meanopennesschange<-data.frame(log2FC=G.GRNvsallTF.trans.allnearTF.meanopennesschange %>% select(log2FC), Type=rep("trans OCRs near all TFs", nrow(G.GRNvsallTF.trans.allnearTF.meanopennesschange %>% select(log2FC))))
G.GRNvsallTF.trans.meanopennesschange<-rbind(G.GRNvsallTF.trans.allnearTF.meanopennesschange,G.GRNvsall.trans.grn.meanopennesschange)


L.GRNvsallTF.trans.allnearTF.meanopennesschange<-data.frame(log2FC=L.GRNvsallTF.trans.allnearTF.meanopennesschange %>% select(log2FC), Type=rep("trans OCRs near all TFs", nrow(L.GRNvsallTF.trans.allnearTF.meanopennesschange %>% select(log2FC))))
L.GRNvsallTF.trans.meanopennesschange<-rbind(L.GRNvsallTF.trans.allnearTF.meanopennesschange,L.GRNvsall.trans.grn.meanopennesschange)

BGL.GRNvsall.transmeanopennesschange<-rbind(B.GRNvsall.trans.meanopennesschange %>% mutate(Stage=rep("B", nrow(B.GRNvsall.trans.meanopennesschange))),G.GRNvsall.trans.meanopennesschange %>% mutate(Stage=rep("G", nrow(G.GRNvsall.trans.meanopennesschange))),L.GRNvsall.trans.meanopennesschange %>% mutate(Stage=rep("L", nrow(L.GRNvsall.trans.meanopennesschange))))

BGL.GRNvsallTF.transmeanopennesschange<-rbind(B.GRNvsallTF.trans.meanopennesschange %>% mutate(Stage=rep("B", nrow(B.GRNvsallTF.trans.meanopennesschange))),G.GRNvsallTF.trans.meanopennesschange %>% mutate(Stage=rep("G", nrow(G.GRNvsallTF.trans.meanopennesschange))),L.GRNvsallTF.trans.meanopennesschange %>% mutate(Stage=rep("L", nrow(L.GRNvsallTF.trans.meanopennesschange))))

BGL.GRNvsallvsallTF.transmeanopennesschange<-rbind(BGL.GRNvsall.transmeanopennesschange,BGL.GRNvsallTF.transmeanopennesschange)
BGL.GRNvsallvsallTF.transmeanopennesschange<-unique(BGL.GRNvsallvsallTF.transmeanopennesschange)

BGL.GRNvsallvsallTF.transmeanopennesschange.plot<-ggplot(BGL.GRNvsallvsallTF.transmeanopennesschange, aes(x=Type, y=abs(log2FC), colour=Type))+
  geom_violin()+
  geom_boxplot(width=0.1,color="black")
#ggsave("BGL.GRNvsallvsallTF.transmeanopennesschange.plot.pdf",BGL.GRNvsallvsallTF.transmeanopennesschange.plot, width=10, height=8, units="in")


ScheffeTest(aov(abs(log2FC)~Type,BGL.GRNvsallvsallTF.transmeanopennesschange))#trans ocrs near TFs and near GRN genes are diff than overall but not from each other #Figure 7D

BGL.GRNvsall.transmeanopennesschange<-unique(BGL.GRNvsall.transmeanopennesschange)



#all cis vs all trans at all stages:
BGL.GRNvsall.cisvstransmeanopennesschange<-rbind(BGL.GRNvsall.cis.meanopennesschange, BGL.GRNvsall.transmeanopennesschange)
BGL.GRNvsall.cisvstransmeanopennesschange<-unique(BGL.GRNvsall.cisvstransmeanopennesschange)
ScheffeTest(aov(abs(log2FC) ~ Type, BGL.GRNvsall.cisvstransmeanopennesschange %>% filter(Type=="all trans OCRs" | Type=="all cis OCRs") %>% filter(Stage=="B")))
ScheffeTest(aov(abs(log2FC) ~ Type, BGL.GRNvsall.cisvstransmeanopennesschange %>% filter(Type=="all trans OCRs" | Type=="all cis OCRs") %>% filter(Stage=="G")))
ScheffeTest(aov(abs(log2FC) ~ Type, BGL.GRNvsall.cisvstransmeanopennesschange %>% filter(Type=="all trans OCRs" | Type=="all cis OCRs") %>% filter(Stage=="L")))# from looking at the "diff" on all 3 of these, i think the difference increases from B to G, but decreases slightly from G to L 

```
```{r}
#make bed files for HOMER
#makehomerbedD <- function(df) {
  temp<-transform(df, col=do.call(rbind, strsplit(PeakCoord, ':', fixed=TRUE)), stringsAsFactors=F) %>% select (-PeakCoord)
temp<-transform(temp, col.3=do.call(rbind, strsplit(col.2, '-', fixed=TRUE)), stringsAsFactors=F) %>% select(-col.2)
temp<-temp[c(2,3,4,1)]
filename<-paste(deparse(substitute(df)),".gff",sep="")
write.table(temp, filename, sep="\t", row.names=F, col.names=F, quote=F)
}

```

How many peaks are near different types of genes?
```{r}
peaksneargenes<-Hery_peak_gene_map_final %>% filter(abs(HEpeakGeneDist_5prime) <=25000 | abs(as.numeric(HEpeakGeneDist_3prime))<=25000)
peaksneargenes<-peaksneargenes %>% dplyr::rename(peakno=peak)
peaksneargenes<-as.data.frame(peaksneargenes)
peaksneargenes[is.na(peaksneargenes)]<-"NA"
peaksneargenes<-peaksneargenes %>% filter(!(H_ery_ID_5prime=="NA" | H_ery_ID_3prime=="NA"))
peaksneargenes<-left_join(peaksneargenes, HEliftedpeakcoordsonly %>% select(-midpeak), by="peakno")
peaksneargenes$PeakCoord<-paste0(peaksneargenes$HEChr,":",peaksneargenes$start, "-",peaksneargenes$end)
peaksneargenes<-inner_join(ase_nolowcounts.CT.peaks.all %>% select(PeakCoord), peaksneargenes %>% select(peakno, PeakCoord), by="PeakCoord") #remove low counts
peaksneargenes_genename<-genesnearpeaks %>% filter(PeakCoord %in% peaksneargenes$PeakCoord)

#how many peaks are near a gene?
numberpeaksw_in25kbofgene<-peaksneargenes_genename %>% select(PeakCoord, H_ery_ID) %>% group_by(H_ery_ID) %>% tally()
numberpeaksw_in25kbofgene<-numberpeaksw_in25kbofgene %>% dplyr::rename(numberofpeaks=n)
numberpeaksw_in25kbofgene<-numberpeaksw_in25kbofgene %>% group_by(numberofpeaks) %>% tally()
numberpeaksw_in25kbofgene<-numberpeaksw_in25kbofgene %>% dplyr::rename(numberofgenes=n)
ggplot(numberpeaksw_in25kbofgene, aes(x=numberofpeaks, y=numberofgenes))+
  geom_col(binwidth=10) #looks like a right-skewed distribution
ggplot(numberpeaksw_in25kbofgene, aes(x=numberofpeaks, y=numberofgenes/(sum(numberofgenes))))+
  geom_col()+
  ylab("Proportion of all genes")+
  xlab("Number of peaks")

numberpeaksw_in25kbofTF<-peaksneargenes_genename %>% filter(Function=="TranscriptionFactor")
numberpeaksw_in25kbofTF<-numberpeaksw_in25kbofTF %>% select(PeakCoord, H_ery_ID) %>% group_by(H_ery_ID) %>% tally()
numberpeaksw_in25kbofTF<-numberpeaksw_in25kbofTF %>% dplyr::rename(numberofpeaks=n)
numberpeaksw_in25kbofTF<-numberpeaksw_in25kbofTF %>% group_by(numberofpeaks) %>% tally()
numberpeaksw_in25kbofTF<-numberpeaksw_in25kbofTF %>% dplyr::rename(numberofgenes=n)
numberpeaksw_in25kbofGRNgene<-peaksneargenes_genename %>% filter(SPU.hit %in% GRNgeneslist$SPU.hit)
numberpeaksw_in25kbofGRNgene<-numberpeaksw_in25kbofGRNgene %>% select(PeakCoord, H_ery_ID) %>% group_by(H_ery_ID) %>% tally()
numberpeaksw_in25kbofGRNgene<-numberpeaksw_in25kbofGRNgene %>% dplyr::rename(numberofpeaks=n)
numberpeaksw_in25kbofGRNgene<-numberpeaksw_in25kbofGRNgene %>% group_by(numberofpeaks) %>% tally()
numberpeaksw_in25kbofGRNgene<-numberpeaksw_in25kbofGRNgene %>% dplyr::rename(numberofgenes=n)
peaksneargenes.plot<-ggplot()+
  geom_smooth(data=numberpeaksw_in25kbofgene, aes(x=numberofpeaks, y=numberofgenes/(sum(numberofgenes)),colour="All genes"))+
  geom_smooth(data=numberpeaksw_in25kbofTF, aes(x=numberofpeaks, y=numberofgenes/(sum(numberofgenes)),colour="TFs"))+
    geom_smooth(data=numberpeaksw_in25kbofGRNgene, aes(x=numberofpeaks, y=numberofgenes/(sum(numberofgenes)),colour="GRN genes"))+
  ylab("Proportion of all genes")+
  xlab("Number of peaks")+
  xlim(0,25)+
  theme_classic()
#For Figure 7A:
#ggsave("peaksneargenes.numberofpeaks.plot.pdf",peaksneargenes.plot, width=10, height=8, units="in")

#are the distributions different?

numberpeaksw_in25kbof3typesofgenes<-rbind(numberpeaksw_in25kbofgene %>% mutate(Type=rep("peaks near all genes",nrow(numberpeaksw_in25kbofgene))),numberpeaksw_in25kbofTF %>% mutate(Type=rep("peaks near TFs",nrow(numberpeaksw_in25kbofTF))),numberpeaksw_in25kbofGRNgene %>% mutate(Type=rep("peaks near GRN genes",nrow(numberpeaksw_in25kbofGRNgene))))

#Are the distributions in Figure 7A statistically different from each other?:
ks.test(numberpeaksw_in25kbofgene$numberofgenes,numberpeaksw_in25kbofTF$numberofgenes) #stat diff
ks.test(numberpeaksw_in25kbofgene$numberofgenes,numberpeaksw_in25kbofGRNgene$numberofgenes) #stat diff
ks.test(numberpeaksw_in25kbofGRNgene$numberofgenes,numberpeaksw_in25kbofTF$numberofgenes) #not stat diff

```
How do regulatory modes of all OCRs compare to regulatory mode of OCRs near GRN genes? (Figure 7B)
```{r}
#CT:
peaksneargenes.CT.all<-ase_nolowcounts.CT.peaks.all %>% filter(PeakCoord %in% peaksneargenes$PeakCoord)
peaksneargenes.CT.all.B<-peaksneargenes.CT.all %>% group_by(CT.B) %>% tally()
peaksneargenes.CT.all.B<-peaksneargenes.CT.all.B$n
peaksneargenes.CT.all.G<-peaksneargenes.CT.all %>% group_by(CT.G) %>% tally()
peaksneargenes.CT.all.G<-peaksneargenes.CT.all.G$n
peaksneargenes.CT.all.L<-peaksneargenes.CT.all %>% group_by(CT.L) %>% tally()
peaksneargenes.CT.all.L<-peaksneargenes.CT.all.L$n
peaksneargenes.CT.all.res<-cbind(Type,peaksneargenes.CT.all.B,peaksneargenes.CT.all.G,peaksneargenes.CT.all.L)
colnames(peaksneargenes.CT.all.res)<-c("Type","CT.B","CT.G","CT.L")
peaksneargenes.CT.all.res<-peaksneargenes.CT.all.res[c(1:3,7,4:6),]
peaksneargenes.CT.all.res<-as.data.frame(peaksneargenes.CT.all.res)
peaksneargenes.CT.all.res$CT.B<-as.numeric(peaksneargenes.CT.all.res$CT.B)
peaksneargenes.CT.all.res$CT.G<-as.numeric(peaksneargenes.CT.all.res$CT.G)
peaksneargenes.CT.all.res$CT.L<-as.numeric(peaksneargenes.CT.all.res$CT.L)
#make into proportions for direct comparison to all CREs:
peaksneargenes.CT.all.res$CT.B<-peaksneargenes.CT.all.res$CT.B/sum(peaksneargenes.CT.all.res$CT.B)
peaksneargenes.CT.all.res$CT.G<-peaksneargenes.CT.all.res$CT.G/sum(peaksneargenes.CT.all.res$CT.G)
peaksneargenes.CT.all.res$CT.L<-peaksneargenes.CT.all.res$CT.L/sum(peaksneargenes.CT.all.res$CT.L)
peaksneargenes.CT.all.melt<-melt(peaksneargenes.CT.all.res, id="Type")
peaksneargenes.CT.all.melt.plot<-ggplot(peaksneargenes.CT.all.melt, aes(x=variable, y=value,color=Type, group=Type))+
  geom_line(size=2)+theme_classic()+theme(panel.background = element_rect(fill = "transparent",colour = NA), plot.background = element_rect(fill = "transparent",colour = NA), legend.background = element_rect(fill = "transparent",colour = NA))+ 
  xlab("") + ylab("Proportion of Peaks") + ggtitle("") + theme(plot.title = element_text(size=24, vjust = 1.5, face = "bold"), axis.text = element_text(size=20), axis.title=element_text(size=22),legend.text = element_text(size=19), legend.title=element_text(size=19), legend.key = element_rect(size = 5, colour = NA), legend.key.size = unit(1.5, 'lines'))+
  ylim(0,1.0)
#ggsave("./peaksneargenes.CT.all.plot.pdf", peaksneargenes.CT.all.melt.plot, width=8, height=6)

Type<-c("All cis", "All trans", "Ambig", "Cis by trans", "Compensatory", "Conserved", "Cis plus trans")
GRN.CT.peaks.all<-right_join(CT.peaks.all.genefunct,GRNgeneslist, by="SPU.hit")
GRN.CT.peaks.all<-na.omit(GRN.CT.peaks.all)
#reorder to match Types order:
GRN.CT.peaks.all$CT.B<-gsub("CbyT","cbyT",GRN.CT.peaks.all$CT.B)
GRN.CT.peaks.all$CT.B<-gsub("CplusT","cplusT",GRN.CT.peaks.all$CT.B)
GRN.CT.peaks.all$CT.G<-gsub("CbyT","cbyT",GRN.CT.peaks.all$CT.G)
GRN.CT.peaks.all$CT.G<-gsub("CplusT","cplusT",GRN.CT.peaks.all$CT.G)
GRN.CT.peaks.all$CT.L<-gsub("CbyT","cbyT",GRN.CT.peaks.all$CT.L)
GRN.CT.peaks.all$CT.L<-gsub("CplusT","cplusT",GRN.CT.peaks.all$CT.L)

GRN.CT.peaks.test.B<-GRN.CT.peaks.all %>% group_by(CT.B) %>% tally()
GRN.CT.peaks.test.B<-GRN.CT.peaks.test.B$n
GRN.CT.peaks.test.G<-GRN.CT.peaks.all %>% group_by(CT.G) %>% tally()
GRN.CT.peaks.test.G<-GRN.CT.peaks.test.G$n
GRN.CT.peaks.test.L<-GRN.CT.peaks.all %>% group_by(CT.L) %>% tally()
GRN.CT.peaks.test.L<-GRN.CT.peaks.test.L$n
GRN.CT.peaks.test.res<-cbind(Type,GRN.CT.peaks.test.B,GRN.CT.peaks.test.G,GRN.CT.peaks.test.L)
colnames(GRN.CT.peaks.test.res)<-c("Type","CT.B","CT.G","CT.L")
GRN.CT.peaks.test.res<-GRN.CT.peaks.test.res[c(1:3,7,4:6),]
GRN.CT.peaks.test.res<-as.data.frame(GRN.CT.peaks.test.res)
GRN.CT.peaks.test.res$CT.B<-as.numeric(GRN.CT.peaks.test.res$CT.B)
GRN.CT.peaks.test.res$CT.G<-as.numeric(GRN.CT.peaks.test.res$CT.G)
GRN.CT.peaks.test.res$CT.L<-as.numeric(GRN.CT.peaks.test.res$CT.L)
#make into proportions for direct comparison to all CREs:
GRN.CT.peaks.test.res.prop<-GRN.CT.peaks.test.res$CT.B/sum(GRN.CT.peaks.test.res$CT.B)
GRN.CT.peaks.test.res.prop<-as.data.frame(GRN.CT.peaks.test.res.prop)
colnames(GRN.CT.peaks.test.res.prop)
GRN.CT.peaks.test.res.prop$CT.G<-GRN.CT.peaks.test.res$CT.G/sum(GRN.CT.peaks.test.res$CT.G)
GRN.CT.peaks.test.res.prop$CT.L<-GRN.CT.peaks.test.res$CT.L/sum(GRN.CT.peaks.test.res$CT.L)
GRN.CT.peaks.test.res.prop$Type<-GRN.CT.peaks.test.res$Type
GRN.CT.test.res.prop<-melt(GRN.CT.peaks.test.res.prop, id="Type")
GRN.CT.test.res.prop.plot<-ggplot(GRN.CT.test.res.prop, aes(x=variable, y=value,color=Type, group=Type))+
  geom_line(size=2)+theme_classic()+theme(panel.background = element_rect(fill = "transparent",colour = NA), plot.background = element_rect(fill = "transparent",colour = NA), legend.background = element_rect(fill = "transparent",colour = NA))+ 
  xlab("") + ylab("Proportion of Peaks") + ggtitle("") + theme(plot.title = element_text(size=24, vjust = 1.5, face = "bold"), axis.text = element_text(size=20), axis.title=element_text(size=22),legend.text = element_text(size=19), legend.title=element_text(size=19), legend.key = element_rect(size = 5, colour = NA), legend.key.size = unit(1.5, 'lines'))+
  ylim(0,1.0)

#ggsave("./GRN.CT.test.res.prop.plot.pdf", GRN.CT.test.res.prop.plot, width=8, height=6)
```

```{r}
#save.image(file='hybrids_forpublication.RData') 
load('hybrids_forpublication.RData')
```


#Connection between DE genes and DA peaks: contingency chi-squared tests in Figure 5A/5B:
```{r}
library(tidyr)
#looking at peaks near (<25kb) cis/trans genes from LY's data (got this file and edited it from the output of his R code):
genesregmode<-read.csv("lingyu.genes.CT.csv")
genesCT.B<-genesregmode %>% select(SPU.hit, gCT.B)
genesCT.G<-genesregmode %>% select(SPU.hit, gCT.G)
genesCT.L<-genesregmode %>% select(SPU.hit, gCT.L)

#make two contingency tables:
#1) from perspective of peaks: rows are DA/not DA, columns are nearest gene is DE, nearest gene not DE
#2) from perspective of genes: rows are DE/not DE, columns are "there is at least one DA peak within 25kb"/"there is not"

#Peaks-first approach:
DApeaks_DEgenes.contingency.B<-inner_join(CT.peaks.all.genefunct %>% dplyr::rename(peak=peakno),(Hery_peak_gene_map_final %>% select(peak)))
DApeaks_DEgenes.contingency.B<-DApeaks_DEgenes.contingency.B %>% filter(PeakCoord %in% genesnearpeaks$PeakCoord) %>% select(PeakCoord, CT.B, SPU.hit)

DApeaks_DEgenes.contingency.B<-left_join(DApeaks_DEgenes.contingency.B, genesCT.B, by="SPU.hit")
DApeaks_DEgenes.contingency.B<-na.omit(DApeaks_DEgenes.contingency.B)
DApeaks_DEgenes.contingency.B$geneDE<-ifelse((DApeaks_DEgenes.contingency.B$gCT.B=="all cis" | DApeaks_DEgenes.contingency.B$gCT.B=="all trans" | DApeaks_DEgenes.contingency.B$gCT.B=="cis x trans" | DApeaks_DEgenes.contingency.B$gCT.B=="cis + trans" | DApeaks_DEgenes.contingency.B$gCT.B=="Compensatory"),"gene_DA","gene_notDE")
DApeaks_DEgenes.contingency.B$peakDA<-ifelse((DApeaks_DEgenes.contingency.B$CT.B!="conserved"& DApeaks_DEgenes.contingency.B$CT.B!="ambig"),"peak_DA","peak_notDA")
DApeaks_DEgenes.contingency.table.B<-DApeaks_DEgenes.contingency.B %>% filter(peakDA=="peak_DA") %>% group_by(geneDE) %>% tally()
DApeaks_DEgenes.contingency.table.B<-DApeaks_DEgenes.contingency.table.B %>% dplyr::rename(PeakisDA=n)
DApeaks_DEgenes.contingency.table.B<-cbind(DApeaks_DEgenes.contingency.table.B,DApeaks_DEgenes.contingency.B %>% filter(peakDA=="peak_notDA") %>% group_by(geneDE) %>% tally() %>% select(-geneDE))
colnames(DApeaks_DEgenes.contingency.table.B)<-c("geneDE","PeakDA","PeaknotDA")
row.names(DApeaks_DEgenes.contingency.table.B)<-DApeaks_DEgenes.contingency.table.B$geneDE
DApeaks_DEgenes.contingency.table.B<-DApeaks_DEgenes.contingency.table.B %>% select(-geneDE)
DApeaks_DEgenes.contingency.table.B[] <- sapply(DApeaks_DEgenes.contingency.table.B[], as.numeric)
DApeaks_DEgenes.contingency.table.B.test<-chisq.test(as.matrix(DApeaks_DEgenes.contingency.table.B))
DApeaks_DEgenes.contingency.table.B.test$observed

DApeaks_DEgenes.contingency.G<-inner_join(CT.peaks.all.genefunct %>% dplyr::rename(peak=peakno),(Hery_peak_gene_map_final %>% select(peak)))
DApeaks_DEgenes.contingency.G<-DApeaks_DEgenes.contingency.G %>% filter(PeakCoord %in% genesnearpeaks$PeakCoord) %>% select(PeakCoord, CT.G, SPU.hit)

DApeaks_DEgenes.contingency.G<-left_join(DApeaks_DEgenes.contingency.G, genesCT.G, by="SPU.hit")
DApeaks_DEgenes.contingency.G<-na.omit(DApeaks_DEgenes.contingency.G)
DApeaks_DEgenes.contingency.G$geneDE<-ifelse((DApeaks_DEgenes.contingency.G$gCT.G=="all cis" | DApeaks_DEgenes.contingency.G$gCT.G=="all trans" | DApeaks_DEgenes.contingency.G$gCT.G=="cis x trans" | DApeaks_DEgenes.contingency.G$gCT.G=="cis + trans" | DApeaks_DEgenes.contingency.G$gCT.G=="Compensatory"),"gene_DA","gene_notDE")
DApeaks_DEgenes.contingency.G$peakDA<-ifelse((DApeaks_DEgenes.contingency.G$CT.G!="conserved"& DApeaks_DEgenes.contingency.G$CT.G!="ambig"),"peak_DA","peak_notDA")
DApeaks_DEgenes.contingency.table.G<-DApeaks_DEgenes.contingency.G %>% filter(peakDA=="peak_DA") %>% group_by(geneDE) %>% tally()
DApeaks_DEgenes.contingency.table.G<-DApeaks_DEgenes.contingency.table.G %>% dplyr::rename(PeakisDA=n)
DApeaks_DEgenes.contingency.table.G<-cbind(DApeaks_DEgenes.contingency.table.G,DApeaks_DEgenes.contingency.G %>% filter(peakDA=="peak_notDA") %>% group_by(geneDE) %>% tally() %>% select(-geneDE))
colnames(DApeaks_DEgenes.contingency.table.G)<-c("geneDE","PeakDA","PeaknotDA")
row.names(DApeaks_DEgenes.contingency.table.G)<-DApeaks_DEgenes.contingency.table.G$geneDE
DApeaks_DEgenes.contingency.table.G<-DApeaks_DEgenes.contingency.table.G %>% select(-geneDE)
DApeaks_DEgenes.contingency.table.G[] <- sapply(DApeaks_DEgenes.contingency.table.G[], as.numeric)
DApeaks_DEgenes.contingency.table.G.test<-chisq.test(as.matrix(DApeaks_DEgenes.contingency.table.G))
DApeaks_DEgenes.contingency.table.G.test

DApeaks_DEgenes.contingency.L<-inner_join(CT.peaks.all.genefunct %>% dplyr::rename(peak=peakno),(Hery_peak_gene_map_final %>% select(peak)))
DApeaks_DEgenes.contingency.L<-DApeaks_DEgenes.contingency.L %>% filter(PeakCoord %in% genesnearpeaks$PeakCoord) %>% select(PeakCoord, CT.L, SPU.hit)

DApeaks_DEgenes.contingency.L<-left_join(DApeaks_DEgenes.contingency.L, genesCT.L, by="SPU.hit")
DApeaks_DEgenes.contingency.L<-na.omit(DApeaks_DEgenes.contingency.L)
DApeaks_DEgenes.contingency.L$geneDE<-ifelse((DApeaks_DEgenes.contingency.L$gCT.L=="all cis" | DApeaks_DEgenes.contingency.L$gCT.L=="all trans" | DApeaks_DEgenes.contingency.L$gCT.L=="cis x trans" | DApeaks_DEgenes.contingency.L$gCT.L=="cis + trans" | DApeaks_DEgenes.contingency.L$gCT.L=="Compensatory"),"gene_DA","gene_notDE")
DApeaks_DEgenes.contingency.L$peakDA<-ifelse((DApeaks_DEgenes.contingency.L$CT.L!="conserved"& DApeaks_DEgenes.contingency.L$CT.L!="ambig"),"peak_DA","peak_notDA")
DApeaks_DEgenes.contingency.table.L<-DApeaks_DEgenes.contingency.L %>% filter(peakDA=="peak_DA") %>% group_by(geneDE) %>% tally()
DApeaks_DEgenes.contingency.table.L<-DApeaks_DEgenes.contingency.table.L %>% dplyr::rename(PeakisDA=n)
DApeaks_DEgenes.contingency.table.L<-cbind(DApeaks_DEgenes.contingency.table.L,DApeaks_DEgenes.contingency.L %>% filter(peakDA=="peak_notDA") %>% group_by(geneDE) %>% tally() %>% select(-geneDE))
colnames(DApeaks_DEgenes.contingency.table.L)<-c("geneDE","PeakDA","PeaknotDA")
row.names(DApeaks_DEgenes.contingency.table.L)<-DApeaks_DEgenes.contingency.table.L$geneDE
DApeaks_DEgenes.contingency.table.L<-DApeaks_DEgenes.contingency.table.L %>% select(-geneDE)
DApeaks_DEgenes.contingency.table.L[] <- sapply(DApeaks_DEgenes.contingency.table.L[], as.numeric)
DApeaks_DEgenes.contingency.table.L.test<-chisq.test(as.matrix(DApeaks_DEgenes.contingency.table.L))
DApeaks_DEgenes.contingency.table.L.test


DApeaks_DEgenes.contingency.table.B.test.resid<-melt(DApeaks_DEgenes.contingency.table.B.test$residuals)
colnames(DApeaks_DEgenes.contingency.table.B.test.resid)<-c("geneDEornot","peakDAornot","value")

DApeaks_DEgenes.contingency.table.B.test.resid.plot<-ggplot(DApeaks_DEgenes.contingency.table.B.test.resid, aes(x=geneDEornot, y=peakDAornot, colour=value, size=abs(value)))+
  geom_point()+
  scale_color_gradient2(low="blue",
  mid = "white",
  high = "red",
  limits=c(-7,7))+
  scale_size(range=c(5,50))

DApeaks_DEgenes.contingency.table.G.test.resid<-melt(DApeaks_DEgenes.contingency.table.G.test$residuals)
colnames(DApeaks_DEgenes.contingency.table.G.test.resid)<-c("geneDEornot","peakDAornot","value")

DApeaks_DEgenes.contingency.table.G.test.resid.plot<-ggplot(DApeaks_DEgenes.contingency.table.G.test.resid, aes(x=geneDEornot, y=peakDAornot, colour=value, size=abs(value)))+
  geom_point()+
  scale_color_gradient2(low="blue",
  mid = "white",
  high = "red",
  limits=c(-7,7))+
  scale_size(range=c(5,50))

DApeaks_DEgenes.contingency.table.L.test.resid<-melt(DApeaks_DEgenes.contingency.table.L.test$residuals)
colnames(DApeaks_DEgenes.contingency.table.L.test.resid)<-c("geneDEornot","peakDAornot","value")

DApeaks_DEgenes.contingency.table.L.test.resid.plot<-ggplot(DApeaks_DEgenes.contingency.table.L.test.resid, aes(x=geneDEornot, y=peakDAornot, colour=value, size=abs(value)))+
  geom_point()+
  scale_color_gradient2(low="blue",
  mid = "white",
  high = "red",
  limits=c(-7,7))+
  scale_size(range=c(5,50))

#ggsave("DApeaks_DEgenes.contingency.table.B.test.resid.plot.pdf",DApeaks_DEgenes.contingency.table.B.test.resid.plot)
#ggsave("DApeaks_DEgenes.contingency.table.G.test.resid.plot.pdf",DApeaks_DEgenes.contingency.table.G.test.resid.plot)
#ggsave("DApeaks_DEgenes.contingency.table.L.test.resid.plot.pdf",DApeaks_DEgenes.contingency.table.L.test.resid.plot)


#now genes-first approach:

DEgenes_DApeaks.contingency.B<-CT.peaks.all.genefunct %>% filter(PeakCoord %in% peaksneargenes$PeakCoord) %>% select(PeakCoord, CT.B, SPU.hit)
DEgenes_DApeaks.contingency.B<-inner_join(DEgenes_DApeaks.contingency.B, genesCT.B, by="SPU.hit")
DEgenes_DApeaks.contingency.B$geneDE<-ifelse((DEgenes_DApeaks.contingency.B$gCT.B=="all cis" | DEgenes_DApeaks.contingency.B$gCT.B=="all trans" | DEgenes_DApeaks.contingency.B$gCT.B=="cis x trans" | DEgenes_DApeaks.contingency.B$gCT.B=="cis + trans" | DEgenes_DApeaks.contingency.B$gCT.B=="Compensatory"),"gene_DA","gene_notDE")
DEgenes_DApeaks.contingency.B$peakDA<-ifelse((DEgenes_DApeaks.contingency.B$CT.B!="conserved"& DEgenes_DApeaks.contingency.B$CT.B!="ambig"),"peak_DA","peak_notDA")
DEgenes_DApeaks.contingency.table.B<-DEgenes_DApeaks.contingency.B %>% filter(geneDE=="gene_DA") %>% group_by(peakDA) %>% tally()
DEgenes_DApeaks.contingency.table.B<-DEgenes_DApeaks.contingency.table.B %>% dplyr::rename(GeneisDA=n)
DEgenes_DApeaks.contingency.table.B<-cbind(DEgenes_DApeaks.contingency.table.B,DEgenes_DApeaks.contingency.B %>% filter(geneDE=="gene_notDE") %>% group_by(peakDA) %>% tally() %>% select(-peakDA))
colnames(DEgenes_DApeaks.contingency.table.B)<-c("peakDA","GeneDA","GenenotDA")
row.names(DEgenes_DApeaks.contingency.table.B)<-DEgenes_DApeaks.contingency.table.B$peakDA
DEgenes_DApeaks.contingency.table.B<-DEgenes_DApeaks.contingency.table.B %>% select(-peakDA)
DEgenes_DApeaks.contingency.table.B[] <- sapply(DEgenes_DApeaks.contingency.table.B[], as.numeric)
DEgenes_DApeaks.contingency.table.B.test<-chisq.test(as.matrix(DEgenes_DApeaks.contingency.table.B))

DEgenes_DApeaks.contingency.G<-CT.peaks.all.genefunct %>% filter(PeakCoord %in% peaksneargenes$PeakCoord) %>% select(PeakCoord, CT.G, SPU.hit)
DEgenes_DApeaks.contingency.G<-inner_join(DEgenes_DApeaks.contingency.G, genesCT.G, by="SPU.hit")
DEgenes_DApeaks.contingency.G$geneDE<-ifelse((DEgenes_DApeaks.contingency.G$gCT.G=="all cis" | DEgenes_DApeaks.contingency.G$gCT.G=="all trans" | DEgenes_DApeaks.contingency.G$gCT.G=="cis x trans" | DEgenes_DApeaks.contingency.G$gCT.G=="cis + trans" | DEgenes_DApeaks.contingency.G$gCT.G=="Compensatory"),"gene_DA","gene_notDE")
DEgenes_DApeaks.contingency.G$peakDA<-ifelse((DEgenes_DApeaks.contingency.G$CT.G!="conserved"& DEgenes_DApeaks.contingency.G$CT.G!="ambig"),"peak_DA","peak_notDA")
DEgenes_DApeaks.contingency.table.G<-DEgenes_DApeaks.contingency.G %>% filter(geneDE=="gene_DA") %>% group_by(peakDA) %>% tally()
DEgenes_DApeaks.contingency.table.G<-DEgenes_DApeaks.contingency.table.G %>% dplyr::rename(GeneisDA=n)
DEgenes_DApeaks.contingency.table.G<-cbind(DEgenes_DApeaks.contingency.table.G,DEgenes_DApeaks.contingency.G %>% filter(geneDE=="gene_notDE") %>% group_by(peakDA) %>% tally() %>% select(-peakDA))
colnames(DEgenes_DApeaks.contingency.table.G)<-c("peakDA","GeneDA","GenenotDA")
row.names(DEgenes_DApeaks.contingency.table.G)<-DEgenes_DApeaks.contingency.table.G$peakDA
DEgenes_DApeaks.contingency.table.G<-DEgenes_DApeaks.contingency.table.G %>% select(-peakDA)
DEgenes_DApeaks.contingency.table.G[] <- sapply(DEgenes_DApeaks.contingency.table.G[], as.numeric)
DEgenes_DApeaks.contingency.table.G.test<-chisq.test(as.matrix(DEgenes_DApeaks.contingency.table.G))

DEgenes_DApeaks.contingency.L<-CT.peaks.all.genefunct %>% filter(PeakCoord %in% peaksneargenes$PeakCoord) %>% select(PeakCoord, CT.L, SPU.hit)
DEgenes_DApeaks.contingency.L<-left_join(DEgenes_DApeaks.contingency.L, genesCT.L, by="SPU.hit")
DEgenes_DApeaks.contingency.L$geneDE<-ifelse((DEgenes_DApeaks.contingency.L$gCT.L=="all cis" | DEgenes_DApeaks.contingency.L$gCT.L=="all trans" | DEgenes_DApeaks.contingency.L$gCT.L=="cis x trans" | DEgenes_DApeaks.contingency.L$gCT.L=="cis + trans" | DEgenes_DApeaks.contingency.L$gCT.L=="Compensatory"),"gene_DA","gene_notDE")
DEgenes_DApeaks.contingency.L$peakDA<-ifelse((DEgenes_DApeaks.contingency.L$CT.L!="conserved"& DEgenes_DApeaks.contingency.L$CT.L!="ambig"),"peak_DA","peak_notDA")
DEgenes_DApeaks.contingency.table.L<-DEgenes_DApeaks.contingency.L %>% filter(geneDE=="gene_DA") %>% group_by(peakDA) %>% tally()
DEgenes_DApeaks.contingency.table.L<-DEgenes_DApeaks.contingency.table.L %>% dplyr::rename(GeneisDA=n)
DEgenes_DApeaks.contingency.table.L<-cbind(DEgenes_DApeaks.contingency.table.L,DEgenes_DApeaks.contingency.L %>% filter(geneDE=="gene_notDE") %>% group_by(peakDA) %>% tally() %>% select(-peakDA))
colnames(DEgenes_DApeaks.contingency.table.L)<-c("peakDA","GeneDA","GenenotDA")
row.names(DEgenes_DApeaks.contingency.table.L)<-DEgenes_DApeaks.contingency.table.L$peakDA
DEgenes_DApeaks.contingency.table.L<-DEgenes_DApeaks.contingency.table.L %>% select(-peakDA)
DEgenes_DApeaks.contingency.table.L[] <- sapply(DEgenes_DApeaks.contingency.table.L[], as.numeric)
DEgenes_DApeaks.contingency.table.L.test<-chisq.test(as.matrix(DEgenes_DApeaks.contingency.table.L))

DEgenes_DApeaks.contingency.table.B.test.resid<-melt(DEgenes_DApeaks.contingency.table.B.test$residuals)
colnames(DEgenes_DApeaks.contingency.table.B.test.resid)<-c("peakDAornot","geneDEornot","value")

DEgenes_DApeaks.contingency.table.B.test.resid.plot<-ggplot(DEgenes_DApeaks.contingency.table.B.test.resid, aes(x=peakDAornot, y=geneDEornot, colour=value, size=abs(value)))+
  geom_point()+
  scale_color_gradient2(low="blue",
  mid = "white",
  high = "red",
  limits=c(-7,7))+
  scale_size(range=c(5,50))

DEgenes_DApeaks.contingency.table.G.test.resid<-melt(DEgenes_DApeaks.contingency.table.G.test$residuals)
colnames(DEgenes_DApeaks.contingency.table.G.test.resid)<-c("peakDAornot","geneDEornot","value")

DEgenes_DApeaks.contingency.table.G.test.resid.plot<-ggplot(DEgenes_DApeaks.contingency.table.G.test.resid, aes(x=peakDAornot, y=geneDEornot, colour=value, size=abs(value)))+
  geom_point()+
  scale_color_gradient2(low="blue",
  mid = "white",
  high = "red",
  limits=c(-7,7))+
  scale_size(range=c(5,50))

DEgenes_DApeaks.contingency.table.L.test.resid<-melt(DEgenes_DApeaks.contingency.table.L.test$residuals)
colnames(DEgenes_DApeaks.contingency.table.L.test.resid)<-c("peakDAornot","geneDEornot","value")

DEgenes_DApeaks.contingency.table.L.test.resid.plot<-ggplot(DEgenes_DApeaks.contingency.table.L.test.resid, aes(x=peakDAornot, y=geneDEornot, colour=value, size=abs(value)))+
  geom_point()+
  scale_color_gradient2(low="blue",
  mid = "white",
  high = "red",
  limits=c(-7,7))+
  scale_size(range=c(5,50))

#ggsave("DEgenes_DApeaks.contingency.table.B.test.resid.plot.pdf",DEgenes_DApeaks.contingency.table.B.test.resid.plot)
#ggsave("DEgenes_DApeaks.contingency.table.G.test.resid.plot.pdf",DEgenes_DApeaks.contingency.table.G.test.resid.plot)
#ggsave("DEgenes_DApeaks.contingency.table.L.test.resid.plot.pdf",DEgenes_DApeaks.contingency.table.L.test.resid.plot)


DApeaks_DEgenes.contingency.table.B.test
DApeaks_DEgenes.contingency.table.G.test
DApeaks_DEgenes.contingency.table.L.test

DEgenes_DApeaks.contingency.table.B.test
DEgenes_DApeaks.contingency.table.G.test
DEgenes_DApeaks.contingency.table.L.test
#all p<.05 except blastulas

```

Now restrict DE genes to only "all cis" and "all trans" genes: which ones are enriched for nearby DA peaks? (Figure 5C):
```{r}
peaksnearcistransgenes.B<-CT.peaks.all.genefunct %>% filter(PeakCoord %in% peaksneargenes$PeakCoord) %>% select(PeakCoord, CT.B, SPU.hit)
peaksnearcistransgenes.B<-left_join(peaksnearcistransgenes.B, genesCT.B, by="SPU.hit")
peaksnearcistransgenes.B.contingency<-peaksnearcistransgenes.B %>% filter(gCT.B=="all cis") %>% group_by(CT.B) %>% tally() %>% dplyr::rename(g_allcis=n)
peaksnearcistransgenes.B.DAornot<-peaksnearcistransgenes.B %>% filter(PeakCoord %in% row.names(ase_nolowcounts.CT.B.all %>% filter(TT_HEvsHTpadj<=0.05))) %>% select(PeakCoord,gCT.B) %>% group_by(gCT.B) %>% tally() %>% dplyr::rename(DA=n)
peaksnearcistransgenes.B.DAornot$notDA<-peaksnearcistransgenes.B %>% filter(PeakCoord %in% row.names(ase_nolowcounts.CT.B.all %>% filter(TT_HEvsHTpadj>0.05))) %>% select(PeakCoord,gCT.B) %>% group_by(gCT.B) %>% tally() %>% select(n)
peaksnearcistransgenes.B.DAornot<-as.data.frame(peaksnearcistransgenes.B.DAornot)
peaksnearcistransgenes.B.DAornot<-peaksnearcistransgenes.B.DAornot[c(1:2),]
row.names(peaksnearcistransgenes.B.DAornot)<-peaksnearcistransgenes.B.DAornot$gCT.B
peaksnearcistransgenes.B.DAornot<-peaksnearcistransgenes.B.DAornot %>% select(-gCT.B)
peaksnearcistransgenes.B.DAornot<-t(peaksnearcistransgenes.B.DAornot)
peaksnearcistransgenes.B.DAornot[] <- sapply(peaksnearcistransgenes.B.DAornot[], as.numeric)
peaksnearcistransgenes.B.DAornot.test<-chisq.test(as.matrix(peaksnearcistransgenes.B.DAornot))

peaksnearcistransgenes.G<-CT.peaks.all.genefunct %>% filter(PeakCoord %in% peaksneargenes$PeakCoord) %>% select(PeakCoord, CT.G, SPU.hit)
peaksnearcistransgenes.G<-left_join(peaksnearcistransgenes.G, genesCT.G, by="SPU.hit")
peaksnearcistransgenes.G.contingency<-peaksnearcistransgenes.G %>% filter(gCT.G=="all cis") %>% group_by(CT.G) %>% tally() %>% dplyr::rename(g_allcis=n)
peaksnearcistransgenes.G.DAornot<-peaksnearcistransgenes.G %>% filter(PeakCoord %in% row.names(ase_nolowcounts.CT.G.all %>% filter(TT_HEvsHTpadj<=0.05))) %>% select(PeakCoord,gCT.G) %>% group_by(gCT.G) %>% tally() %>% dplyr::rename(DA=n)
peaksnearcistransgenes.G.DAornot$notDA<-peaksnearcistransgenes.G %>% filter(PeakCoord %in% row.names(ase_nolowcounts.CT.G.all %>% filter(TT_HEvsHTpadj>0.05))) %>% select(PeakCoord,gCT.G) %>% group_by(gCT.G) %>% tally() %>% select(n)
peaksnearcistransgenes.G.DAornot<-as.data.frame(peaksnearcistransgenes.G.DAornot)
peaksnearcistransgenes.G.DAornot<-peaksnearcistransgenes.G.DAornot[c(1:2),]
row.names(peaksnearcistransgenes.G.DAornot)<-peaksnearcistransgenes.G.DAornot$gCT.G
peaksnearcistransgenes.G.DAornot<-peaksnearcistransgenes.G.DAornot %>% select(-gCT.G)
peaksnearcistransgenes.G.DAornot<-t(peaksnearcistransgenes.G.DAornot)
peaksnearcistransgenes.G.DAornot[] <- sapply(peaksnearcistransgenes.G.DAornot[], as.numeric)
peaksnearcistransgenes.G.DAornot.test<-chisq.test(as.matrix(peaksnearcistransgenes.G.DAornot))

peaksnearcistransgenes.L<-CT.peaks.all.genefunct %>% filter(PeakCoord %in% peaksneargenes$PeakCoord) %>% select(PeakCoord, CT.L, SPU.hit)
peaksnearcistransgenes.L<-left_join(peaksnearcistransgenes.L, genesCT.L, by="SPU.hit")
peaksnearcistransgenes.L.contingency<-peaksnearcistransgenes.L %>% filter(gCT.L=="all cis") %>% group_by(CT.L) %>% tally() %>% dplyr::rename(g_allcis=n)
peaksnearcistransgenes.L.DAornot<-peaksnearcistransgenes.L %>% filter(PeakCoord %in% row.names(ase_nolowcounts.CT.L.all %>% filter(TT_HEvsHTpadj<=0.05))) %>% select(PeakCoord,gCT.L) %>% group_by(gCT.L) %>% tally() %>% dplyr::rename(DA=n)
peaksnearcistransgenes.L.DAornot$notDA<-peaksnearcistransgenes.L %>% filter(PeakCoord %in% row.names(ase_nolowcounts.CT.L.all %>% filter(TT_HEvsHTpadj>0.05))) %>% select(PeakCoord,gCT.L) %>% group_by(gCT.L) %>% tally() %>% select(n)
peaksnearcistransgenes.L.DAornot<-as.data.frame(peaksnearcistransgenes.L.DAornot)
peaksnearcistransgenes.L.DAornot<-peaksnearcistransgenes.L.DAornot[c(1:2),]
row.names(peaksnearcistransgenes.L.DAornot)<-peaksnearcistransgenes.L.DAornot$gCT.L
peaksnearcistransgenes.L.DAornot<-peaksnearcistransgenes.L.DAornot %>% select(-gCT.L)
peaksnearcistransgenes.L.DAornot<-t(peaksnearcistransgenes.L.DAornot)
peaksnearcistransgenes.L.DAornot[] <- sapply(peaksnearcistransgenes.L.DAornot[], as.numeric)
peaksnearcistransgenes.L.DAornot.test<-chisq.test(as.matrix(peaksnearcistransgenes.L.DAornot))

peaksnearcistransgenes.B.DAornot.resid<-melt(peaksnearcistransgenes.B.DAornot.test$residuals)
colnames(peaksnearcistransgenes.B.DAornot.resid)<-c("peaks","genes","value")

peaksnearcistransgenes.B.DAornot.resid.plot<-ggplot(peaksnearcistransgenes.B.DAornot.resid, aes(x=peaks, y=genes, colour=value, size=abs(value)))+
  geom_point()+
  scale_color_gradient2(low="blue",
  mid = "white",
  high = "red",
  limits=c(-6,6))+
  scale_size(range=c(5,20))

#ggsave("peaksnearcistransgenes.B.DAornot.resid.plot.pdf",peaksnearcistransgenes.B.DAornot.resid.plot) 

peaksnearcistransgenes.G.DAornot.resid<-melt(peaksnearcistransgenes.G.DAornot.test$residuals)
colnames(peaksnearcistransgenes.G.DAornot.resid)<-c("peaks","genes","value")

peaksnearcistransgenes.G.DAornot.resid.plot<-ggplot(peaksnearcistransgenes.G.DAornot.resid, aes(x=peaks, y=genes, colour=value, size=abs(value)))+
  geom_point()+
  scale_color_gradient2(low="blue",
  mid = "white",
  high = "red",
  limits=c(-6,6))+
  scale_size(range=c(5,20))

#ggsave("peaksnearcistransgenes.G.DAornot.resid.plot.pdf",peaksnearcistransgenes.G.DAornot.resid.plot) 

peaksnearcistransgenes.L.DAornot.resid<-melt(peaksnearcistransgenes.L.DAornot.test$residuals)
colnames(peaksnearcistransgenes.L.DAornot.resid)<-c("peaks","genes","value")

peaksnearcistransgenes.L.DAornot.resid.plot<-ggplot(peaksnearcistransgenes.L.DAornot.resid, aes(x=peaks, y=genes, colour=value, size=abs(value)))+
  geom_point()+
  scale_color_gradient2(low="blue",
  mid = "white",
  high = "red",
  limits=c(-6,6))+
  scale_size(range=c(5,20))

#ggsave("peaksnearcistransgenes.L.DAornot.resid.plot.pdf",peaksnearcistransgenes.L.DAornot.resid.plot)

peaksnearcistransgenes.B.DAornot.test
peaksnearcistransgenes.G.DAornot.test
peaksnearcistransgenes.L.DAornot.test

```

#Peaks that change openness from stage to stage:
```{r}
#DE between stages but just within each "species"
ase_nolowcounts.CT_liftedHery.BGL.desc<-data.frame(individual=factor(c(4,7,2,5,8,3,6,9)),
                  stage=factor(c("B","B","G","G","G","L","L","L")))
ase_nolowcounts_Heryonly<-ase_nolowcounts %>% select(Chr,start,end,HeryB4,HeryB7,HeryG2,HeryG5,HeryG8,HeryL3,HeryL6,HeryL9)
rownames(ase_nolowcounts_Heryonly)<-paste0(ase_nolowcounts_Heryonly$Chr,":",ase_nolowcounts_Heryonly$start, "-",ase_nolowcounts_Heryonly$end)
ase_nolowcounts_Heryonly<-ase_nolowcounts_Heryonly %>% select(-Chr,-start,-end)

ase_nolowcounts_Heryonlydds<-DESeqDataSetFromMatrix(ase_nolowcounts_Heryonly,
                                        ase_nolowcounts.CT_liftedHery.BGL.desc,
                                        design = ~stage)
ase_nolowcounts_Heryonlydds<-DESeq(ase_nolowcounts_Heryonlydds)
ase_nolowcounts_Heryonly.BtoG<-results(ase_nolowcounts_Heryonlydds, contrast=c("stage","B","G"))
ase_nolowcounts_Heryonly.BtoG<-as.data.frame(ase_nolowcounts_Heryonly.BtoG)

ase_nolowcounts_Heryonly.GtoL<-results(ase_nolowcounts_Heryonlydds, contrast=c("stage","G","L"))
ase_nolowcounts_Heryonly.GtoL<-as.data.frame(ase_nolowcounts_Heryonly.GtoL)
View(ase_nolowcounts_Heryonly.BtoG)
View(ase_nolowcounts)

ase_nolowcounts.CT_liftedhtub.BGL.desc<-data.frame(individual=factor(c(4,7,2,5,8,3,6,11)),
                  stage=factor(c("B","B","G","G","G","L","L","L")))
ase_nolowcounts_htubonly<-ase_nolowcounts %>% select(Chr,start,end,htubB4,htubB7,htubG2,htubG5,htubG8,htubL3,htubL6,htubL11)
rownames(ase_nolowcounts_htubonly)<-paste0(ase_nolowcounts_htubonly$Chr,":",ase_nolowcounts_htubonly$start, "-",ase_nolowcounts_htubonly$end)
ase_nolowcounts_htubonly<-ase_nolowcounts_htubonly %>% select(-Chr,-start,-end)

ase_nolowcounts_htubonlydds<-DESeqDataSetFromMatrix(ase_nolowcounts_htubonly,
                                        ase_nolowcounts.CT_liftedhtub.BGL.desc,
                                        design = ~stage)
ase_nolowcounts_htubonlydds<-DESeq(ase_nolowcounts_htubonlydds)
ase_nolowcounts_htubonly.BtoG<-results(ase_nolowcounts_htubonlydds, contrast=c("stage","B","G"))
ase_nolowcounts_htubonly.BtoG<-as.data.frame(ase_nolowcounts_htubonly.BtoG)

ase_nolowcounts_htubonly.GtoL<-results(ase_nolowcounts_htubonlydds, contrast=c("stage","G","L"))
ase_nolowcounts_htubonly.GtoL<-as.data.frame(ase_nolowcounts_htubonly.GtoL)


ase_nolowcounts.CT_liftedHybHE.BGL.desc<-data.frame(individual=factor(c(12,13,7,8,9,1,2,11)),
                  stage=factor(c("B","B","G","G","G","L","L","L")))
ase_nolowcounts_HybHEonly<-ase_nolowcounts %>% select(Chr,start,end,HybHEB12,HybHEB13,HybHEG7,HybHEG8,HybHEG9,HybHEL1,HybHEL2, HybHEL11)
rownames(ase_nolowcounts_HybHEonly)<-paste0(ase_nolowcounts_HybHEonly$Chr,":",ase_nolowcounts_HybHEonly$start, "-",ase_nolowcounts_HybHEonly$end)
ase_nolowcounts_HybHEonly<-ase_nolowcounts_HybHEonly %>% select(-Chr,-start,-end)

ase_nolowcounts_HybHEonlydds<-DESeqDataSetFromMatrix(ase_nolowcounts_HybHEonly,
                                        ase_nolowcounts.CT_liftedHybHE.BGL.desc,
                                        design = ~stage)
ase_nolowcounts_HybHEonlydds<-DESeq(ase_nolowcounts_HybHEonlydds)
ase_nolowcounts_HybHEonly.BtoG<-results(ase_nolowcounts_HybHEonlydds, contrast=c("stage","B","G"))
ase_nolowcounts_HybHEonly.BtoG<-as.data.frame(ase_nolowcounts_HybHEonly.BtoG)

ase_nolowcounts_HybHEonly.GtoL<-results(ase_nolowcounts_HybHEonlydds, contrast=c("stage","G","L"))
ase_nolowcounts_HybHEonly.GtoL<-as.data.frame(ase_nolowcounts_HybHEonly.GtoL)


ase_nolowcounts.CT_liftedHybHT.BGL.desc<-data.frame(individual=factor(c(12,13,7,8,9,1,2,11)),
                  stage=factor(c("B","B","G","G","G","L","L","L")))
ase_nolowcounts_HybHTonly<-ase_nolowcounts %>% select(Chr,start,end,HybHTB12,HybHTB13,HybHTG7,HybHTG8,HybHTG9,HybHTL1,HybHTL2, HybHTL11)
rownames(ase_nolowcounts_HybHTonly)<-paste0(ase_nolowcounts_HybHTonly$Chr,":",ase_nolowcounts_HybHTonly$start, "-",ase_nolowcounts_HybHTonly$end)
ase_nolowcounts_HybHTonly<-ase_nolowcounts_HybHTonly %>% select(-Chr,-start,-end)

ase_nolowcounts_HybHTonlydds<-DESeqDataSetFromMatrix(ase_nolowcounts_HybHTonly,
                                        ase_nolowcounts.CT_liftedHybHT.BGL.desc,
                                        design = ~stage)
ase_nolowcounts_HybHTonlydds<-DESeq(ase_nolowcounts_HybHTonlydds)
ase_nolowcounts_HybHTonly.BtoG<-results(ase_nolowcounts_HybHTonlydds, contrast=c("stage","B","G"))
ase_nolowcounts_HybHTonly.BtoG<-as.data.frame(ase_nolowcounts_HybHTonly.BtoG)

ase_nolowcounts_HybHTonly.GtoL<-results(ase_nolowcounts_HybHTonlydds, contrast=c("stage","G","L"))
ase_nolowcounts_HybHTonly.GtoL<-as.data.frame(ase_nolowcounts_HybHTonly.GtoL)


ase_nolowcounts.CT_liftedHyb.BGL.desc<-data.frame(individual=factor(c(12,13,7,8,9,1,2,11)),
                  stage=factor(c("B","B","G","G","G","L","L","L")))
ase_nolowcounts_Hybonly<-ase_nolowcounts %>% select(Chr,start,end,HybB12,HybB13,HybG7,HybG8,HybG9,HybL1,HybL2, HybL11)
rownames(ase_nolowcounts_Hybonly)<-paste0(ase_nolowcounts_Hybonly$Chr,":",ase_nolowcounts_Hybonly$start, "-",ase_nolowcounts_Hybonly$end)
ase_nolowcounts_Hybonly<-ase_nolowcounts_Hybonly %>% select(-Chr,-start,-end)

ase_nolowcounts_Hybonlydds<-DESeqDataSetFromMatrix(ase_nolowcounts_Hybonly,
                                        ase_nolowcounts.CT_liftedHyb.BGL.desc,
                                       design = ~stage)
ase_nolowcounts_Hybonlydds<-DESeq(ase_nolowcounts_Hybonlydds)
ase_nolowcounts_Hybonly.BtoG<-results(ase_nolowcounts_Hybonlydds, contrast=c("stage","B","G"))
ase_nolowcounts_Hybonly.BtoG<-as.data.frame(ase_nolowcounts_Hybonly.BtoG)

ase_nolowcounts_Hybonly.GtoL<-results(ase_nolowcounts_Hybonlydds, contrast=c("stage","G","L"))
ase_nolowcounts_Hybonly.GtoL<-as.data.frame(ase_nolowcounts_Hybonly.GtoL)

#ocr's that change between stages--table for paper
ocrs.debetweenstages.parents<-data.frame(
  HtubBtoG=nrow(ase_nolowcounts_htubonly.BtoG %>% filter(padj<=.1)),
  HtubGtoL=nrow(ase_nolowcounts_htubonly.GtoL %>% filter(padj<=.1)),
  HeryBtoG=nrow(ase_nolowcounts_Heryonly.BtoG %>% filter(padj<=.1)),
  HeryGtoL=nrow(ase_nolowcounts_Heryonly.GtoL %>% filter(padj<=.1)),
  HybHTBtoG=nrow(ase_nolowcounts_HybHTonly.BtoG %>% filter(padj<=.1)),
  HybHTGtoL=nrow(ase_nolowcounts_HybHTonly.GtoL %>% filter(padj<=.1)),
  HybHEBtoG=nrow(ase_nolowcounts_HybHEonly.BtoG %>% filter(padj<=.1)),
  HybHEGtoL=nrow(ase_nolowcounts_HybHEonly.GtoL %>% filter(padj<=.1))
)

ocrs.debetweenstages.parents #Table S7
```

Proximal vs distal peaks analysis:
```{r}
proximalpeak_gene_map<-Hery_peak_gene_map_final %>% filter(abs(HEpeakGeneDist_5prime)<abs(HEpeakGeneDist_3prime)&(abs(HEpeakGeneDist_5prime)<=500)|abs(HEpeakGeneDist_3prime<abs(HEpeakGeneDist_5prime) & (abs(HEpeakGeneDist_3prime)<=500)))
proximalpeak_gene_map<-left_join(proximalpeak_gene_map,HEliftedpeakcoordsonly %>% select(-midpeak) %>% dplyr::rename(peak=peakno))
proximalpeak_gene_map$peaksize<-abs(proximalpeak_gene_map$start - proximalpeak_gene_map$end)
proximalpeak_gene_map$PeakCoord<-paste0(proximalpeak_gene_map$HEChr,":",proximalpeak_gene_map$start, "-",proximalpeak_gene_map$end)
proximalpeak_gene_map<-inner_join(proximalpeak_gene_map %>% select(PeakCoord,peaksize),ase_nolowcounts.CT.peaks.all) 
distalpeak_gene_map<-anti_join(ase_nolowcounts.CT.peaks.all,proximalpeak_gene_map)
distalpeak_gene_map<-left_join(distalpeak_gene_map,HEliftedpeakcoordsonly %>% select(-midpeak) %>% dplyr::rename(peak=peakno) %>% mutate(PeakCoord=paste0(HEliftedpeakcoordsonly$HEChr,":",HEliftedpeakcoordsonly$start, "-",HEliftedpeakcoordsonly$end)))
distalpeak_gene_map$peaksize<-abs(distalpeak_gene_map$start - distalpeak_gene_map$end)
distalpeak_gene_map<-left_join(distalpeak_gene_map %>% select(PeakCoord,peaksize), HEliftedpeakcoordsonly %>% mutate(PeakCoord=paste0(HEliftedpeakcoordsonly$HEChr,":",HEliftedpeakcoordsonly$start, "-",HEliftedpeakcoordsonly$end)) %>% dplyr::rename(peak=peakno))
distalpeak_gene_map<-left_join(distalpeak_gene_map,Hery_peak_gene_map_final)
distalpeak_gene_map<-distalpeak_gene_map %>% filter(abs(HEpeakGeneDist_5prime)<25000|abs(HEpeakGeneDist_3prime<25000))

#compare size of prox vs distal peaks:
summary(proximalpeak_gene_map$peaksize)
summary(distalpeak_gene_map$peaksize) #proximal are bigger than distal

#View(proximalpeak_gene_map %>% group_by(CT.B) %>% tally()) 
#View(proximalpeak_gene_map %>% group_by(CT.G) %>% tally()) 
#View(proximalpeak_gene_map %>% group_by(CT.L) %>% tally()) 
#at all stages trans is over twice as common as cis, which is not true of peaks overall

proximalpeak_tallies<-data.frame(CT.B=proximalpeak_gene_map %>% group_by(CT.B) %>% tally())
colnames(proximalpeak_tallies)<-c("Type","CT.B")
proximalpeak_tallies<-cbind(proximalpeak_tallies, proximalpeak_gene_map %>% group_by(CT.G) %>% tally() %>% select(n) %>% dplyr::rename(CT.G=n))
proximalpeak_tallies<-cbind(proximalpeak_tallies, proximalpeak_gene_map %>% group_by(CT.L) %>% tally() %>% select(n) %>% dplyr::rename(CT.L=n))

distalpeak_gene_map<-inner_join(distalpeak_gene_map %>% select(PeakCoord,peak),ase_nolowcounts.CT.peaks.all) 
distalpeak_tallies<-data.frame(CT.B=distalpeak_gene_map %>% group_by(CT.B) %>% tally())
colnames(distalpeak_tallies)<-c("Type","CT.B")
distalpeak_tallies<-cbind(distalpeak_tallies, distalpeak_gene_map %>% group_by(CT.G) %>% tally() %>% select(n) %>% dplyr::rename(CT.G=n))
distalpeak_tallies<-cbind(distalpeak_tallies, distalpeak_gene_map %>% group_by(CT.L) %>% tally() %>% select(n) %>% dplyr::rename(CT.L=n))

proximalpeak_tallies<-melt(proximalpeak_tallies, id="Type")
proximalpeak_tallies.plot<-ggplot(proximalpeak_tallies, aes(x=variable, y=value/sum(value),color=Type, group=Type))+
  geom_line(linewidth=2)+theme_classic()+theme(panel.background = element_rect(fill = "transparent",colour = NA), plot.background = element_rect(fill = "transparent",colour = NA), legend.background = element_rect(fill = "transparent",colour = NA))+ 
  xlab("") + ylab("Number of Peaks") + ggtitle("") + theme(plot.title = element_text(size=24, vjust = 1.5, face = "bold"), axis.text = element_text(size=20), axis.title=element_text(size=22),legend.text = element_text(size=19), legend.title=element_text(size=19), legend.key = element_rect(linewidth = 5, colour = NA), legend.key.size = unit(1.5, 'lines'))+
  ylim(0,.2)
#ggsave("proximalpeak_tallies.plot.pdf", proximalpeak_tallies.plot, width=8,height=6, units="in") #Figure 6A, proximal peaks plot

distalpeak_tallies<-melt(distalpeak_tallies, id="Type")
distalpeak_tallies.plot<-ggplot(distalpeak_tallies, aes(x=variable, y=value/sum(value),color=Type, group=Type))+
  geom_line(size=2)+theme_classic()+theme(panel.background = element_rect(fill = "transparent",colour = NA), plot.background = element_rect(fill = "transparent",colour = NA), legend.background = element_rect(fill = "transparent",colour = NA))+ 
  xlab("") + ylab("Number of Peaks") + ggtitle("") + theme(plot.title = element_text(size=24, vjust = 1.5, face = "bold"), axis.text = element_text(size=20), axis.title=element_text(size=22),legend.text = element_text(size=19), legend.title=element_text(size=19), legend.key = element_rect(size = 5, colour = NA), legend.key.size = unit(1.5, 'lines'))+
    ylim(0,.2)

#ggsave("distalpeak_tallies.plot.pdf", distalpeak_tallies.plot, width=8,height=6, units="in") #Figure 6A, distal peaks plot

#testing if trans is more likely in prox than in all peaks (Table S9): 
#blastula:
prop.test(x = c(1128, 11616-1128), n = c(sum(134+1128+764+22+24+1375+27),sum(ase_nolowcounts.CT.test.res$B)-sum(134+1128+764+22+24+1375+27))) #signif
#gastrula:
prop.test(x = c(1286,15235-1286), n = c(sum(230+1286+677+78+25+1130+48),sum(ase_nolowcounts.CT.test.res$G)-sum(230+1286+677+78+25+1130+48))) #signif
#larva:
prop.test(x = c(1358,13426-1358), n = c(sum(247+1358+590+24+32+1159+64),sum(ase_nolowcounts.CT.test.res$L)-sum(247+1358+590+24+32+1159+64))) #signif

#testing if trans is more likely in distal than in all peaks:
#blastula:
prop.test(x = c(8475, 11616-8475), n = c(sum(8537+8475+20974+1084+836+48038+616),sum(ase_nolowcounts.CT.test.res$B)-sum(8537+8475+20974+1084+836+48038+616))) #signif for LESS likely to be trans
11616/sum(ase_nolowcounts.CT.test.res$B)

#gastrula:
prop.test(x = c(11110,15235-11110), n = c(sum(13246+11110+19865+3016+1090+38468+1765),sum(ase_nolowcounts.CT.test.res$G)-sum(13246+11110+19865+3016+1090+38468+1765))) #signif for LESS likely to be trans
#larva:
prop.test(x = c(9691,13426-9691), n = c(sum(14513+9691+18263+1275+1855+40977+1986),sum(ase_nolowcounts.CT.test.res$L)-sum(14513+9691+18263+1275+1855+40977+1986))) #signif for LESS Likely to be trans


#now doing same tests for proximal trans vs distal trans:
#blastula:
prop.test(x = c(1128, 8475), n = c(sum(134+1128+764+22+24+1375+27),sum(8537+8475+20974+1084+836+48038+616))) #signif
#gastrula:
prop.test(x = c(1286,11110), n = c(sum(230+1286+677+78+25+1130+48),sum(13246+11110+19865+3016+1090+38468+1765))) #signif
#larva:
prop.test(x = c(1358,9691), n = c(sum(247+1358+590+24+32+1159+64),sum(14513+9691+18263+1275+1855+40977+1986))) #signif


#are proximal or distal more likely to be DA than peaks as a whole? (Table S8)
#these numbers came from proximalpeak_tallies, distalpeak_tallies, and ase_nolowcounts.CT.test.res
#proximal vs all not-proximal peaks:
#blastula
prop.test(x = c(1335,25454-1335), n = c(3474, 117391-3474))
#gastrula
prop.test(x = c(1667,39263-1667), n = c(3474, 117391-3474))
#larva
prop.test(x = c(1725,38032-1725), n = c(3474, 117391-3474))
#signif at all 3 stages for enriched in DA

#distal vs all not-distal peaks
#blastula
prop.test(x = c(19548,25454-19548), n = c(88560, 117391-88560))
#gastrula
prop.test(x = c(30227,39263-30227), n = c(88560, 117391-88560))
#larva
prop.test(x = c(29320,38032-29320), n = c(88560, 117391-88560))
#signif at all 3 stages for enriched in DA

```
```{r}
#now looking at effect size at each stage:
B.proximal.meanexprchange<-data.frame(abslog2FC=ase_nolowcounts.CT.B.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.B.all)) %>%  filter(PeakCoord %in% proximalpeak_gene_map$PeakCoord) %>% select(log2FC_TT_HEvsHT) %>% abs(),
                                        ProximalorDistal=rep("Proximal",length(proximalpeak_gene_map$PeakCoord)))
B.distal.meanexprchange<-data.frame(abslog2FC=ase_nolowcounts.CT.B.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.B.all)) %>%  filter(PeakCoord %in% distalpeak_gene_map$PeakCoord) %>% select(log2FC_TT_HEvsHT) %>% abs(),
                                        ProximalorDistal=rep("distal",length(distalpeak_gene_map$PeakCoord)))
B.proximaldistal.meanexprchange<-rbind(B.proximal.meanexprchange,B.distal.meanexprchange)
B.proximaldistal.meanexprchange.plot<-ggplot(B.proximaldistal.meanexprchange, aes(x=ProximalorDistal, y=log2FC_TT_HEvsHT,colour=ProximalorDistal))+
  scale_color_manual(values=c("red","blue"),name = "Proximal or Distal")+
  geom_boxplot(width=0.1,color="black")+
  geom_violin(fill=NA)+
  theme_bw()+
  theme(legend.position='none')+
  theme(axis.title.x=element_blank(),
        axis.ticks.x=element_blank(),
        text = element_text(size=20))+
  background_grid(major="none",minor="none")+
  ylim(0,15)

G.proximal.meanexprchange<-data.frame(abslog2FC=ase_nolowcounts.CT.G.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.G.all)) %>%  filter(PeakCoord %in% proximalpeak_gene_map$PeakCoord) %>% select(log2FC_TT_HEvsHT) %>% abs(),
                                        ProximalorDistal=rep("Proximal",length(proximalpeak_gene_map$PeakCoord)))
G.distal.meanexprchange<-data.frame(abslog2FC=ase_nolowcounts.CT.G.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.G.all)) %>%  filter(PeakCoord %in% distalpeak_gene_map$PeakCoord) %>% select(log2FC_TT_HEvsHT) %>% abs(),
                                        ProximalorDistal=rep("distal",length(distalpeak_gene_map$PeakCoord)))
G.proximaldistal.meanexprchange<-rbind(G.proximal.meanexprchange,G.distal.meanexprchange)
View(G.proximaldistal.meanexprchange)
G.proximaldistal.meanexprchange.plot<-ggplot(G.proximaldistal.meanexprchange, aes(x=ProximalorDistal, y=log2FC_TT_HEvsHT,colour=ProximalorDistal))+
  scale_color_manual(values=c("red","blue"),name = "Regulatory Mode")+
  geom_boxplot(width=0.1,color="black")+
  geom_violin(fill=NA)+
  theme_bw()+
  theme(legend.position='none')+
  theme(axis.title.x=element_blank(),
        axis.ticks.x=element_blank(),
        text = element_text(size=20))+
  background_grid(major="none",minor="none")+
  ylim(0,15)

L.proximal.meanexprchange<-data.frame(abslog2FC=ase_nolowcounts.CT.L.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.L.all)) %>%  filter(PeakCoord %in% proximalpeak_gene_map$PeakCoord) %>% select(log2FC_TT_HEvsHT) %>% abs(),
                                        ProximalorDistal=rep("Proximal",length(proximalpeak_gene_map$PeakCoord)))
L.distal.meanexprchange<-data.frame(abslog2FC=ase_nolowcounts.CT.L.all %>% mutate(PeakCoord=row.names(ase_nolowcounts.CT.L.all)) %>%  filter(PeakCoord %in% distalpeak_gene_map$PeakCoord) %>% select(log2FC_TT_HEvsHT) %>% abs(),
                                        ProximalorDistal=rep("distal",length(distalpeak_gene_map$PeakCoord)))
L.proximaldistal.meanexprchange<-rbind(L.proximal.meanexprchange,L.distal.meanexprchange)
L.proximaldistal.meanexprchange.plot<-ggplot(L.proximaldistal.meanexprchange, aes(x=ProximalorDistal, y=log2FC_TT_HEvsHT,colour=ProximalorDistal))+
  scale_color_manual(values=c("red","blue"),name = "Regulatory Mode")+
  geom_boxplot(width=0.1,color="black")+
  geom_violin(fill=NA)+
  theme_bw()+
  theme(legend.position='none')+
  theme(axis.title.x=element_blank(),
        axis.ticks.x=element_blank(),
        text = element_text(size=20))+
  background_grid(major="none",minor="none")+
  ylim(0,15)

#because the variances of proximal vs distal are different, i will use a welch's test of unequal variance to see if the means are statsitcally different:
t.test(B.proximal.meanexprchange$log2FC_TT_HEvsHT,B.distal.meanexprchange$log2FC_TT_HEvsHT) #stat diff
t.test(G.proximal.meanexprchange$log2FC_TT_HEvsHT,G.distal.meanexprchange$log2FC_TT_HEvsHT) #stat diff
t.test(L.proximal.meanexprchange$log2FC_TT_HEvsHT,L.distal.meanexprchange$log2FC_TT_HEvsHT) #stat diff
#^^ for Figure 6B


#ggsave("proximaldistal.meanexprchange.BGL.pdf",(plot_grid(B.proximaldistal.meanexprchange.plot,G.proximaldistal.meanexprchange.plot,L.proximaldistal.meanexprchange.plot, nrow=1)),width=10, height=6) #Figure 6B

#make list of peaks for HOMER motif enrichment of proximal vs distal: (Table S10)
##makehomerbedD(proximalpeak_gene_map %>% select(PeakCoord, peak))
##makehomerbedD(distalpeak_gene_map %>% select(PeakCoord, peak))
```






Enriched heat maps on proximal peaks (Figure S1):
```{r}
library(EnrichedHeatmap)
library(data.table)
library(rtracklayer)
#the package requires bed files and bedgraph files so I'll make a bed of proximal peaks:
proximalpeaks<-proximalpeak_gene_map %>% select(PeakCoord, peak)
#makehomerbedD(proximalpeaks)

#and i need a bedgraph which is bed + counts. let's do blastula, Hery to start:
proximalpeaks<-transform(proximalpeaks, col=do.call(rbind, strsplit(PeakCoord, ':', fixed=TRUE)), stringsAsFactors=F) %>% select (-PeakCoord)
proximalpeaks<-transform(proximalpeaks, col=do.call(rbind, strsplit(col.2, '-', fixed=TRUE)), stringsAsFactors=F) %>% select(-col.2,-peak)
colnames(proximalpeaks)<-c("Chr","start","end")
proximalpeaks$start<-as.numeric(proximalpeaks$start)
proximalpeaks$end<-as.numeric(proximalpeaks$end)

proximalpeaks.Hery.aveB<-left_join(proximalpeaks, ase_nolowcounts %>% select(Chr,start,end,HeryB4,HeryB7))
proximalpeaks.Hery.aveB$HeryB4<-cpm(proximalpeaks.Hery.aveB$HeryB4) #need to use normalized counts to be able to combine multiple samples
proximalpeaks.Hery.aveB$HeryB7<-cpm(proximalpeaks.Hery.aveB$HeryB7) 
proximalpeaks.Hery.aveB$aveB<-(proximalpeaks.Hery.aveB$HeryB4+proximalpeaks.Hery.aveB$HeryB7)/2
proximalpeaks.Hery.aveB<-proximalpeaks.Hery.aveB %>% select(-HeryB4,-HeryB7)
proximalpeaks.Hery.aveB<-proximalpeaks.Hery.aveB %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
proximalpeaks.Hery.aveB<-proximalpeaks.Hery.aveB %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveB),0L))
proximalpeaks.Hery.aveB$overlap<-proximalpeaks.Hery.aveB$start-proximalpeaks.Hery.aveB$prevend
proximalpeaks.Hery.aveB$overlap<-ifelse(lead(proximalpeaks.Hery.aveB$overlap<0),"erase me",proximalpeaks.Hery.aveB$overlap)
proximalpeaks.Hery.aveB$start<-ifelse(proximalpeaks.Hery.aveB$overlap<0,proximalpeaks.Hery.aveB$prevstart,proximalpeaks.Hery.aveB$start)
proximalpeaks.Hery.aveB$aveB<-ifelse(proximalpeaks.Hery.aveB$overlap<0,proximalpeaks.Hery.aveB$aveB+proximalpeaks.Hery.aveB$prevpeak,proximalpeaks.Hery.aveB$aveB)
proximalpeaks.Hery.aveB<-proximalpeaks.Hery.aveB %>% filter(!overlap=="erase me")
proximalpeaks.Hery.aveB<-proximalpeaks.Hery.aveB %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(proximalpeaks.Hery.aveB,"proximalpeaks.Hery.aveB.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig

#back to enrichedheatmap:
targets <- makeGRangesFromDataFrame(
  df = fread("proximalpeaks.gff", header = FALSE, data.table = FALSE),
  seqnames.field = "V1", start.field = "V2", end.field = "V3")

#/ We take the center of each region/peak and want to extend by 5kb each direction:
ExtendSize <- 5000
targets.extended  <- resize(targets, fix = "center", width = ExtendSize*2)

#/ We load the relevant parts of the bigwig file into R. 
#/ This is more efficient than reading the entire file. 
Hery.B.BigWig <- rtracklayer::import("proximalpeaks.Hery.aveB.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
#/ Create the normalizedMatrix that EnrichedHeatmap accepts as input.
#/ We use the targets center (width=1) because from what I understand normalizeMatrix
#/ does not allow to turn off its "extend" option. Therefore we trick it by simply
#/ providing the peak centers and then let the function extend it by our predefined window size.
Hery.B.normMatrix <- normalizeToMatrix(signal = Hery.B.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      #/ minimal value to the 99th percentile
                                target_ratio = 0,
                                mean_mode = "w0",       #/ see ?EnrichedHeatmap on other options
                                value_column = "score", #/ = the name of the 4th column of the bigwig
                                extend = ExtendSize)

#/ Make a color gradient that covers the range of normMatrix from 0 to the 99th percentile.
#/ The percentile avoids outliers to skew the heatmap:
#col_fun = circlize::colorRamp2(quantile(Hery.B.normMatrix, c(0, .99)), c("darkblue", "darkgoldenrod1"))

#/ heatmap function:
Hery.B.EH <- EnrichedHeatmap( mat = Hery.B.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from above
                       column_title = "Proximal Peaks: Hery x Hery Blastula", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                           ylim=c(0,300))
                       )
) #/ end of EnrichedHeatmap function


#now repeat for other stages of Hery
#gastrula:
proximalpeaks.Hery.aveG<-left_join(proximalpeaks, ase_nolowcounts %>% select(Chr,start,end,HeryG2,HeryG5,HeryG8))
proximalpeaks.Hery.aveG$HeryG2<-cpm(proximalpeaks.Hery.aveG$HeryG2)
proximalpeaks.Hery.aveG$HeryG5<-cpm(proximalpeaks.Hery.aveG$HeryG5) 
proximalpeaks.Hery.aveG$HeryG8<-cpm(proximalpeaks.Hery.aveG$HeryG8) 
proximalpeaks.Hery.aveG$aveG<-(proximalpeaks.Hery.aveG$HeryG2+proximalpeaks.Hery.aveG$HeryG5+proximalpeaks.Hery.aveG$HeryG8)/3
proximalpeaks.Hery.aveG<-proximalpeaks.Hery.aveG %>% select(-HeryG2,-HeryG5,-HeryG8)
proximalpeaks.Hery.aveG<-proximalpeaks.Hery.aveG %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
proximalpeaks.Hery.aveG<-proximalpeaks.Hery.aveG %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveG),0L))
proximalpeaks.Hery.aveG$overlap<-proximalpeaks.Hery.aveG$start-proximalpeaks.Hery.aveG$prevend
proximalpeaks.Hery.aveG$overlap<-ifelse(lead(proximalpeaks.Hery.aveG$overlap<0),"erase me",proximalpeaks.Hery.aveG$overlap)
proximalpeaks.Hery.aveG$start<-ifelse(proximalpeaks.Hery.aveG$overlap<0,proximalpeaks.Hery.aveG$prevstart,proximalpeaks.Hery.aveG$start)
proximalpeaks.Hery.aveG$aveG<-ifelse(proximalpeaks.Hery.aveG$overlap<0,proximalpeaks.Hery.aveG$aveG+proximalpeaks.Hery.aveG$prevpeak,proximalpeaks.Hery.aveG$aveG)
proximalpeaks.Hery.aveG<-proximalpeaks.Hery.aveG %>% filter(!overlap=="erase me")
proximalpeaks.Hery.aveG<-proximalpeaks.Hery.aveG %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(proximalpeaks.Hery.aveG,"proximalpeaks.Hery.aveG.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#skip making target, it is the same as before
Hery.G.BigWig <- rtracklayer::import("proximalpeaks.Hery.aveG.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
Hery.G.normMatrix <- normalizeToMatrix(signal = Hery.G.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      
                                target_ratio = 0,
                                mean_mode = "w0",       
                                value_column = "score", 
                                extend = ExtendSize)
Hery.G.EH <- EnrichedHeatmap( mat = Hery.G.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from blastula
                       column_title = "Proximal Peaks: Hery x Hery Gastrula", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                           ylim=c(0,300))
                       )
) 

#larva:
proximalpeaks.Hery.aveL<-left_join(proximalpeaks, ase_nolowcounts %>% select(Chr,start,end,HeryL3,HeryL6,HeryL9))
proximalpeaks.Hery.aveL$HeryL3<-cpm(proximalpeaks.Hery.aveL$HeryL3)
proximalpeaks.Hery.aveL$HeryL6<-cpm(proximalpeaks.Hery.aveL$HeryL6) 
proximalpeaks.Hery.aveL$HeryL9<-cpm(proximalpeaks.Hery.aveL$HeryL9) 
proximalpeaks.Hery.aveL$aveL<-(proximalpeaks.Hery.aveL$HeryL3+proximalpeaks.Hery.aveL$HeryL6+proximalpeaks.Hery.aveL$HeryL9)/3
proximalpeaks.Hery.aveL<-proximalpeaks.Hery.aveL %>% select(-HeryL3,-HeryL6,-HeryL9)
proximalpeaks.Hery.aveL<-proximalpeaks.Hery.aveL %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
proximalpeaks.Hery.aveL<-proximalpeaks.Hery.aveL %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveL),0L))
proximalpeaks.Hery.aveL$overlap<-proximalpeaks.Hery.aveL$start-proximalpeaks.Hery.aveL$prevend
proximalpeaks.Hery.aveL$overlap<-ifelse(lead(proximalpeaks.Hery.aveL$overlap<0),"erase me",proximalpeaks.Hery.aveL$overlap)
proximalpeaks.Hery.aveL$start<-ifelse(proximalpeaks.Hery.aveL$overlap<0,proximalpeaks.Hery.aveL$prevstart,proximalpeaks.Hery.aveL$start)
proximalpeaks.Hery.aveL$aveL<-ifelse(proximalpeaks.Hery.aveL$overlap<0,proximalpeaks.Hery.aveL$aveL+proximalpeaks.Hery.aveL$prevpeak,proximalpeaks.Hery.aveL$aveL)
proximalpeaks.Hery.aveL<-proximalpeaks.Hery.aveL %>% filter(!overlap=="erase me")
proximalpeaks.Hery.aveL<-proximalpeaks.Hery.aveL %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(proximalpeaks.Hery.aveL,"proximalpeaks.Hery.aveL.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#skip making target, it is the same as before
Hery.L.BigWig <- rtracklayer::import("proximalpeaks.Hery.aveL.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
Hery.L.normMatrix <- normalizeToMatrix(signal = Hery.L.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      
                                target_ratio = 0,
                                mean_mode = "w0",       
                                value_column = "score", 
                                extend = ExtendSize)
Hery.L.EH <- EnrichedHeatmap( mat = Hery.L.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from blastula
                       column_title = "Proximal Peaks: Hery x Hery Larva", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                           ylim=c(0,300))
                       )
) 

#now for HTUB:
proximalpeaks.htub.aveB<-left_join(proximalpeaks, ase_nolowcounts %>% select(Chr,start,end,htubB4,htubB7))
proximalpeaks.htub.aveB$htubB4<-cpm(proximalpeaks.htub.aveB$htubB4) 
proximalpeaks.htub.aveB$htubB7<-cpm(proximalpeaks.htub.aveB$htubB7) 
proximalpeaks.htub.aveB$aveB<-(proximalpeaks.htub.aveB$htubB4+proximalpeaks.htub.aveB$htubB7)/2
proximalpeaks.htub.aveB<-proximalpeaks.htub.aveB %>% select(-htubB4,-htubB7)
proximalpeaks.htub.aveB<-proximalpeaks.htub.aveB %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
proximalpeaks.htub.aveB<-proximalpeaks.htub.aveB %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveB),0L))
proximalpeaks.htub.aveB$overlap<-proximalpeaks.htub.aveB$start-proximalpeaks.htub.aveB$prevend
proximalpeaks.htub.aveB$overlap<-ifelse(lead(proximalpeaks.htub.aveB$overlap<0),"erase me",proximalpeaks.htub.aveB$overlap)
proximalpeaks.htub.aveB$start<-ifelse(proximalpeaks.htub.aveB$overlap<0,proximalpeaks.htub.aveB$prevstart,proximalpeaks.htub.aveB$start)
proximalpeaks.htub.aveB$aveB<-ifelse(proximalpeaks.htub.aveB$overlap<0,proximalpeaks.htub.aveB$aveB+proximalpeaks.htub.aveB$prevpeak,proximalpeaks.htub.aveB$aveB)
proximalpeaks.htub.aveB<-proximalpeaks.htub.aveB %>% filter(!overlap=="erase me")
proximalpeaks.htub.aveB<-proximalpeaks.htub.aveB %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(proximalpeaks.htub.aveB,"proximalpeaks.htub.aveB.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#/ We load the relevant parts of the bigwig file into R. 
#/ This is more efficient than reading the entire file. 
htub.B.BigWig <- rtracklayer::import("proximalpeaks.htub.aveB.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
#/ Create the normalizedMatrix that EnrichedHeatmap accepts as input.
#/ We use the targets center (width=1) because from what I understand normalizeMatrix
#/ does not allow to turn off its "extend" option. Therefore we trick it by simply
#/ providing the peak centers and then let the function extend it by our predefined window size.
htub.B.normMatrix <- normalizeToMatrix(signal = htub.B.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      #/ minimal value to the 99th percentile
                                target_ratio = 0,
                                mean_mode = "w0",       #/ see ?EnrichedHeatmap on other options
                                value_column = "score", #/ = the name of the 4th column of the bigwig
                                extend = ExtendSize)
#/ heatmap function:
htub.B.EH <- EnrichedHeatmap( mat = htub.B.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from hery blastula
                       column_title = "Proximal Peaks: htub x htub Blastula", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                           ylim=c(0,300))
                       )
) #/ end of EnrichedHeatmap function

#now repeat for other stages of htub
#gastrula:
proximalpeaks.htub.aveG<-left_join(proximalpeaks, ase_nolowcounts %>% select(Chr,start,end,htubG2,htubG5,htubG8))
proximalpeaks.htub.aveG$htubG2<-cpm(proximalpeaks.htub.aveG$htubG2)
proximalpeaks.htub.aveG$htubG5<-cpm(proximalpeaks.htub.aveG$htubG5) 
proximalpeaks.htub.aveG$htubG8<-cpm(proximalpeaks.htub.aveG$htubG8) 
proximalpeaks.htub.aveG$aveG<-(proximalpeaks.htub.aveG$htubG2+proximalpeaks.htub.aveG$htubG5+proximalpeaks.htub.aveG$htubG8)/3
proximalpeaks.htub.aveG<-proximalpeaks.htub.aveG %>% select(-htubG2,-htubG5,-htubG8)
proximalpeaks.htub.aveG<-proximalpeaks.htub.aveG %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
proximalpeaks.htub.aveG<-proximalpeaks.htub.aveG %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveG),0L))
proximalpeaks.htub.aveG$overlap<-proximalpeaks.htub.aveG$start-proximalpeaks.htub.aveG$prevend
proximalpeaks.htub.aveG$overlap<-ifelse(lead(proximalpeaks.htub.aveG$overlap<0),"erase me",proximalpeaks.htub.aveG$overlap)
proximalpeaks.htub.aveG$start<-ifelse(proximalpeaks.htub.aveG$overlap<0,proximalpeaks.htub.aveG$prevstart,proximalpeaks.htub.aveG$start)
proximalpeaks.htub.aveG$aveG<-ifelse(proximalpeaks.htub.aveG$overlap<0,proximalpeaks.htub.aveG$aveG+proximalpeaks.htub.aveG$prevpeak,proximalpeaks.htub.aveG$aveG)
proximalpeaks.htub.aveG<-proximalpeaks.htub.aveG %>% filter(!overlap=="erase me")
proximalpeaks.htub.aveG<-proximalpeaks.htub.aveG %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(proximalpeaks.htub.aveG,"proximalpeaks.htub.aveG.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#skip making target, it is the same as before
htub.G.BigWig <- rtracklayer::import("proximalpeaks.htub.aveG.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
htub.G.normMatrix <- normalizeToMatrix(signal = htub.G.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      
                                target_ratio = 0,
                                mean_mode = "w0",       
                                value_column = "score", 
                                extend = ExtendSize)
htub.G.EH <- EnrichedHeatmap( mat = htub.G.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from above
                       column_title = "Proximal Peaks: htub x htub Gastrula", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                           ylim=c(0,300))
                       )
) 

#larva:
proximalpeaks.htub.aveL<-left_join(proximalpeaks, ase_nolowcounts %>% select(Chr,start,end,htubL3,htubL6,htubL11))
proximalpeaks.htub.aveL$htubL3<-cpm(proximalpeaks.htub.aveL$htubL3)
proximalpeaks.htub.aveL$htubL6<-cpm(proximalpeaks.htub.aveL$htubL6) 
proximalpeaks.htub.aveL$htubL11<-cpm(proximalpeaks.htub.aveL$htubL11) 
proximalpeaks.htub.aveL$aveL<-(proximalpeaks.htub.aveL$htubL3+proximalpeaks.htub.aveL$htubL6+proximalpeaks.htub.aveL$htubL11)/3
proximalpeaks.htub.aveL<-proximalpeaks.htub.aveL %>% select(-htubL3,-htubL6,-htubL11)
proximalpeaks.htub.aveL<-proximalpeaks.htub.aveL %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
proximalpeaks.htub.aveL<-proximalpeaks.htub.aveL %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveL),0L))
proximalpeaks.htub.aveL$overlap<-proximalpeaks.htub.aveL$start-proximalpeaks.htub.aveL$prevend
proximalpeaks.htub.aveL$overlap<-ifelse(lead(proximalpeaks.htub.aveL$overlap<0),"erase me",proximalpeaks.htub.aveL$overlap)
proximalpeaks.htub.aveL$start<-ifelse(proximalpeaks.htub.aveL$overlap<0,proximalpeaks.htub.aveL$prevstart,proximalpeaks.htub.aveL$start)
proximalpeaks.htub.aveL$aveL<-ifelse(proximalpeaks.htub.aveL$overlap<0,proximalpeaks.htub.aveL$aveL+proximalpeaks.htub.aveL$prevpeak,proximalpeaks.htub.aveL$aveL)
proximalpeaks.htub.aveL<-proximalpeaks.htub.aveL %>% filter(!overlap=="erase me")
proximalpeaks.htub.aveL<-proximalpeaks.htub.aveL %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(proximalpeaks.htub.aveL,"proximalpeaks.htub.aveL.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#skip making target, it is the same as before
htub.L.BigWig <- rtracklayer::import("proximalpeaks.htub.aveL.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
htub.L.normMatrix <- normalizeToMatrix(signal = htub.L.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      
                                target_ratio = 0,
                                mean_mode = "w0",       
                                value_column = "score", 
                                extend = ExtendSize)
htub.L.EH <- EnrichedHeatmap( mat = htub.L.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from above
                       column_title = "Proximal Peaks: Htub x Htub Larva", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                         ylim=c(0,300))
                       )
) 

pdf("Hery.B.EH.proximal.pdf")
draw(Hery.B.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

pdf("Hery.G.EH.proximal.pdf")
draw(Hery.G.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

pdf("Hery.L.EH.proximal.pdf")
draw(Hery.L.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

pdf("htub.B.EH.proximal.pdf")
draw(htub.B.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

pdf("htub.G.EH.proximal.pdf")
draw(htub.G.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

pdf("htub.L.EH.proximal.pdf")
draw(htub.L.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

proximalpeaks.HybHE.aveB<-left_join(proximalpeaks, ase_nolowcounts %>% select(Chr,start,end,HybHEB12,HybHEB13))
proximalpeaks.HybHE.aveB$HybHEB12<-cpm(proximalpeaks.HybHE.aveB$HybHEB12) 
proximalpeaks.HybHE.aveB$HybHEB13<-cpm(proximalpeaks.HybHE.aveB$HybHEB13) 
proximalpeaks.HybHE.aveB$aveB<-(proximalpeaks.HybHE.aveB$HybHEB12+proximalpeaks.HybHE.aveB$HybHEB13)/2
proximalpeaks.HybHE.aveB<-proximalpeaks.HybHE.aveB %>% select(-HybHEB12,-HybHEB13)
proximalpeaks.HybHE.aveB<-proximalpeaks.HybHE.aveB %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
proximalpeaks.HybHE.aveB<-proximalpeaks.HybHE.aveB %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveB),0L))
proximalpeaks.HybHE.aveB$overlap<-proximalpeaks.HybHE.aveB$start-proximalpeaks.HybHE.aveB$prevend
proximalpeaks.HybHE.aveB$overlap<-ifelse(lead(proximalpeaks.HybHE.aveB$overlap<0),"erase me",proximalpeaks.HybHE.aveB$overlap)
proximalpeaks.HybHE.aveB$start<-ifelse(proximalpeaks.HybHE.aveB$overlap<0,proximalpeaks.HybHE.aveB$prevstart,proximalpeaks.HybHE.aveB$start)
proximalpeaks.HybHE.aveB$aveB<-ifelse(proximalpeaks.HybHE.aveB$overlap<0,proximalpeaks.HybHE.aveB$aveB+proximalpeaks.HybHE.aveB$prevpeak,proximalpeaks.HybHE.aveB$aveB)
proximalpeaks.HybHE.aveB<-proximalpeaks.HybHE.aveB %>% filter(!overlap=="erase me")
proximalpeaks.HybHE.aveB<-proximalpeaks.HybHE.aveB %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(proximalpeaks.HybHE.aveB,"proximalpeaks.HybHE.aveB.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#/ We load the relevant parts of the bigwig file into R. 
#/ This is more efficient than reading the entire file. 
HybHE.B.BigWig <- rtracklayer::import("proximalpeaks.HybHE.aveB.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
#/ Create the normalizedMatrix that EnrichedHeatmap accepts as input.
#/ We use the targets center (width=1) because from what I understand normalizeMatrix
#/ does not allow to turn off its "extend" option. Therefore we trick it by simply
#/ providing the peak centers and then let the function extend it by our predefined window size.
HybHE.B.normMatrix <- normalizeToMatrix(signal = HybHE.B.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      #/ minimal value to the 99th percentile
                                target_ratio = 0,
                                mean_mode = "w0",       #/ see ?EnrichedHeatmap on other options
                                value_column = "score", #/ = the name of the 4th column of the bigwig
                                extend = ExtendSize)
#/ heatmap function:
HybHE.B.EH <- EnrichedHeatmap( mat = HybHE.B.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from hery blastula
                       column_title = "Proximal Peaks: HybHE x HybHE Blastula", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                           ylim=c(0,300))
                       )
) #/ end of EnrichedHeatmap function

#now repeat for other stages of HybHE
#gastrula:
proximalpeaks.HybHE.aveG<-left_join(proximalpeaks, ase_nolowcounts %>% select(Chr,start,end,HybHEG7,HybHEG8,HybHEG9))
proximalpeaks.HybHE.aveG$HybHEG7<-cpm(proximalpeaks.HybHE.aveG$HybHEG7)
proximalpeaks.HybHE.aveG$HybHEG8<-cpm(proximalpeaks.HybHE.aveG$HybHEG8) 
proximalpeaks.HybHE.aveG$HybHEG9<-cpm(proximalpeaks.HybHE.aveG$HybHEG9) 
proximalpeaks.HybHE.aveG$aveG<-(proximalpeaks.HybHE.aveG$HybHEG7+proximalpeaks.HybHE.aveG$HybHEG8+proximalpeaks.HybHE.aveG$HybHEG9)/3
proximalpeaks.HybHE.aveG<-proximalpeaks.HybHE.aveG %>% select(-HybHEG7,-HybHEG8,-HybHEG9)
proximalpeaks.HybHE.aveG<-proximalpeaks.HybHE.aveG %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
proximalpeaks.HybHE.aveG<-proximalpeaks.HybHE.aveG %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveG),0L))
proximalpeaks.HybHE.aveG$overlap<-proximalpeaks.HybHE.aveG$start-proximalpeaks.HybHE.aveG$prevend
proximalpeaks.HybHE.aveG$overlap<-ifelse(lead(proximalpeaks.HybHE.aveG$overlap<0),"erase me",proximalpeaks.HybHE.aveG$overlap)
proximalpeaks.HybHE.aveG$start<-ifelse(proximalpeaks.HybHE.aveG$overlap<0,proximalpeaks.HybHE.aveG$prevstart,proximalpeaks.HybHE.aveG$start)
proximalpeaks.HybHE.aveG$aveG<-ifelse(proximalpeaks.HybHE.aveG$overlap<0,proximalpeaks.HybHE.aveG$aveG+proximalpeaks.HybHE.aveG$prevpeak,proximalpeaks.HybHE.aveG$aveG)
proximalpeaks.HybHE.aveG<-proximalpeaks.HybHE.aveG %>% filter(!overlap=="erase me")
proximalpeaks.HybHE.aveG<-proximalpeaks.HybHE.aveG %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(proximalpeaks.HybHE.aveG,"proximalpeaks.HybHE.aveG.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#skip making target, it is the same as before
HybHE.G.BigWig <- rtracklayer::import("proximalpeaks.HybHE.aveG.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
HybHE.G.normMatrix <- normalizeToMatrix(signal = HybHE.G.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      
                                target_ratio = 0,
                                mean_mode = "w0",       
                                value_column = "score", 
                                extend = ExtendSize)
HybHE.G.EH <- EnrichedHeatmap( mat = HybHE.G.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from above
                       column_title = "Proximal Peaks: HybHE x HybHE Gastrula", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                           ylim=c(0,300))
                       )
) 

#larva:
proximalpeaks.HybHE.aveL<-left_join(proximalpeaks, ase_nolowcounts %>% select(Chr,start,end,HybHEL1,HybHEL2,HybHEL11))
proximalpeaks.HybHE.aveL$HybHEL1<-cpm(proximalpeaks.HybHE.aveL$HybHEL1)
proximalpeaks.HybHE.aveL$HybHEL2<-cpm(proximalpeaks.HybHE.aveL$HybHEL2) 
proximalpeaks.HybHE.aveL$HybHEL11<-cpm(proximalpeaks.HybHE.aveL$HybHEL11) 
proximalpeaks.HybHE.aveL$aveL<-(proximalpeaks.HybHE.aveL$HybHEL1+proximalpeaks.HybHE.aveL$HybHEL2+proximalpeaks.HybHE.aveL$HybHEL11)/3
proximalpeaks.HybHE.aveL<-proximalpeaks.HybHE.aveL %>% select(-HybHEL1,-HybHEL2,-HybHEL11)
proximalpeaks.HybHE.aveL<-proximalpeaks.HybHE.aveL %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
proximalpeaks.HybHE.aveL<-proximalpeaks.HybHE.aveL %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveL),0L))
proximalpeaks.HybHE.aveL$overlap<-proximalpeaks.HybHE.aveL$start-proximalpeaks.HybHE.aveL$prevend
proximalpeaks.HybHE.aveL$overlap<-ifelse(lead(proximalpeaks.HybHE.aveL$overlap<0),"erase me",proximalpeaks.HybHE.aveL$overlap)
proximalpeaks.HybHE.aveL$start<-ifelse(proximalpeaks.HybHE.aveL$overlap<0,proximalpeaks.HybHE.aveL$prevstart,proximalpeaks.HybHE.aveL$start)
proximalpeaks.HybHE.aveL$aveL<-ifelse(proximalpeaks.HybHE.aveL$overlap<0,proximalpeaks.HybHE.aveL$aveL+proximalpeaks.HybHE.aveL$prevpeak,proximalpeaks.HybHE.aveL$aveL)
proximalpeaks.HybHE.aveL<-proximalpeaks.HybHE.aveL %>% filter(!overlap=="erase me")
proximalpeaks.HybHE.aveL<-proximalpeaks.HybHE.aveL %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(proximalpeaks.HybHE.aveL,"proximalpeaks.HybHE.aveL.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#skip making target, it is the same as before
HybHE.L.BigWig <- rtracklayer::import("proximalpeaks.HybHE.aveL.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
HybHE.L.normMatrix <- normalizeToMatrix(signal = HybHE.L.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      
                                target_ratio = 0,
                                mean_mode = "w0",       
                                value_column = "score", 
                                extend = ExtendSize)
HybHE.L.EH <- EnrichedHeatmap( mat = HybHE.L.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from above
                       column_title = "Proximal Peaks: HybHE Larva", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                         ylim=c(0,300))
                       )
) 

#now for HybHT:
proximalpeaks.HybHT.aveB<-left_join(proximalpeaks, ase_nolowcounts %>% select(Chr,start,end,HybHTB12,HybHTB13))
proximalpeaks.HybHT.aveB$HybHTB12<-cpm(proximalpeaks.HybHT.aveB$HybHTB12) 
proximalpeaks.HybHT.aveB$HybHTB13<-cpm(proximalpeaks.HybHT.aveB$HybHTB13) 
proximalpeaks.HybHT.aveB$aveB<-(proximalpeaks.HybHT.aveB$HybHTB12+proximalpeaks.HybHT.aveB$HybHTB13)/2
proximalpeaks.HybHT.aveB<-proximalpeaks.HybHT.aveB %>% select(-HybHTB12,-HybHTB13)
proximalpeaks.HybHT.aveB<-proximalpeaks.HybHT.aveB %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
proximalpeaks.HybHT.aveB<-proximalpeaks.HybHT.aveB %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveB),0L))
proximalpeaks.HybHT.aveB$overlap<-proximalpeaks.HybHT.aveB$start-proximalpeaks.HybHT.aveB$prevend
proximalpeaks.HybHT.aveB$overlap<-ifelse(lead(proximalpeaks.HybHT.aveB$overlap<0),"erase me",proximalpeaks.HybHT.aveB$overlap)
proximalpeaks.HybHT.aveB$start<-ifelse(proximalpeaks.HybHT.aveB$overlap<0,proximalpeaks.HybHT.aveB$prevstart,proximalpeaks.HybHT.aveB$start)
proximalpeaks.HybHT.aveB$aveB<-ifelse(proximalpeaks.HybHT.aveB$overlap<0,proximalpeaks.HybHT.aveB$aveB+proximalpeaks.HybHT.aveB$prevpeak,proximalpeaks.HybHT.aveB$aveB)
proximalpeaks.HybHT.aveB<-proximalpeaks.HybHT.aveB %>% filter(!overlap=="erase me")
proximalpeaks.HybHT.aveB<-proximalpeaks.HybHT.aveB %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(proximalpeaks.HybHT.aveB,"proximalpeaks.HybHT.aveB.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#/ We load the relevant parts of the bigwig file into R. 
#/ This is more efficient than reading the entire file. 
HybHT.B.BigWig <- rtracklayer::import("proximalpeaks.HybHT.aveB.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
#/ Create the normalizedMatrix that EnrichedHeatmap accepts as input.
#/ We use the targets center (width=1) because from what I understand normalizeMatrix
#/ does not allow to turn off its "extend" option. Therefore we trick it by simply
#/ providing the peak centers and then let the function extend it by our predefined window size.
HybHT.B.normMatrix <- normalizeToMatrix(signal = HybHT.B.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      #/ minimal value to the 99th percentile
                                target_ratio = 0,
                                mean_mode = "w0",       #/ see ?EnrichedHeatmap on other options
                                value_column = "score", #/ = the name of the 4th column of the bigwig
                                extend = ExtendSize)
#/ heatmap function:
HybHT.B.EH <- EnrichedHeatmap( mat = HybHT.B.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from hery blastula
                       column_title = "Proximal Peaks: HybHT x HybHT Blastula", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                           ylim=c(0,300))
                       )
) #/ end of EnrichedHeatmap function

#now repeat for other stages of HybHT
#gastrula:
proximalpeaks.HybHT.aveG<-left_join(proximalpeaks, ase_nolowcounts %>% select(Chr,start,end,HybHTG7,HybHTG8,HybHTG9))
proximalpeaks.HybHT.aveG$HybHTG7<-cpm(proximalpeaks.HybHT.aveG$HybHTG7)
proximalpeaks.HybHT.aveG$HybHTG8<-cpm(proximalpeaks.HybHT.aveG$HybHTG8) 
proximalpeaks.HybHT.aveG$HybHTG9<-cpm(proximalpeaks.HybHT.aveG$HybHTG9) 
proximalpeaks.HybHT.aveG$aveG<-(proximalpeaks.HybHT.aveG$HybHTG7+proximalpeaks.HybHT.aveG$HybHTG8+proximalpeaks.HybHT.aveG$HybHTG9)/3
proximalpeaks.HybHT.aveG<-proximalpeaks.HybHT.aveG %>% select(-HybHTG7,-HybHTG8,-HybHTG9)
proximalpeaks.HybHT.aveG<-proximalpeaks.HybHT.aveG %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
proximalpeaks.HybHT.aveG<-proximalpeaks.HybHT.aveG %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveG),0L))
proximalpeaks.HybHT.aveG$overlap<-proximalpeaks.HybHT.aveG$start-proximalpeaks.HybHT.aveG$prevend
proximalpeaks.HybHT.aveG$overlap<-ifelse(lead(proximalpeaks.HybHT.aveG$overlap<0),"erase me",proximalpeaks.HybHT.aveG$overlap)
proximalpeaks.HybHT.aveG$start<-ifelse(proximalpeaks.HybHT.aveG$overlap<0,proximalpeaks.HybHT.aveG$prevstart,proximalpeaks.HybHT.aveG$start)
proximalpeaks.HybHT.aveG$aveG<-ifelse(proximalpeaks.HybHT.aveG$overlap<0,proximalpeaks.HybHT.aveG$aveG+proximalpeaks.HybHT.aveG$prevpeak,proximalpeaks.HybHT.aveG$aveG)
proximalpeaks.HybHT.aveG<-proximalpeaks.HybHT.aveG %>% filter(!overlap=="erase me")
proximalpeaks.HybHT.aveG<-proximalpeaks.HybHT.aveG %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(proximalpeaks.HybHT.aveG,"proximalpeaks.HybHT.aveG.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#skip making target, it is the same as before
HybHT.G.BigWig <- rtracklayer::import("proximalpeaks.HybHT.aveG.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
HybHT.G.normMatrix <- normalizeToMatrix(signal = HybHT.G.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      
                                target_ratio = 0,
                                mean_mode = "w0",       
                                value_column = "score", 
                                extend = ExtendSize)
HybHT.G.EH <- EnrichedHeatmap( mat = HybHT.G.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from above
                       column_title = "Proximal Peaks: HybHT x HybHT Gastrula", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                           ylim=c(0,300))
                       )
) 

#larva:
proximalpeaks.HybHT.aveL<-left_join(proximalpeaks, ase_nolowcounts %>% select(Chr,start,end,HybHTL1,HybHTL2,HybHTL11))
proximalpeaks.HybHT.aveL$HybHTL1<-cpm(proximalpeaks.HybHT.aveL$HybHTL1)
proximalpeaks.HybHT.aveL$HybHTL2<-cpm(proximalpeaks.HybHT.aveL$HybHTL2) 
proximalpeaks.HybHT.aveL$HybHTL11<-cpm(proximalpeaks.HybHT.aveL$HybHTL11) 
proximalpeaks.HybHT.aveL$aveL<-(proximalpeaks.HybHT.aveL$HybHTL1+proximalpeaks.HybHT.aveL$HybHTL2+proximalpeaks.HybHT.aveL$HybHTL11)/3
proximalpeaks.HybHT.aveL<-proximalpeaks.HybHT.aveL %>% select(-HybHTL1,-HybHTL2,-HybHTL11)
proximalpeaks.HybHT.aveL<-proximalpeaks.HybHT.aveL %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
proximalpeaks.HybHT.aveL<-proximalpeaks.HybHT.aveL %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveL),0L))
proximalpeaks.HybHT.aveL$overlap<-proximalpeaks.HybHT.aveL$start-proximalpeaks.HybHT.aveL$prevend
proximalpeaks.HybHT.aveL$overlap<-ifelse(lead(proximalpeaks.HybHT.aveL$overlap<0),"erase me",proximalpeaks.HybHT.aveL$overlap)
proximalpeaks.HybHT.aveL$start<-ifelse(proximalpeaks.HybHT.aveL$overlap<0,proximalpeaks.HybHT.aveL$prevstart,proximalpeaks.HybHT.aveL$start)
proximalpeaks.HybHT.aveL$aveL<-ifelse(proximalpeaks.HybHT.aveL$overlap<0,proximalpeaks.HybHT.aveL$aveL+proximalpeaks.HybHT.aveL$prevpeak,proximalpeaks.HybHT.aveL$aveL)
proximalpeaks.HybHT.aveL<-proximalpeaks.HybHT.aveL %>% filter(!overlap=="erase me")
proximalpeaks.HybHT.aveL<-proximalpeaks.HybHT.aveL %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(proximalpeaks.HybHT.aveL,"proximalpeaks.HybHT.aveL.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#skip making target, it is the same as before
HybHT.L.BigWig <- rtracklayer::import("proximalpeaks.HybHT.aveL.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
HybHT.L.normMatrix <- normalizeToMatrix(signal = HybHT.L.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      
                                target_ratio = 0,
                                mean_mode = "w0",       
                                value_column = "score", 
                                extend = ExtendSize)
HybHT.L.EH <- EnrichedHeatmap( mat = HybHT.L.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from above
                       column_title = "Proximal Peaks: HybHT Larva", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                         ylim=c(0,300))
                       )
) 


pdf("HybHE.B.EH.proximal.pdf")
draw(HybHE.B.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

pdf("HybHE.G.EH.proximal.pdf")
draw(HybHE.G.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

pdf("HybHE.L.EH.proximal.pdf")
draw(HybHE.L.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

pdf("HybHT.B.EH.proximal.pdf")
draw(HybHT.B.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

pdf("HybHT.G.EH.proximal.pdf")
draw(HybHT.G.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

pdf("HybHT.L.EH.proximal.pdf")
draw(HybHT.L.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()
```
```{r}
#remake color scale and then go back and re-rerun all the plot making steps. this will normalize the colors across all grpahs.
common_min = min(c(Hery.B.normMatrix, Hery.G.normMatrix,Hery.L.normMatrix,htub.B.normMatrix, htub.G.normMatrix,htub.L.normMatrix,HybHE.B.normMatrix, HybHE.G.normMatrix,HybHE.L.normMatrix,HybHT.B.normMatrix, HybHT.G.normMatrix,HybHT.L.normMatrix))
common_max = max(c(Hery.B.normMatrix, Hery.G.normMatrix,Hery.L.normMatrix,htub.B.normMatrix, htub.G.normMatrix,htub.L.normMatrix,HybHE.B.normMatrix, HybHE.G.normMatrix,HybHE.L.normMatrix,HybHT.B.normMatrix, HybHT.G.normMatrix,HybHT.L.normMatrix))
col_fun = circlize::colorRamp2(c(common_min, common_max), c("darkblue", "darkgoldenrod1"))

```


Enriched heatmaps for distal peaks (Figure S2):



```{r}
#the package requires bed files and bedgraph files so I'll make a bed of distal peaks:
distalpeaks<-distalpeak_gene_map %>% select(PeakCoord, peak)
#makehomerbedD(distalpeaks)

#and i need a bedgraph which apparently is bed + counts. let's do blastula, Hery to start:
distalpeaks<-transform(distalpeaks, col=do.call(rbind, strsplit(PeakCoord, ':', fixed=TRUE)), stringsAsFactors=F) %>% select (-PeakCoord)
distalpeaks<-transform(distalpeaks, col=do.call(rbind, strsplit(col.2, '-', fixed=TRUE)), stringsAsFactors=F) %>% select(-col.2,-peak)
colnames(distalpeaks)<-c("Chr","start","end")
distalpeaks$start<-as.numeric(distalpeaks$start)
distalpeaks$end<-as.numeric(distalpeaks$end)

distalpeaks.Hery.aveB<-left_join(distalpeaks, ase_nolowcounts %>% select(Chr,start,end,HeryB4,HeryB7))
distalpeaks.Hery.aveB$HeryB4<-cpm(distalpeaks.Hery.aveB$HeryB4) #greg told me to do raw counts but if i want to combine samples i need to do normd counts (don't know a way to feed multiple bedgraphs into enrichedheatmap at once)
distalpeaks.Hery.aveB$HeryB7<-cpm(distalpeaks.Hery.aveB$HeryB7) 
distalpeaks.Hery.aveB$aveB<-(distalpeaks.Hery.aveB$HeryB4+distalpeaks.Hery.aveB$HeryB7)/2
distalpeaks.Hery.aveB<-distalpeaks.Hery.aveB %>% select(-HeryB4,-HeryB7)
distalpeaks.Hery.aveB<-distalpeaks.Hery.aveB %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
distalpeaks.Hery.aveB<-distalpeaks.Hery.aveB %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveB),0L))
distalpeaks.Hery.aveB$overlap<-distalpeaks.Hery.aveB$start-distalpeaks.Hery.aveB$prevend
distalpeaks.Hery.aveB$overlap<-ifelse(lead(distalpeaks.Hery.aveB$overlap<0),"erase me",distalpeaks.Hery.aveB$overlap)
distalpeaks.Hery.aveB$start<-ifelse(distalpeaks.Hery.aveB$overlap<0,distalpeaks.Hery.aveB$prevstart,distalpeaks.Hery.aveB$start)
distalpeaks.Hery.aveB$aveB<-ifelse(distalpeaks.Hery.aveB$overlap<0,distalpeaks.Hery.aveB$aveB+distalpeaks.Hery.aveB$prevpeak,distalpeaks.Hery.aveB$aveB)
distalpeaks.Hery.aveB<-distalpeaks.Hery.aveB %>% filter(!overlap=="erase me")
distalpeaks.Hery.aveB<-distalpeaks.Hery.aveB %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(distalpeaks.Hery.aveB,"distalpeaks.Hery.aveB.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
targets <- makeGRangesFromDataFrame(
  df = fread("distalpeaks.gff", header = FALSE, data.table = FALSE),
  seqnames.field = "V1", start.field = "V2", end.field = "V3")

#/ We take the center of each region/peak and want to extend by 5kb each direction:
ExtendSize <- 5000
targets.extended  <- resize(targets, fix = "center", width = ExtendSize*2)

#/ We load the relevant parts of the bigwig file into R. 
#/ This is more efficient than reading the entire file. 
Hery.B.BigWig <- rtracklayer::import("distalpeaks.Hery.aveB.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
#/ Create the normalizedMatrix that EnrichedHeatmap accepts as input.
#/ We use the targets center (width=1) because from what I understand normalizeMatrix
#/ does not allow to turn off its "extend" option. Therefore we trick it by simply
#/ providing the peak centers and then let the function extend it by our predefined window size.
Hery.B.normMatrix <- normalizeToMatrix(signal = Hery.B.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      #/ minimal value to the 99th percentile
                                target_ratio = 0,
                                mean_mode = "w0",       #/ see ?EnrichedHeatmap on other options
                                value_column = "score", #/ = the name of the 4th column of the bigwig
                                extend = ExtendSize)

#/ Make a color gradient that covers the range of normMatrix from 0 to the 99th percentile.
#/ The percentile avoids outliers to skew the heatmap:
col_fun = circlize::colorRamp2(quantile(Hery.B.normMatrix, c(0, .99)), c("darkblue", "darkgoldenrod1"))

#/ heatmap function:
Hery.B.EH <- EnrichedHeatmap( mat = Hery.B.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from above
                       column_title = "Distal Peaks: Hery x Hery Blastula", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                           ylim=c(0,50))
                       )
) #/ end of EnrichedHeatmap function


#now repeat for other stages of Hery
#gastrula:
distalpeaks.Hery.aveG<-left_join(distalpeaks, ase_nolowcounts %>% select(Chr,start,end,HeryG2,HeryG5,HeryG8))
distalpeaks.Hery.aveG$HeryG2<-cpm(distalpeaks.Hery.aveG$HeryG2)
distalpeaks.Hery.aveG$HeryG5<-cpm(distalpeaks.Hery.aveG$HeryG5) 
distalpeaks.Hery.aveG$HeryG8<-cpm(distalpeaks.Hery.aveG$HeryG8) 
distalpeaks.Hery.aveG$aveG<-(distalpeaks.Hery.aveG$HeryG2+distalpeaks.Hery.aveG$HeryG5+distalpeaks.Hery.aveG$HeryG8)/3
distalpeaks.Hery.aveG<-distalpeaks.Hery.aveG %>% select(-HeryG2,-HeryG5,-HeryG8)
distalpeaks.Hery.aveG<-distalpeaks.Hery.aveG %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
distalpeaks.Hery.aveG<-distalpeaks.Hery.aveG %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveG),0L))
distalpeaks.Hery.aveG$overlap<-distalpeaks.Hery.aveG$start-distalpeaks.Hery.aveG$prevend
distalpeaks.Hery.aveG$overlap<-ifelse(lead(distalpeaks.Hery.aveG$overlap<0),"erase me",distalpeaks.Hery.aveG$overlap)
distalpeaks.Hery.aveG$start<-ifelse(distalpeaks.Hery.aveG$overlap<0,distalpeaks.Hery.aveG$prevstart,distalpeaks.Hery.aveG$start)
distalpeaks.Hery.aveG$aveG<-ifelse(distalpeaks.Hery.aveG$overlap<0,distalpeaks.Hery.aveG$aveG+distalpeaks.Hery.aveG$prevpeak,distalpeaks.Hery.aveG$aveG)
distalpeaks.Hery.aveG<-distalpeaks.Hery.aveG %>% filter(!overlap=="erase me")
distalpeaks.Hery.aveG<-distalpeaks.Hery.aveG %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(distalpeaks.Hery.aveG,"distalpeaks.Hery.aveG.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#skip making target, it is the same as before
Hery.G.BigWig <- rtracklayer::import("distalpeaks.Hery.aveG.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
Hery.G.normMatrix <- normalizeToMatrix(signal = Hery.G.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      
                                target_ratio = 0,
                                mean_mode = "w0",       
                                value_column = "score", 
                                extend = ExtendSize)
Hery.G.EH <- EnrichedHeatmap( mat = Hery.G.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from blastula
                       column_title = "Distal Peaks: Hery x Hery Gastrula", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                           ylim=c(0,50))
                       )
) 

#larva:
distalpeaks.Hery.aveL<-left_join(distalpeaks, ase_nolowcounts %>% select(Chr,start,end,HeryL3,HeryL6,HeryL9))
distalpeaks.Hery.aveL$HeryL3<-cpm(distalpeaks.Hery.aveL$HeryL3)
distalpeaks.Hery.aveL$HeryL6<-cpm(distalpeaks.Hery.aveL$HeryL6) 
distalpeaks.Hery.aveL$HeryL9<-cpm(distalpeaks.Hery.aveL$HeryL9) 
distalpeaks.Hery.aveL$aveL<-(distalpeaks.Hery.aveL$HeryL3+distalpeaks.Hery.aveL$HeryL6+distalpeaks.Hery.aveL$HeryL9)/3
distalpeaks.Hery.aveL<-distalpeaks.Hery.aveL %>% select(-HeryL3,-HeryL6,-HeryL9)
distalpeaks.Hery.aveL<-distalpeaks.Hery.aveL %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
distalpeaks.Hery.aveL<-distalpeaks.Hery.aveL %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveL),0L))
distalpeaks.Hery.aveL$overlap<-distalpeaks.Hery.aveL$start-distalpeaks.Hery.aveL$prevend
distalpeaks.Hery.aveL$overlap<-ifelse(lead(distalpeaks.Hery.aveL$overlap<0),"erase me",distalpeaks.Hery.aveL$overlap)
distalpeaks.Hery.aveL$start<-ifelse(distalpeaks.Hery.aveL$overlap<0,distalpeaks.Hery.aveL$prevstart,distalpeaks.Hery.aveL$start)
distalpeaks.Hery.aveL$aveL<-ifelse(distalpeaks.Hery.aveL$overlap<0,distalpeaks.Hery.aveL$aveL+distalpeaks.Hery.aveL$prevpeak,distalpeaks.Hery.aveL$aveL)
distalpeaks.Hery.aveL<-distalpeaks.Hery.aveL %>% filter(!overlap=="erase me")
distalpeaks.Hery.aveL<-distalpeaks.Hery.aveL %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(distalpeaks.Hery.aveL,"distalpeaks.Hery.aveL.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#skip making target, it is the same as before
Hery.L.BigWig <- rtracklayer::import("distalpeaks.Hery.aveL.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
Hery.L.normMatrix <- normalizeToMatrix(signal = Hery.L.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      
                                target_ratio = 0,
                                mean_mode = "w0",       
                                value_column = "score", 
                                extend = ExtendSize)
Hery.L.EH <- EnrichedHeatmap( mat = Hery.L.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from blastula
                       column_title = "Distal Peaks: Hery x Hery Larva", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                           ylim=c(0,50))
                       )
) 

#now for HTUB:
distalpeaks.htub.aveB<-left_join(distalpeaks, ase_nolowcounts %>% select(Chr,start,end,htubB4,htubB7))
distalpeaks.htub.aveB$htubB4<-cpm(distalpeaks.htub.aveB$htubB4) 
distalpeaks.htub.aveB$htubB7<-cpm(distalpeaks.htub.aveB$htubB7) 
distalpeaks.htub.aveB$aveB<-(distalpeaks.htub.aveB$htubB4+distalpeaks.htub.aveB$htubB7)/2
distalpeaks.htub.aveB<-distalpeaks.htub.aveB %>% select(-htubB4,-htubB7)
distalpeaks.htub.aveB<-distalpeaks.htub.aveB %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
distalpeaks.htub.aveB<-distalpeaks.htub.aveB %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveB),0L))
distalpeaks.htub.aveB$overlap<-distalpeaks.htub.aveB$start-distalpeaks.htub.aveB$prevend
distalpeaks.htub.aveB$overlap<-ifelse(lead(distalpeaks.htub.aveB$overlap<0),"erase me",distalpeaks.htub.aveB$overlap)
distalpeaks.htub.aveB$start<-ifelse(distalpeaks.htub.aveB$overlap<0,distalpeaks.htub.aveB$prevstart,distalpeaks.htub.aveB$start)
distalpeaks.htub.aveB$aveB<-ifelse(distalpeaks.htub.aveB$overlap<0,distalpeaks.htub.aveB$aveB+distalpeaks.htub.aveB$prevpeak,distalpeaks.htub.aveB$aveB)
distalpeaks.htub.aveB<-distalpeaks.htub.aveB %>% filter(!overlap=="erase me")
distalpeaks.htub.aveB<-distalpeaks.htub.aveB %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(distalpeaks.htub.aveB,"distalpeaks.htub.aveB.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#/ We load the relevant parts of the bigwig file into R. 
#/ This is more efficient than reading the entire file. 
htub.B.BigWig <- rtracklayer::import("distalpeaks.htub.aveB.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
#/ Create the normalizedMatrix that EnrichedHeatmap accepts as input.
#/ We use the targets center (width=1) because from what I understand normalizeMatrix
#/ does not allow to turn off its "extend" option. Therefore we trick it by simply
#/ providing the peak centers and then let the function extend it by our predefined window size.
htub.B.normMatrix <- normalizeToMatrix(signal = htub.B.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      #/ minimal value to the 99th percentile
                                target_ratio = 0,
                                mean_mode = "w0",       #/ see ?EnrichedHeatmap on other options
                                value_column = "score", #/ = the name of the 4th column of the bigwig
                                extend = ExtendSize)
#/ heatmap function:
htub.B.EH <- EnrichedHeatmap( mat = htub.B.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from hery blastula
                       column_title = "Distal Peaks: htub x htub Blastula", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                           ylim=c(0,50))
                       )
) #/ end of EnrichedHeatmap function

#now repeat for other stages of htub
#gastrula:
distalpeaks.htub.aveG<-left_join(distalpeaks, ase_nolowcounts %>% select(Chr,start,end,htubG2,htubG5,htubG8))
distalpeaks.htub.aveG$htubG2<-cpm(distalpeaks.htub.aveG$htubG2)
distalpeaks.htub.aveG$htubG5<-cpm(distalpeaks.htub.aveG$htubG5) 
distalpeaks.htub.aveG$htubG8<-cpm(distalpeaks.htub.aveG$htubG8) 
distalpeaks.htub.aveG$aveG<-(distalpeaks.htub.aveG$htubG2+distalpeaks.htub.aveG$htubG5+distalpeaks.htub.aveG$htubG8)/3
distalpeaks.htub.aveG<-distalpeaks.htub.aveG %>% select(-htubG2,-htubG5,-htubG8)
distalpeaks.htub.aveG<-distalpeaks.htub.aveG %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
distalpeaks.htub.aveG<-distalpeaks.htub.aveG %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveG),0L))
distalpeaks.htub.aveG$overlap<-distalpeaks.htub.aveG$start-distalpeaks.htub.aveG$prevend
distalpeaks.htub.aveG$overlap<-ifelse(lead(distalpeaks.htub.aveG$overlap<0),"erase me",distalpeaks.htub.aveG$overlap)
distalpeaks.htub.aveG$start<-ifelse(distalpeaks.htub.aveG$overlap<0,distalpeaks.htub.aveG$prevstart,distalpeaks.htub.aveG$start)
distalpeaks.htub.aveG$aveG<-ifelse(distalpeaks.htub.aveG$overlap<0,distalpeaks.htub.aveG$aveG+distalpeaks.htub.aveG$prevpeak,distalpeaks.htub.aveG$aveG)
distalpeaks.htub.aveG<-distalpeaks.htub.aveG %>% filter(!overlap=="erase me")
distalpeaks.htub.aveG<-distalpeaks.htub.aveG %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(distalpeaks.htub.aveG,"distalpeaks.htub.aveG.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#skip making target, it is the same as before
htub.G.BigWig <- rtracklayer::import("distalpeaks.htub.aveG.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
htub.G.normMatrix <- normalizeToMatrix(signal = htub.G.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      
                                target_ratio = 0,
                                mean_mode = "w0",       
                                value_column = "score", 
                                extend = ExtendSize)
htub.G.EH <- EnrichedHeatmap( mat = htub.G.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from above
                       column_title = "Distal Peaks: htub x htub Gastrula", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                           ylim=c(0,50))
                       )
) 

#larva:
distalpeaks.htub.aveL<-left_join(distalpeaks, ase_nolowcounts %>% select(Chr,start,end,htubL3,htubL6,htubL11))
distalpeaks.htub.aveL$htubL3<-cpm(distalpeaks.htub.aveL$htubL3)
distalpeaks.htub.aveL$htubL6<-cpm(distalpeaks.htub.aveL$htubL6) 
distalpeaks.htub.aveL$htubL11<-cpm(distalpeaks.htub.aveL$htubL11) 
distalpeaks.htub.aveL$aveL<-(distalpeaks.htub.aveL$htubL3+distalpeaks.htub.aveL$htubL6+distalpeaks.htub.aveL$htubL11)/3
distalpeaks.htub.aveL<-distalpeaks.htub.aveL %>% select(-htubL3,-htubL6,-htubL11)
distalpeaks.htub.aveL<-distalpeaks.htub.aveL %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
distalpeaks.htub.aveL<-distalpeaks.htub.aveL %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveL),0L))
distalpeaks.htub.aveL$overlap<-distalpeaks.htub.aveL$start-distalpeaks.htub.aveL$prevend
distalpeaks.htub.aveL$overlap<-ifelse(lead(distalpeaks.htub.aveL$overlap<0),"erase me",distalpeaks.htub.aveL$overlap)
distalpeaks.htub.aveL$start<-ifelse(distalpeaks.htub.aveL$overlap<0,distalpeaks.htub.aveL$prevstart,distalpeaks.htub.aveL$start)
distalpeaks.htub.aveL$aveL<-ifelse(distalpeaks.htub.aveL$overlap<0,distalpeaks.htub.aveL$aveL+distalpeaks.htub.aveL$prevpeak,distalpeaks.htub.aveL$aveL)
distalpeaks.htub.aveL<-distalpeaks.htub.aveL %>% filter(!overlap=="erase me")
distalpeaks.htub.aveL<-distalpeaks.htub.aveL %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(distalpeaks.htub.aveL,"distalpeaks.htub.aveL.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#skip making target, it is the same as before
htub.L.BigWig <- rtracklayer::import("distalpeaks.htub.aveL.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
htub.L.normMatrix <- normalizeToMatrix(signal = htub.L.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      
                                target_ratio = 0,
                                mean_mode = "w0",       
                                value_column = "score", 
                                extend = ExtendSize)
htub.L.EH <- EnrichedHeatmap( mat = htub.L.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from above
                       column_title = "Distal Peaks: Htub x Htub Larva", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                         ylim=c(0,50))
                       )
) 

pdf("Hery.B.EH.distal.pdf")
draw(Hery.B.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

pdf("Hery.G.EH.distal.pdf")
draw(Hery.G.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

pdf("Hery.L.EH.distal.pdf")
draw(Hery.L.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

pdf("htub.B.EH.distal.pdf")
draw(htub.B.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

pdf("htub.G.EH.distal.pdf")
draw(htub.G.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

pdf("htub.L.EH.distal.pdf")
draw(htub.L.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

#now for HybHE:
distalpeaks.HybHE.aveB<-left_join(distalpeaks, ase_nolowcounts %>% select(Chr,start,end,HybHEB12,HybHEB13))
distalpeaks.HybHE.aveB$HybHEB12<-cpm(distalpeaks.HybHE.aveB$HybHEB12) 
distalpeaks.HybHE.aveB$HybHEB13<-cpm(distalpeaks.HybHE.aveB$HybHEB13) 
distalpeaks.HybHE.aveB$aveB<-(distalpeaks.HybHE.aveB$HybHEB12+distalpeaks.HybHE.aveB$HybHEB13)/2
distalpeaks.HybHE.aveB<-distalpeaks.HybHE.aveB %>% select(-HybHEB12,-HybHEB13)
distalpeaks.HybHE.aveB<-distalpeaks.HybHE.aveB %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
distalpeaks.HybHE.aveB<-distalpeaks.HybHE.aveB %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveB),0L))
distalpeaks.HybHE.aveB$overlap<-distalpeaks.HybHE.aveB$start-distalpeaks.HybHE.aveB$prevend
distalpeaks.HybHE.aveB$overlap<-ifelse(lead(distalpeaks.HybHE.aveB$overlap<0),"erase me",distalpeaks.HybHE.aveB$overlap)
distalpeaks.HybHE.aveB$start<-ifelse(distalpeaks.HybHE.aveB$overlap<0,distalpeaks.HybHE.aveB$prevstart,distalpeaks.HybHE.aveB$start)
distalpeaks.HybHE.aveB$aveB<-ifelse(distalpeaks.HybHE.aveB$overlap<0,distalpeaks.HybHE.aveB$aveB+distalpeaks.HybHE.aveB$prevpeak,distalpeaks.HybHE.aveB$aveB)
distalpeaks.HybHE.aveB<-distalpeaks.HybHE.aveB %>% filter(!overlap=="erase me")
distalpeaks.HybHE.aveB<-distalpeaks.HybHE.aveB %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(distalpeaks.HybHE.aveB,"distalpeaks.HybHE.aveB.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#/ We load the relevant parts of the bigwig file into R. 
#/ This is more efficient than reading the entire file. 
HybHE.B.BigWig <- rtracklayer::import("distalpeaks.HybHE.aveB.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
#/ Create the normalizedMatrix that EnrichedHeatmap accepts as input.
#/ We use the targets center (width=1) because from what I understand normalizeMatrix
#/ does not allow to turn off its "extend" option. Therefore we trick it by simply
#/ providing the peak centers and then let the function extend it by our predefined window size.
HybHE.B.normMatrix <- normalizeToMatrix(signal = HybHE.B.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      #/ minimal value to the 99th percentile
                                target_ratio = 0,
                                mean_mode = "w0",       #/ see ?EnrichedHeatmap on other options
                                value_column = "score", #/ = the name of the 4th column of the bigwig
                                extend = ExtendSize)
#/ heatmap function:
HybHE.B.EH <- EnrichedHeatmap( mat = HybHE.B.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from hery blastula
                       column_title = "Distal Peaks: HybHE x HybHE Blastula", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                           ylim=c(0,50))
                       )
) #/ end of EnrichedHeatmap function

#now repeat for other stages of HybHE
#gastrula:
distalpeaks.HybHE.aveG<-left_join(distalpeaks, ase_nolowcounts %>% select(Chr,start,end,HybHEG7,HybHEG8,HybHEG9))
distalpeaks.HybHE.aveG$HybHEG7<-cpm(distalpeaks.HybHE.aveG$HybHEG7)
distalpeaks.HybHE.aveG$HybHEG8<-cpm(distalpeaks.HybHE.aveG$HybHEG8) 
distalpeaks.HybHE.aveG$HybHEG9<-cpm(distalpeaks.HybHE.aveG$HybHEG9) 
distalpeaks.HybHE.aveG$aveG<-(distalpeaks.HybHE.aveG$HybHEG7+distalpeaks.HybHE.aveG$HybHEG8+distalpeaks.HybHE.aveG$HybHEG9)/3
distalpeaks.HybHE.aveG<-distalpeaks.HybHE.aveG %>% select(-HybHEG7,-HybHEG8,-HybHEG9)
distalpeaks.HybHE.aveG<-distalpeaks.HybHE.aveG %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
distalpeaks.HybHE.aveG<-distalpeaks.HybHE.aveG %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveG),0L))
distalpeaks.HybHE.aveG$overlap<-distalpeaks.HybHE.aveG$start-distalpeaks.HybHE.aveG$prevend
distalpeaks.HybHE.aveG$overlap<-ifelse(lead(distalpeaks.HybHE.aveG$overlap<0),"erase me",distalpeaks.HybHE.aveG$overlap)
distalpeaks.HybHE.aveG$start<-ifelse(distalpeaks.HybHE.aveG$overlap<0,distalpeaks.HybHE.aveG$prevstart,distalpeaks.HybHE.aveG$start)
distalpeaks.HybHE.aveG$aveG<-ifelse(distalpeaks.HybHE.aveG$overlap<0,distalpeaks.HybHE.aveG$aveG+distalpeaks.HybHE.aveG$prevpeak,distalpeaks.HybHE.aveG$aveG)
distalpeaks.HybHE.aveG<-distalpeaks.HybHE.aveG %>% filter(!overlap=="erase me")
distalpeaks.HybHE.aveG<-distalpeaks.HybHE.aveG %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(distalpeaks.HybHE.aveG,"distalpeaks.HybHE.aveG.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#skip making target, it is the same as before
HybHE.G.BigWig <- rtracklayer::import("distalpeaks.HybHE.aveG.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
HybHE.G.normMatrix <- normalizeToMatrix(signal = HybHE.G.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      
                                target_ratio = 0,
                                mean_mode = "w0",       
                                value_column = "score", 
                                extend = ExtendSize)
HybHE.G.EH <- EnrichedHeatmap( mat = HybHE.G.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from above
                       column_title = "Distal Peaks: HybHE x HybHE Gastrula", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                           ylim=c(0,50))
                       )
) 

#larva:
distalpeaks.HybHE.aveL<-left_join(distalpeaks, ase_nolowcounts %>% select(Chr,start,end,HybHEL1,HybHEL2,HybHEL11))
distalpeaks.HybHE.aveL$HybHEL1<-cpm(distalpeaks.HybHE.aveL$HybHEL1)
distalpeaks.HybHE.aveL$HybHEL2<-cpm(distalpeaks.HybHE.aveL$HybHEL2) 
distalpeaks.HybHE.aveL$HybHEL11<-cpm(distalpeaks.HybHE.aveL$HybHEL11) 
distalpeaks.HybHE.aveL$aveL<-(distalpeaks.HybHE.aveL$HybHEL1+distalpeaks.HybHE.aveL$HybHEL2+distalpeaks.HybHE.aveL$HybHEL11)/3
distalpeaks.HybHE.aveL<-distalpeaks.HybHE.aveL %>% select(-HybHEL1,-HybHEL2,-HybHEL11)
distalpeaks.HybHE.aveL<-distalpeaks.HybHE.aveL %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
distalpeaks.HybHE.aveL<-distalpeaks.HybHE.aveL %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveL),0L))
distalpeaks.HybHE.aveL$overlap<-distalpeaks.HybHE.aveL$start-distalpeaks.HybHE.aveL$prevend
distalpeaks.HybHE.aveL$overlap<-ifelse(lead(distalpeaks.HybHE.aveL$overlap<0),"erase me",distalpeaks.HybHE.aveL$overlap)
distalpeaks.HybHE.aveL$start<-ifelse(distalpeaks.HybHE.aveL$overlap<0,distalpeaks.HybHE.aveL$prevstart,distalpeaks.HybHE.aveL$start)
distalpeaks.HybHE.aveL$aveL<-ifelse(distalpeaks.HybHE.aveL$overlap<0,distalpeaks.HybHE.aveL$aveL+distalpeaks.HybHE.aveL$prevpeak,distalpeaks.HybHE.aveL$aveL)
distalpeaks.HybHE.aveL<-distalpeaks.HybHE.aveL %>% filter(!overlap=="erase me")
distalpeaks.HybHE.aveL<-distalpeaks.HybHE.aveL %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(distalpeaks.HybHE.aveL,"distalpeaks.HybHE.aveL.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#skip making target, it is the same as before
HybHE.L.BigWig <- rtracklayer::import("distalpeaks.HybHE.aveL.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
HybHE.L.normMatrix <- normalizeToMatrix(signal = HybHE.L.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      
                                target_ratio = 0,
                                mean_mode = "w0",       
                                value_column = "score", 
                                extend = ExtendSize)
HybHE.L.EH <- EnrichedHeatmap( mat = HybHE.L.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from above
                       column_title = "Distal Peaks: HybHE Larva", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                         ylim=c(0,50))
                       )
) 

#now for HybHT:
distalpeaks.HybHT.aveB<-left_join(distalpeaks, ase_nolowcounts %>% select(Chr,start,end,HybHTB12,HybHTB13))
distalpeaks.HybHT.aveB$HybHTB12<-cpm(distalpeaks.HybHT.aveB$HybHTB12) 
distalpeaks.HybHT.aveB$HybHTB13<-cpm(distalpeaks.HybHT.aveB$HybHTB13) 
distalpeaks.HybHT.aveB$aveB<-(distalpeaks.HybHT.aveB$HybHTB12+distalpeaks.HybHT.aveB$HybHTB13)/2
distalpeaks.HybHT.aveB<-distalpeaks.HybHT.aveB %>% select(-HybHTB12,-HybHTB13)
distalpeaks.HybHT.aveB<-distalpeaks.HybHT.aveB %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
distalpeaks.HybHT.aveB<-distalpeaks.HybHT.aveB %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveB),0L))
distalpeaks.HybHT.aveB$overlap<-distalpeaks.HybHT.aveB$start-distalpeaks.HybHT.aveB$prevend
distalpeaks.HybHT.aveB$overlap<-ifelse(lead(distalpeaks.HybHT.aveB$overlap<0),"erase me",distalpeaks.HybHT.aveB$overlap)
distalpeaks.HybHT.aveB$start<-ifelse(distalpeaks.HybHT.aveB$overlap<0,distalpeaks.HybHT.aveB$prevstart,distalpeaks.HybHT.aveB$start)
distalpeaks.HybHT.aveB$aveB<-ifelse(distalpeaks.HybHT.aveB$overlap<0,distalpeaks.HybHT.aveB$aveB+distalpeaks.HybHT.aveB$prevpeak,distalpeaks.HybHT.aveB$aveB)
distalpeaks.HybHT.aveB<-distalpeaks.HybHT.aveB %>% filter(!overlap=="erase me")
distalpeaks.HybHT.aveB<-distalpeaks.HybHT.aveB %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(distalpeaks.HybHT.aveB,"distalpeaks.HybHT.aveB.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#/ We load the relevant parts of the bigwig file into R. 
#/ This is more efficient than reading the entire file. 
HybHT.B.BigWig <- rtracklayer::import("distalpeaks.HybHT.aveB.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
#/ Create the normalizedMatrix that EnrichedHeatmap accepts as input.
#/ We use the targets center (width=1) because from what I understand normalizeMatrix
#/ does not allow to turn off its "extend" option. Therefore we trick it by simply
#/ providing the peak centers and then let the function extend it by our predefined window size.
HybHT.B.normMatrix <- normalizeToMatrix(signal = HybHT.B.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      #/ minimal value to the 99th percentile
                                target_ratio = 0,
                                mean_mode = "w0",       #/ see ?EnrichedHeatmap on other options
                                value_column = "score", #/ = the name of the 4th column of the bigwig
                                extend = ExtendSize)
#/ heatmap function:
HybHT.B.EH <- EnrichedHeatmap( mat = HybHT.B.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from hery blastula
                       column_title = "Distal Peaks: HybHT x HybHT Blastula", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                           ylim=c(0,50))
                       )
) #/ end of EnrichedHeatmap function

#now repeat for other stages of HybHT
#gastrula:
distalpeaks.HybHT.aveG<-left_join(distalpeaks, ase_nolowcounts %>% select(Chr,start,end,HybHTG7,HybHTG8,HybHTG9))
distalpeaks.HybHT.aveG$HybHTG7<-cpm(distalpeaks.HybHT.aveG$HybHTG7)
distalpeaks.HybHT.aveG$HybHTG8<-cpm(distalpeaks.HybHT.aveG$HybHTG8) 
distalpeaks.HybHT.aveG$HybHTG9<-cpm(distalpeaks.HybHT.aveG$HybHTG9) 
distalpeaks.HybHT.aveG$aveG<-(distalpeaks.HybHT.aveG$HybHTG7+distalpeaks.HybHT.aveG$HybHTG8+distalpeaks.HybHT.aveG$HybHTG9)/3
distalpeaks.HybHT.aveG<-distalpeaks.HybHT.aveG %>% select(-HybHTG7,-HybHTG8,-HybHTG9)
distalpeaks.HybHT.aveG<-distalpeaks.HybHT.aveG %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
distalpeaks.HybHT.aveG<-distalpeaks.HybHT.aveG %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveG),0L))
distalpeaks.HybHT.aveG$overlap<-distalpeaks.HybHT.aveG$start-distalpeaks.HybHT.aveG$prevend
distalpeaks.HybHT.aveG$overlap<-ifelse(lead(distalpeaks.HybHT.aveG$overlap<0),"erase me",distalpeaks.HybHT.aveG$overlap)
distalpeaks.HybHT.aveG$start<-ifelse(distalpeaks.HybHT.aveG$overlap<0,distalpeaks.HybHT.aveG$prevstart,distalpeaks.HybHT.aveG$start)
distalpeaks.HybHT.aveG$aveG<-ifelse(distalpeaks.HybHT.aveG$overlap<0,distalpeaks.HybHT.aveG$aveG+distalpeaks.HybHT.aveG$prevpeak,distalpeaks.HybHT.aveG$aveG)
distalpeaks.HybHT.aveG<-distalpeaks.HybHT.aveG %>% filter(!overlap=="erase me")
distalpeaks.HybHT.aveG<-distalpeaks.HybHT.aveG %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(distalpeaks.HybHT.aveG,"distalpeaks.HybHT.aveG.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#skip making target, it is the same as before
HybHT.G.BigWig <- rtracklayer::import("distalpeaks.HybHT.aveG.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
HybHT.G.normMatrix <- normalizeToMatrix(signal = HybHT.G.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      
                                target_ratio = 0,
                                mean_mode = "w0",       
                                value_column = "score", 
                                extend = ExtendSize)
HybHT.G.EH <- EnrichedHeatmap( mat = HybHT.G.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from above
                       column_title = "Distal Peaks: HybHT x HybHT Gastrula", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                           ylim=c(0,50))
                       )
) 

#larva:
distalpeaks.HybHT.aveL<-left_join(distalpeaks, ase_nolowcounts %>% select(Chr,start,end,HybHTL1,HybHTL2,HybHTL11))
distalpeaks.HybHT.aveL$HybHTL1<-cpm(distalpeaks.HybHT.aveL$HybHTL1)
distalpeaks.HybHT.aveL$HybHTL2<-cpm(distalpeaks.HybHT.aveL$HybHTL2) 
distalpeaks.HybHT.aveL$HybHTL11<-cpm(distalpeaks.HybHT.aveL$HybHTL11) 
distalpeaks.HybHT.aveL$aveL<-(distalpeaks.HybHT.aveL$HybHTL1+distalpeaks.HybHT.aveL$HybHTL2+distalpeaks.HybHT.aveL$HybHTL11)/3
distalpeaks.HybHT.aveL<-distalpeaks.HybHT.aveL %>% select(-HybHTL1,-HybHTL2,-HybHTL11)
distalpeaks.HybHT.aveL<-distalpeaks.HybHT.aveL %>% arrange(Chr,start,end)
#cleaning up stupid overlapping peaks (no clue why these are here):
distalpeaks.HybHT.aveL<-distalpeaks.HybHT.aveL %>% mutate(prevstart=ifelse(Chr==lag(Chr),lag(start),0L)) %>% mutate(prevend=ifelse(Chr==lag(Chr),lag(end),0L)) %>% mutate(prevpeak=ifelse(Chr==lag(Chr),lag(aveL),0L))
distalpeaks.HybHT.aveL$overlap<-distalpeaks.HybHT.aveL$start-distalpeaks.HybHT.aveL$prevend
distalpeaks.HybHT.aveL$overlap<-ifelse(lead(distalpeaks.HybHT.aveL$overlap<0),"erase me",distalpeaks.HybHT.aveL$overlap)
distalpeaks.HybHT.aveL$start<-ifelse(distalpeaks.HybHT.aveL$overlap<0,distalpeaks.HybHT.aveL$prevstart,distalpeaks.HybHT.aveL$start)
distalpeaks.HybHT.aveL$aveL<-ifelse(distalpeaks.HybHT.aveL$overlap<0,distalpeaks.HybHT.aveL$aveL+distalpeaks.HybHT.aveL$prevpeak,distalpeaks.HybHT.aveL$aveL)
distalpeaks.HybHT.aveL<-distalpeaks.HybHT.aveL %>% filter(!overlap=="erase me")
distalpeaks.HybHT.aveL<-distalpeaks.HybHT.aveL %>% select(-prevstart,-prevend,-prevpeak,-overlap)
write.table(distalpeaks.HybHT.aveL,"distalpeaks.HybHT.aveL.bg",quote=F,row.names=F, col.names=F,sep="\t") #this is the bedgraph, upload to hardac to get bigwig using bedGraphtoBigWig
#back to enrichedheatmap:
#skip making target, it is the same as before
HybHT.L.BigWig <- rtracklayer::import("distalpeaks.HybHT.aveL.sorted.bw", 
                              format = "BigWig", 
                              selection = BigWigSelection(targets.extended))
HybHT.L.normMatrix <- normalizeToMatrix(signal = HybHT.L.BigWig, 
                                target = resize(targets, fix = "center", width = 1), 
                                background = 0, 
                                keep = c(0, 0.99),      
                                target_ratio = 0,
                                mean_mode = "w0",       
                                value_column = "score", 
                                extend = ExtendSize)
HybHT.L.EH <- EnrichedHeatmap( mat = HybHT.L.normMatrix, 
                       pos_line = FALSE, #/ no dashed lines around the start
                       border = FALSE,   #/ no box around heatmap
                       col = col_fun,    #/ color gradients from above
                       column_title = "Distal Peaks: HybHT Larva", #/ column title 
                       column_title_gp = gpar(fontsize = 15, fontfamily = "sans"),
                       use_raster = TRUE, raster_quality = 10, raster_device = "png",
                       #/ turn off background colors
                       rect_gp = gpar(col = "transparent"), 
                       #/ legend options:
                       heatmap_legend_param = list(
                         legend_direction = "horizontal",
                         title = "normalized counts"),
                       #/ options for the profile plot on top of the heatmap:
                       top_annotation = HeatmapAnnotation(
                         enriched = anno_enriched(
                           gp = gpar(col = "black", lty = 1, lwd=2),
                           col="black",
                         ylim=c(0,50))
                       )
) 

pdf("HybHE.B.EH.distal.pdf")
draw(HybHE.B.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

pdf("HybHE.G.EH.distal.pdf")
draw(HybHE.G.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

pdf("HybHE.L.EH.distal.pdf")
draw(HybHE.L.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

pdf("HybHT.B.EH.distal.pdf")
draw(HybHT.B.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

pdf("HybHT.G.EH.distal.pdf")
draw(HybHT.G.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()

pdf("HybHT.L.EH.distal.pdf")
draw(HybHT.L.EH,                                
     heatmap_legend_side = "bottom",     
     annotation_legend_side = "bottom",  
     padding = unit(c(4, 4, 4, 4), "mm") 
)
dev.off()
```

IGV-style tracks for real examples of "conserved" (Fugre 2D), "cis" (Figure 4C) and "trans" (Figure 4D) peaks 
```{r}
library(patchwork)
combinedHybGSE_bbsplit_fHEnoMT_reresorted_q5.reciplifted.bedgraph<-read.table("combinedHybGSE_bbsplit_fHEnoMT_reresorted_q5.reciplifted.bedgraph")
colnames(combinedHybGSE_bbsplit_fHEnoMT_reresorted_q5.reciplifted.bedgraph)<-c("chr","start","end","count")
combinedHybGSE_bbsplit_fHT_reresorted_q5.liftedtofHE.bedgraph<-read.table("combinedHybGSE_bbsplit_fHT_reresorted_q5.liftedtofHE.bedgraph")
colnames(combinedHybGSE_bbsplit_fHT_reresorted_q5.liftedtofHE.bedgraph)<-c("chr","start","end","count")
combinedHeryGSE_bbsplit_andAMBIG_fHEnoMT_reresorted_q5.reciplifted.bedgraph<-read.table("combinedHeryGSE_bbsplit_andAMBIG_fHEnoMT_reresorted_q5.reciplifted.bedgraph")
colnames(combinedHeryGSE_bbsplit_andAMBIG_fHEnoMT_reresorted_q5.reciplifted.bedgraph)<-c("chr","start","end","count")
combinedHtubGSE_bbsplit_andAMBIG_fHT_reresorted_q5.liftedtofHE.bedgraph<-read.table("combinedHtubGSE_bbsplit_andAMBIG_fHT_reresorted_q5.liftedtofHE.bedgraph")
colnames(combinedHtubGSE_bbsplit_andAMBIG_fHT_reresorted_q5.liftedtofHE.bedgraph)<-c("chr","start","end","count")

#write a function to pull up any region
combinedHeryGSE_bbsplit_andAMBIG_fHEnoMT_reresorted_q5.reciplifted.bedgraph$chr<-gsub("chr","",combinedHeryGSE_bbsplit_andAMBIG_fHEnoMT_reresorted_q5.reciplifted.bedgraph$chr)
combinedHeryGSE_bbsplit_andAMBIG_fHEnoMT_reresorted_q5.reciplifted.bedgraph$chr<-as.numeric(combinedHeryGSE_bbsplit_andAMBIG_fHEnoMT_reresorted_q5.reciplifted.bedgraph$chr)

combinedHybGSE_bbsplit_fHEnoMT_reresorted_q5.reciplifted.bedgraph$chr<-gsub("chr","",combinedHybGSE_bbsplit_fHEnoMT_reresorted_q5.reciplifted.bedgraph$chr)
combinedHybGSE_bbsplit_fHEnoMT_reresorted_q5.reciplifted.bedgraph$chr<-as.numeric(combinedHybGSE_bbsplit_fHEnoMT_reresorted_q5.reciplifted.bedgraph$chr)

combinedHybGSE_bbsplit_fHT_reresorted_q5.liftedtofHE.bedgraph$chr<-gsub("chr","",combinedHybGSE_bbsplit_fHT_reresorted_q5.liftedtofHE.bedgraph$chr)
combinedHybGSE_bbsplit_fHT_reresorted_q5.liftedtofHE.bedgraph$chr<-as.numeric(combinedHybGSE_bbsplit_fHT_reresorted_q5.liftedtofHE.bedgraph$chr)

combinedHtubGSE_bbsplit_andAMBIG_fHT_reresorted_q5.liftedtofHE.bedgraph$chr<-gsub("chr","",combinedHtubGSE_bbsplit_andAMBIG_fHT_reresorted_q5.liftedtofHE.bedgraph$chr)
combinedHtubGSE_bbsplit_andAMBIG_fHT_reresorted_q5.liftedtofHE.bedgraph$chr<-as.numeric(combinedHtubGSE_bbsplit_andAMBIG_fHT_reresorted_q5.liftedtofHE.bedgraph$chr)

custom_atactrack<-
  function(customchr,customstart,customend){
  Hery_track_height<-max(combinedHeryGSE_bbsplit_andAMBIG_fHEnoMT_reresorted_q5.reciplifted.bedgraph %>% filter(chr==customchr) %>% filter(start>=customstart) %>% filter(end<=customend) %>% select(count))
  HybHE_track_height<-max(combinedHybGSE_bbsplit_fHEnoMT_reresorted_q5.reciplifted.bedgraph %>% filter(chr==customchr) %>% filter(start>=customstart) %>% filter(end<=customend) %>% select(count))
  HybHT_track_height<-max(combinedHybGSE_bbsplit_fHT_reresorted_q5.liftedtofHE.bedgraph %>% filter(chr==customchr) %>% filter(start>=customstart) %>% filter(end<=customend) %>% select(count))
  Htub_track_height<-max(combinedHtubGSE_bbsplit_andAMBIG_fHT_reresorted_q5.liftedtofHE.bedgraph %>% filter(chr==customchr) %>% filter(start>=customstart) %>% filter(end<=customend) %>% select(count))
  allmax<-max(c(Hery_track_height,HybHE_track_height,HybHT_track_height,Htub_track_height))
  
  combinedHery_G<-ggplot(combinedHeryGSE_bbsplit_andAMBIG_fHEnoMT_reresorted_q5.reciplifted.bedgraph %>% filter(chr==customchr))+
  stat_smooth(geom="area",aes(start,count),fill="orange",method="loess",span=.2)+
  xlim(customstart,customend)+
  ylim(0,allmax)+
  theme_classic()

  combinedHybHE_G<-ggplot(combinedHybGSE_bbsplit_fHEnoMT_reresorted_q5.reciplifted.bedgraph %>% filter(chr==customchr))+
  stat_smooth(geom="area",aes(start,count),fill="orange", alpha=.5,method="loess",span=.15)+
  xlim(customstart,customend)+
  ylim(0,allmax)+
  theme_classic()
      
  combinedHybHT_G<-ggplot(combinedHybGSE_bbsplit_fHT_reresorted_q5.liftedtofHE.bedgraph %>% filter(chr==customchr))+
  stat_smooth(geom="area",aes(start,count),fill="darkgreen", alpha=.5,method="loess",span=.15)+
  xlim(customstart,customend)+
  ylim(0,allmax)+
  theme_classic()
  
    combinedHtub_G<-ggplot(combinedHtubGSE_bbsplit_andAMBIG_fHT_reresorted_q5.liftedtofHE.bedgraph %>% filter(chr==customchr))+
  stat_smooth(geom="area",aes(start,count),fill="darkgreen",method="loess",span=.15)+
  xlim(customstart,customend)+
  ylim(0,allmax)+
  theme_classic()
  
print(combinedHery_G / 
combinedHybHE_G /
combinedHybHT_G /
combinedHtub_G)

  }
#conserved
conservedplot<-custom_atactrack(
15,5243273,5243923
)
#ggsave("realconservedexample.svg",conservedplot,width=7,height=7) #Figure 2D

#cis
cisplot<-custom_atactrack(5,33066580,33066913)
#ggsave("realcisexample.svg",cisplot,width=7,height=7) #Figure 4C

#trans
transplot<-custom_atactrack(16,11534048,11534545)
#ggsave("realtransexample.svg",transplot,width=7,height=7) #Figure 4D

```
```{r}
#save.image(file='hybrids_forpublication.RData') #do not do this unless you are sure!
load(file='hybrids_forpublication.RData')
```

